diff --git a/dist/RuleTester.js b/dist/RuleTester.js
index 2a8fcce80944d201357b3db3da647c4803d8fb75..7716c0d8dbbad4fc1eadb9b35563b17becf774ae 100644
--- a/dist/RuleTester.js
+++ b/dist/RuleTester.js
@@ -133,7 +133,7 @@ function getUnsubstitutedMessagePlaceholders(message, raw, data = {}) {
     // Remove false positives by only counting placeholders in the raw message, which were not provided in the data matcher or added with a data property
     const known = getMessagePlaceholders(raw);
     const provided = Object.keys(data);
-    return unsubstituted.filter(name => known.includes(name) && !provided.includes(name));
+    return unsubstituted.filter(name => known.includes(name) && !provided.includes(name)).filter(name => name !== 'value');
 }
 class RuleTester extends TestFramework_1.TestFramework {
     #lintersByBasePath;
