<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/component-class-suffix.ts
  - ../../tests/rules/component-class-suffix/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/component-class-suffix`

Classes decorated with @Component must have suffix "Component" (or custom) in their name. See more at https://angular.io/styleguide#style-02-03

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * Default: `["Component"]`
   */
  suffixes?: string[];
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```ts
@Component({
  selector: 'sg-foo-bar'
})
class Test {}
      ~~~~
```

```ts
@Component({
  selector: 'sgBarFoo'
})
class TestPage {}
      ~~~~~~~~
```

```ts
@Component({
  selector: 'sgBarFoo'
})
class TestPage {}
      ~~~~~~~~
```

```ts
@Component({
  selector: 'sgBarFoo'
})
class TestDirective {}
      ~~~~~~~~~~~~~
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```ts
@Component({
  selector: 'sg-foo-bar',
  template: '<foo-bar [foo]="bar">{{baz + 42}}</foo-bar>'
})
class TestComponent {}
```

```ts
@Directive({
  selector: '[myHighlight]'
})
class TestDirective {}
```

```ts
@Pipe({
  selector: 'sg-test-pipe'
})
class TestPipe {}
```

```ts
@Injectable()
class TestService {}
```

```ts
class TestEmpty {}
```
