<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/computed-must-return.ts
  - ../../tests/rules/computed-must-return/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/computed-must-return`

Ensures that computed() returns a value. Omitting the value is likely a mistake.

- Type: problem

<br>

## Rationale

`computed()` is used to transform signal values. If no value is returned from `computed()`, it's likely a mistake.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
computed(() => {});
~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
computed(() => {
~~~~~~~~~~~~~~~~
  y();
  ~~~~
});
~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
computed(function() {
~~~~~~~~~~~~~~~~~~~~~
  y();
  ~~~
});
~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
computed(() => {
~~~~~~~~~~~~~~~~
  return;
  ~~~~~~~
});
~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
computed(function() {
~~~~~~~~~~~~~~~~~~~~~
  return;
  ~~~~~~~
});
~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test {
  x = computed(() => {
      ~~~~~~~~~~~~~~~~
    y();
    ~~~
  });
  ~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
computed(() => {}, { equal: (a, b) => {} });
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
computed(() => {
~~~~~~~~~~~~~~~~
  const nested = () => {
  ~~~~~~~~~~~~~~~~~~~~~~
    return 2;
    ~~~~~~~~~
  }
  ~
});
~~
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const computed = 2;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class computed {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const x = computed;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
computed()
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test {
    computed = 2;
  }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test {
    computed() {};
  }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test {
    x = computed(() => y());
  }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
function computedFactory() {
    return computed(() => {
      return 2;
    });
  }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
computed(() => {
    return 2;
  });
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
computed(function() {
    return 2;
  });
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
computed(() => {
    return () => {};
  });
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
computed(() => {
    return undefined;
  });
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
Test.computed(() => {});
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
instance.computed(() => {});
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
computed(() => {
      if (condition()) {
        return;
      }

      return 2;
    })
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/computed-must-return": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
computed(() => {
      if (condition()) {
        return;
      }

      return 2;
    })
```

</details>

<br>
