<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/component-class-suffix.ts
  - ../../tests/rules/component-class-suffix/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/component-class-suffix`

Classes decorated with @Component must have suffix "Component" (or custom) in their name. See more at https://angular.io/styleguide#style-02-03

- Type: suggestion

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * Default: `["Component"]`
   */
  suffixes?: string[];
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/component-class-suffix": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({
  selector: 'sg-foo-bar'
})
class Test {}
      ~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/component-class-suffix": [
      "error",
      {
        "suffixes": [
          "Component",
          "View"
        ]
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({
  selector: 'sgBarFoo'
})
class TestPage {}
      ~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/component-class-suffix": [
      "error",
      {
        "suffixes": [
          "Component"
        ]
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({
  selector: 'sgBarFoo'
})
class TestPage {}
      ~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/component-class-suffix": [
      "error",
      {
        "suffixes": [
          "Page"
        ]
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({
  selector: 'sgBarFoo'
})
class TestDirective {}
      ~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/component-class-suffix": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  selector: 'sg-foo-bar',
  template: '<foo-bar [foo]="bar">{{baz + 42}}</foo-bar>'
})
class TestComponent {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/component-class-suffix": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({
  selector: '[myHighlight]'
})
class TestDirective {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/component-class-suffix": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Pipe({
  selector: 'sg-test-pipe'
})
class TestPipe {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/component-class-suffix": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Injectable()
class TestService {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/component-class-suffix": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class TestEmpty {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/component-class-suffix": [
      "error",
      {
        "suffixes": [
          "Page"
        ]
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  selector: 'sgBarFoo'
})
class TestPage {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/component-class-suffix": [
      "error",
      {
        "suffixes": [
          "Page",
          "View"
        ]
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  selector: 'sgBarFoo'
})
class TestPage {}
```

</details>

<br>
