<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/component-max-inline-declarations.ts
  - ../../tests/rules/component-max-inline-declarations/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/component-max-inline-declarations`

Enforces a maximum number of lines in inline template, styles and animations. See more at https://angular.io/guide/styleguide#style-05-04

- Type: suggestion

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * Default: `3`
   */
  template?: number;
  /**
   * Default: `3`
   */
  styles?: number;
  /**
   * Default: `15`
   */
  animations?: number;
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/component-max-inline-declarations": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({
  template: `
            ~
    <div>first line</div>
    <div>second line</div>
    <div>third line</div>
    <div>fourth line</div>
  `
  ~
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/component-max-inline-declarations": [
      "error",
      {
        "template": 0
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({
  template: '<div>first line</div>'
            ~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/component-max-inline-declarations": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({
  styles: [
          ~
    `
      div {
        display: block;
        height: 40px;
      }
    `
  ]
  ~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/component-max-inline-declarations": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({
  styles: `
          ~
    div {
      display: block;
      height: 40px;
    }
  `
  ~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/component-max-inline-declarations": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({
  styles: [
          ~
    `
      div {
        display: block;
      }
    `,
    `
      span {
        width: 30px;
      }
    `
  ]
  ~
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/component-max-inline-declarations": [
      "error",
      {
        "styles": 0
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({
  styles: ['div { display: none; }']
          ~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/component-max-inline-declarations": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({
  animations: [{
              ~
    transformPanelWrap: trigger('transformPanelWrap', [
      transition('* => void', query('@transformPanel', [animateChild()], {optional: true})),
    ]),
    transformPanel: trigger('transformPanel', [
      state('void', style({
        transform: 'scaleY(0.8)',
        minWidth: '100%',
        opacity: 0
      })),
      state('showing', style({
        opacity: 1,
        minWidth: 'calc(100% + 32px)',
        transform: 'scaleY(1)'
      })),
      state('next', style({height: '0px', visibility: 'hidden'}))
    ])
  }]
   ~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/component-max-inline-declarations": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({
  animations: [
              ~
    trigger('dialogContainer', [
      transition('* => void', query('@transformPanel', [animateChild()], {optional: true}))
    ]),
    trigger('transformPanel', [
      state('void', style({
        transform: 'scaleY(0.8)',
        minWidth: '100%',
        opacity: 0
      })),
      state('showing', style({
        opacity: 1,
        minWidth: 'calc(100% + 32px)',
        transform: 'scaleY(1)'
      }))
    ]),
    trigger('transformPanel', [
      state('void', style({opacity: 0, transform: 'scale(1, 0)'}))
    ])
  ]
  ~
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/component-max-inline-declarations": [
      "error",
      {
        "animations": 2
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({
  animations: [{
              ~
    transformPanel: trigger('transformPanel', [
      state('void', style({opacity: 0, transform: 'scale(1, 0)'}))
    ])
  }]
   ~
})
class Test {}
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/component-max-inline-declarations": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  template: '<div>just one line template</div>'
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/component-max-inline-declarations": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  styles: ['div { display: none; }']
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/component-max-inline-declarations": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  styles: 'div { display: none; }'
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/component-max-inline-declarations": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  animations: [state('void', style({opacity: 0, transform: 'scale(1, 0)'}))]
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/component-max-inline-declarations": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  styleUrls: ['./foobar.scss'],
  templateUrl: './foobar.html',
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/component-max-inline-declarations": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  animations: [
    state('void', style({opacity: 0, transform: 'scale(1, 0)'}))
  ],
  templateUrl: './foobar.html',
})
class Test {}
```

</details>

<br>
