<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/consistent-component-styles.ts
  - ../../tests/rules/consistent-component-styles/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/consistent-component-styles`

Ensures consistent usage of `styles`/`styleUrls`/`styleUrl` within Component metadata

- Type: suggestion
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
type Options = "array" | "string";

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  styles: [':host { display: block; }']
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  standalone: true,
  imports: [MatButtonModule],
  styles: [':host { display: block; }'],
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  providers: []
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  styleUrls: ['./test.component.css']
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  standalone: true,
  imports: [MatButtonModule],
  styleUrls: ['./test.component.css'],
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  providers: []
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
const type = 'block';
@Component({
  styles: [`:host\t{ display: ${type}; }`]
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  styles: [':host\t{ display: block; }']
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  styles: [':host{ display: block; }']
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  styles: [":host{ display: block; }"]
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  styles: [`:host{ display: block; }`]
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  styleUrls: [`./test.component.css`]
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "string"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  styles: [':host { display: block; }']
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "string"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  styleUrls: ['./test.component.css']
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "array"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  styles: ':host { display: block; }'
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "array"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  standalone: true,
  imports: [MatButtonModule],
  styles: ':host { display: block; }',
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~
  providers: []
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "array"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  styleUrl: './test.component.css',
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "array"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
const type = 'block';
@Component({
  styles: `:host\t{ display: ${type}; }`
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "array"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  styles: ':host\t{ display: block; }'
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "array"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  styles: ':host{ display: block; }'
          ~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "array"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  styles: ":host{ display: block; }"
          ~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "array"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  styles: `:host{ display: block; }`
          ~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  styles: ':host { display: block; }',
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  styles: `
    :host { display: block; }
  `,
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  selector: 'my-test',
  standalone: true,
  imports: [CommonModule],
  styles: `
    :host { display: block; }
  `,
  providers: [FooService]
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  styles: [
    ':host { display: block; }',
    `.foo { color: red; }`
  ],
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  styleUrl: `./test.component.css`,
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  styleUrls: [
    '../shared.css',
    `./test.component.css`
  ],
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  selector: 'my-test',
  standalone: true,
  imports: [CommonModule],
  styleUrls: [
    '../shared.css',
    `./test.component.css`
  ],
  providers: [FooService]
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "string"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  styles: ':host { display: block; }',
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "string"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  styles: \`
    :host { display: block; }
  \`,
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "string"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  selector: 'my-test',
  standalone: true,
  imports: [CommonModule],
  styles: \`
    :host { display: block; }
  \`,
  providers: [FooService]
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "string"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  styles: [
    ':host { display: block; }',
    \`.foo { color: red; }\`
  ],
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "string"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  styleUrl: \`./test.component.css\`,
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "string"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  styleUrls: [
    '../shared.css',
    \`./test.component.css\`
  ],
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "string"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  selector: 'my-test',
  standalone: true,
  imports: [CommonModule],
  styleUrls: [
    '../shared.css',
    \`./test.component.css\`
  ],
  providers: [FooService]
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "array"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  styles: [':host { display: block; }'],
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "array"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  styles: [
    \`
      :host { display: block; }
    \`
  ],
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "array"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  selector: 'my-test',
  standalone: true,
  imports: [CommonModule],
  styles: [
    \`
      :host { display: block; }
    \`
  ],
  providers: [FooService]
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "array"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  styles: [
    ':host { display: block; }',
    \`.foo { color: red; }\`
  ],
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "array"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  styleUrls: [\`./test.component.css\`],
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "array"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  styleUrls: [
    '../shared.css',
    \`./test.component.css\`
  ],
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/consistent-component-styles": [
      "error",
      "array"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  selector: 'my-test',
  standalone: true,
  imports: [CommonModule],
  styleUrls: [
    '../shared.css',
    \`./test.component.css\`
  ],
  providers: [FooService]
})
class Test {}
```

</details>

<br>
