<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/contextual-lifecycle.ts
  - ../../tests/rules/contextual-lifecycle/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/contextual-lifecycle`

Ensures that lifecycle methods are used in a correct context

- Type: problem
- Category: Possible Errors

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```ts
@Component()
class Test {
  ngDoBootstrap() {}
  ~~~~~~~~~~~~~
}
```

```ts
@Directive()
class Test {
  ngDoBootstrap() {}
  ~~~~~~~~~~~~~
}
```

```ts
@Injectable()
class Test {
  ngAfterContentChecked() { console.log('AfterContentChecked'); }
  ~~~~~~~~~~~~~~~~~~~~~
}
```

```ts
@Injectable()
class Test {
  ngAfterContentInit() { console.log('ngAfterContentInit'); }
  ~~~~~~~~~~~~~~~~~~
}
```

```ts
@Injectable()
class Test {
  ngAfterViewInit() { console.log('ngAfterViewInit'); }
  ~~~~~~~~~~~~~~~
}
```

```ts
@Injectable()
class Test {
  ngDoBootstrap() {}
  ~~~~~~~~~~~~~
}
```

```ts
@Injectable()
class Test {
  ngDoCheck() { console.log('ngDoCheck'); }
  ~~~~~~~~~
}
```

```ts
@Injectable()
class Test {
  ngOnChanges() { console.log('ngOnChanges'); }
  ~~~~~~~~~~~
}
```

```ts
@Injectable()
class Test {
  ngOnInit() { console.log('ngOnInit'); }
  ~~~~~~~~
}
```

```ts
@NgModule()
class Test {
  ngAfterContentChecked() { console.log('ngAfterContentChecked'); }
  ~~~~~~~~~~~~~~~~~~~~~
}
```

```ts
@NgModule()
class Test {
  ngAfterContentInit() { console.log('ngAfterContentInit'); }
  ~~~~~~~~~~~~~~~~~~
}
```

```ts
@NgModule()
class Test {
  ngAfterViewChecked() { console.log('ngAfterViewChecked'); }
  ~~~~~~~~~~~~~~~~~~
}
```

```ts
@NgModule()
class Test {
  ngAfterViewInit() { console.log('ngAfterViewInit'); }
  ~~~~~~~~~~~~~~~
}
```

```ts
@NgModule()
class Test {
  ngDoCheck() { console.log('ngDoCheck'); }
  ~~~~~~~~~
}
```

```ts
@NgModule()
class Test {
  ngOnChanges() { console.log('ngOnChanges'); }
  ~~~~~~~~~~~
}
```

```ts
@NgModule()
class Test {
  ngOnInit() { console.log('ngOnInit'); }
  ~~~~~~~~
}
```

```ts
@NgModule()
class Test {
  ngOnDestroy() { console.log('ngOnDestroy'); }
  ~~~~~~~~~~~
}
```

```ts
@Pipe()
class Test {
  ngAfterContentChecked() { console.log('ngAfterContentChecked'); }
  ~~~~~~~~~~~~~~~~~~~~~
}
```

```ts
@Pipe()
class Test {
  ngAfterContentInit() { console.log('ngAfterContentInit'); }
  ~~~~~~~~~~~~~~~~~~
}
```

```ts
@Pipe()
class Test {
  ngAfterViewChecked() { console.log('ngAfterViewChecked'); }
  ~~~~~~~~~~~~~~~~~~
}
```

```ts
@Pipe()
class Test {
  ngAfterViewInit() { console.log('ngAfterViewInit'); }
  ~~~~~~~~~~~~~~~
}
```

```ts
@Pipe()
class Test {
  ngDoBootstrap() {}
  ~~~~~~~~~~~~~
}
```

```ts
@Pipe()
class Test {
  ngDoCheck() { console.log('ngDoCheck'); }
  ~~~~~~~~~
}
```

```ts
@Pipe()
class Test {
  ngOnChanges() { console.log('ngOnChanges'); }
  ~~~~~~~~~~~
}
```

```ts
@Pipe()
class Test {
  ngOnInit() { console.log('ngOnInit'); }
  ~~~~~~~~
}
```

```ts
@Pipe()
class Test implements DoCheck {
  constructor() {}

  ngDoCheck() {}
  ~~~~~~~~~
}

@Directive()
class TestDirective implements OnInit {
  ngOnInit() {
    console.log('Initialized');
  }
}
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```ts
@Injectable()
class Test {
  ngOnDestroy() { console.log('OnDestroy'); }
}
```

```ts
@Component()
class Test {
  ngAfterContentChecked() { console.log('AfterContentChecked'); }
}
```

```ts
@Component()
class Test {
  ngAfterContentInit() { console.log('AfterContentInit'); }
}
```

```ts
@Component()
class Test {
  ngAfterViewChecked() { console.log('AfterViewChecked'); }
}
```

```ts
@Component()
class Test {
  ngAfterViewInit() { console.log('AfterViewInit'); }
}
```

```ts
@Component()
class Test {
  ngDoCheck() { console.log('DoCheck'); }
}
```

```ts
@Component()
class Test {
  ngOnChanges() { console.log('OnChanges'); }
}
```

```ts
@Component()
class Test {
  ngOnDestroy() { console.log('OnDestroy'); }
}
```

```ts
@Component()
class Test {
  ngOnInit() { console.log('OnInit'); }
}
```

```ts
@Directive()
class Test {
  ngAfterContentChecked() { console.log('AfterContentChecked'); }
}
```

```ts
@Directive()
class Test {
  ngAfterContentInit() { console.log('AfterContentInit'); }
}
```

```ts
@Directive()
class Test {
  ngAfterViewChecked() { console.log('AfterViewChecked'); }
}
```

```ts
@Directive()
class Test {
  ngAfterViewInit() { console.log('AfterViewInit'); }
}
```

```ts
@Directive()
class Test {
  ngDoCheck() { console.log('DoCheck'); }
}
```

```ts
@Directive()
class Test {
  ngOnChanges() { console.log('OnChanges'); }
}
```

```ts
@Directive()
class Test {
  ngOnDestroy() { console.log('OnDestroy'); }
}
```

```ts
@Directive()
class Test {
  ngOnInit() { console.log('OnInit'); }
}
```

```ts
@Injectable()
class Test {
  ngOnDestroy() { console.log('OnDestroy'); }
}
```

```ts
@NgModule()
class Test {
  ngDoBootstrap() {}
}
```

```ts
@Pipe()
class Test {
  ngOnDestroy() { console.log('OnDestroy'); }
}
```
