<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/contextual-decorator.ts
  - ../../tests/rules/contextual-decorator/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/contextual-decorator`

Ensures that classes use contextual decorators in its body

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```ts
@Injectable({
  providedIn: 'root'
})
class Test {
  @Input()
  ~~~~~~~~
  get label(): string {
    return this._label;
  }
  set label(value: string) {
    this._label = value;
  }
  private _label: string;
}
```

```ts
@Injectable({
  providedIn: 'root'
})
class Test {
  @Input()
  ~~~~~~~~
  set label(value: string) {
    this._label = value;
  }
  get label(): string {
    return this._label;
  }
  private _label: string;
}
```

```ts
@Injectable({
  providedIn: 'root'
})
class Test {
  @ViewChild(Pane)
  ~~~~~~~~~~~~~~~~
  set label(value: Pane) {
    doSomething();
  }
}
```

```ts
@Injectable({
  providedIn: 'root'
})
class Test {
  @HostListener('mouseover')
  ~~~~~~~~~~~~~~~~~~~~~~~~~~
  mouseOver() {
    this.doSomething();
  }
}
```

```ts
@Injectable({
  providedIn: 'root'
})
class Test {
  constructor(
    @Attribute('test') private readonly test: string
    ~~~~~~~~~~~~~~~~~~
  ) {}
}
```

```ts
@Injectable({
  providedIn: 'root'
})
class Test {
  @ContentChild(Pane) pane: Pane;
  ~~~~~~~~~~~~~~~~~~~
}
```

```ts
@Injectable({
  providedIn: 'root'
})
class Test {
  @ContentChildren(Pane, { descendants: true }) arbitraryNestedPanes: QueryList<Pane>;
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
}
```

```ts
@Injectable({
  providedIn: 'root'
})
class Test {
  @HostBinding('class.card-outline') private isCardOutline: boolean;
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
}
```

```ts
@NgModule({
  providers: []
})
class Test {
  @Input() label: string;
  ~~~~~~~~
}
```

```ts
@NgModule({
  providers: []
})
class Test {
  @Output() emitter = new EventEmitter<void>();
  ~~~~~~~~~
}
```

```ts
@NgModule({
  providers: []
})
class Test {
  @ViewChild(Pane) pane: Pane;
  ~~~~~~~~~~~~~~~~
}
```

```ts
@NgModule({
  providers: []
})
class Test {
  @ViewChildren(Pane) panes: QueryList<Pane>;
  ~~~~~~~~~~~~~~~~~~~
}
```

```ts
@NgModule({
  providers: []
})
class Test {
  @ContentChild(Pane) pane: Pane;
  ~~~~~~~~~~~~~~~~~~~

  @Input()
  ~~~~~~~~
  get label(): string {
    return this._label;
  }
  set label(value: string) {
    this._label = value;
  }
  private _label: string;

  @ViewChild(Pane)
  ~~~~~~~~~~~~~~~~
  set label(value: Pane) {
    doSomething();
  }

  get type(): string {
    return this._type;
  }
  set type(value: string) {
    this._type = value;
  }
  private _type: string;

  private prop: string | undefined;

  constructor(
    @Attribute('test') private readonly test: string,
    ~~~~~~~~~~~~~~~~~~
    @Inject(LOCALE_ID) localeId: string
  ) {}

  @HostListener('mouseover')
  ~~~~~~~~~~~~~~~~~~~~~~~~~~
  mouseOver() {
    this.doSomething();
  }

  clickHandler(): void {}
}
```

```ts
@Pipe({
  name: 'test'
})
class Test {
  @Input()
  ~~~~~~~~
  get label(): string {
    return this._label;
  }
  set label(value: string) {
    this._label = value;
  }
  private _label: string;
}
```

```ts
@Pipe({
  name: 'test'
})
class Test {
  @Input()
  ~~~~~~~~
  set label(value: string) {
    this._label = value;
  }
  get label(): string {
    return this._label;
  }
  private _label: string;
}
```

```ts
@Pipe({
  name: 'test'
})
class Test {
  @ViewChild(Pane)
  ~~~~~~~~~~~~~~~~
  set label(value: Pane) {
    doSomething();
  }
}
```

```ts
@Pipe({
  name: 'test'
})
class Test {
  @HostListener('mouseover')
  ~~~~~~~~~~~~~~~~~~~~~~~~~~
  mouseOver() {
    this.doSomething();
  }
}
```

```ts
@Pipe({
  name: 'test'
})
class Test {
  constructor(
    @Attribute('test') private readonly test: string
    ~~~~~~~~~~~~~~~~~~
  ) {}
}
```

```ts
@Pipe({
  name: 'test'
})
class Test {
  @ContentChild(Pane) pane: Pane;
  ~~~~~~~~~~~~~~~~~~~
}
```

```ts
@Pipe({
  name: 'test'
})
class Test {
  @ContentChildren(Pane, { descendants: true }) arbitraryNestedPanes: QueryList<Pane>;
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
}
```

```ts
@Pipe({
  name: 'test'
})
class Test {
  @HostBinding('class.card-outline') private isCardOutline: boolean;
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
}
```

```ts
@Pipe({
  name: 'test'
})
class Test {
  @Input() label: string;
  ~~~~~~~~
}
```

```ts
@Pipe({
  name: 'test'
})
class Test {
  @Output() emitter = new EventEmitter<void>();
  ~~~~~~~~~
}
```

```ts
@Pipe({
  name: 'test'
})
class Test {
  @ViewChild(Pane) pane: Pane;
  ~~~~~~~~~~~~~~~~
}
```

```ts
@Pipe({
  name: 'test'
})
class Test {
  @ViewChildren(Pane) panes: QueryList<Pane>;
  ~~~~~~~~~~~~~~~~~~~
}
```

```ts
@Pipe({
  name: 'test'
})
class Test {
  @ContentChild(Pane) pane: Pane;
  ~~~~~~~~~~~~~~~~~~~

  @Input()
  ~~~~~~~~
  get label(): string {
    return this._label;
  }
  set label(value: string) {
    this._label = value;
  }
  private _label: string;

  @ViewChild(Pane)
  ~~~~~~~~~~~~~~~~
  set label(value: Pane) {
    doSomething();
  }

  get type(): string {
    return this._type;
  }
  set type(value: string) {
    this._type = value;
  }
  private _type: string;

  private prop: string | undefined;

  constructor(
    @Attribute('test') private readonly test: string,
    ~~~~~~~~~~~~~~~~~~
    @Inject(LOCALE_ID) localeId: string
  ) {}

  @HostListener('mouseover')
  ~~~~~~~~~~~~~~~~~~~~~~~~~~
  mouseOver() {
    this.doSomething();
  }

  clickHandler(): void {}
}
```

```ts
@Directive({
  selector: 'test'
})
class TestDirective {
  @Input() label: string;
}

@Pipe({
  name: 'test'
})
class Test {
  @Input() label: string;
  ~~~~~~~~
}
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:
