<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/directive-class-suffix.ts
  - ../../tests/rules/directive-class-suffix/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/directive-class-suffix`

Classes decorated with @Directive must have suffix "Directive" (or custom) in their name. See more at https://angular.io/styleguide#style-02-03

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * Default: `["Directive"]`
   */
  suffixes?: string[];
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```ts
@Directive({
  selector: 'sg-foo-bar'
})
class Test {}
      ~~~~
```

```ts
@Directive({
  selector: 'sg-foo-bar'
})
class TestDirectivePage implements AsyncValidator {}
      ~~~~~~~~~~~~~~~~~
```

```ts
@Directive({
  selector: 'sgBarFoo'
})
class TestPageDirective {}
      ~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```ts
@Directive({
  selector: 'sgBarFoo'
})
class TestDirective {}
```

```ts
@Directive({
  selector: 'sgBarFoo'
})
class TestValidator implements Validator {}
```

```ts
@Directive({
  selector: 'sgBarFoo'
})
class TestValidator implements AsyncValidator {}
```

```ts
@Directive
class Test {}
```

```ts
@Directive()
class Test {}
```

```ts
@Component({
  selector: 'sg-bar-foo'
})
class TestComponent {}
```

```ts
@Pipe({
  name: 'sgPipe'
})
class TestPipe {}
```

```ts
@Injectable()
class TestService {}
```

```ts
class TestEmpty {}
```
