<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/directive-selector.ts
  - ../../tests/rules/directive-selector/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/directive-selector`

Directive selectors should follow given naming rules. See more at https://angular.io/guide/styleguide#style-02-06 and https://angular.io/guide/styleguide#style-02-08.

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  type?: string | ("element" | "attribute")[];
  prefix?: string | unknown[];
  style?: "camelCase" | "kebab-case";
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```ts
@Directive({
  selector: 'app-foo-bar'
            ~~~~~~~~~~~~~
})
class Test {}
```

```ts
@Directive({
  selector: '[app-foo-bar]'
            ~~~~~~~~~~~~~~~
})
class Test {}
```

```ts
@Directive({
  selector: 'app-foo-bar[baz].app'
            ~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

```ts
@Directive({
  selector: '[app-bar-foo]'
            ~~~~~~~~~~~~~~~
})
class Test {}
```

```ts
@Directive({
  selector: 'appFooBar'
            ~~~~~~~~~~~
})
class Test {}
```

```ts
@Directive({
  selector: 'app'
            ~~~~~
})
class Test {}
```

```ts
@Directive({
  selector: `app-foo-bar`
            ~~~~~~~~~~~~~
})
class Test {}
```

```ts
@Directive({
  selector: '[appFooBar]'
            ~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:
