<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/directive-selector.ts
  - ../../tests/rules/directive-selector/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/directive-selector`

Directive selectors should follow given naming rules. See more at https://angular.io/guide/styleguide#style-02-06 and https://angular.io/guide/styleguide#style-02-08.

- Type: suggestion

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  type?: string | ("element" | "attribute")[];
  prefix?: string | unknown[];
  style?: "camelCase" | "kebab-case";
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": "element",
        "prefix": "bar",
        "style": "kebab-case"
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive({
  selector: 'app-foo-bar'
            ~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": "attribute",
        "prefix": [
          "cd",
          "ng"
        ],
        "style": "kebab-case"
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive({
  selector: '[app-foo-bar]'
            ~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": "element",
        "prefix": [
          "foo",
          "cd",
          "ng"
        ],
        "style": "kebab-case"
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive({
  selector: 'app-foo-bar[baz].app'
            ~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": "attribute",
        "prefix": "app",
        "style": "camelCase"
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive({
  selector: '[app-bar-foo]'
            ~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": "element",
        "prefix": "app",
        "style": "kebab-case"
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive({
  selector: 'appFooBar'
            ~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": "element",
        "prefix": "app",
        "style": "kebab-case"
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive({
  selector: 'app'
            ~~~~~
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": "attribute",
        "prefix": [
          "app",
          "ng"
        ],
        "style": "kebab-case"
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive({
  selector: `app-foo-bar`
            ~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": "element",
        "prefix": [
          "app",
          "ng"
        ],
        "style": "camelCase"
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive({
  selector: '[appFooBar]'
            ~~~~~~~~~~~~~
})
class Test {}
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": "element",
        "prefix": "app",
        "style": "kebab-case"
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({
  selector: 'app-foo-bar'
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": "attribute",
        "prefix": [
          "app",
          "ng"
        ],
        "style": "kebab-case"
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({
  selector: '[app-foo-bar]'
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": "element",
        "prefix": [
          "app",
          "cd",
          "ng"
        ],
        "style": "kebab-case"
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({
  selector: 'app-foo-bar[baz].app'
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": "element",
        "prefix": [
          "app",
          "cd",
          "ngg"
        ],
        "style": "kebab-case"
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({ selector: 'app-bar' }) class TestOne {}
@Directive({ selector: 'ngg-bar' }) class TestTwo {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": "element",
        "prefix": "app",
        "style": "camelCase"
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({
  selector: 'appBarFoo'
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": "element",
        "prefix": "app1",
        "style": "kebab-case"
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({
  selector: 'app1-foo-bar'
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": "element",
        "prefix": "app",
        "style": "kebab-case"
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const selectorName = 'appFooBar';
@Directive({
  selector: selectorName
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": "attribute",
        "prefix": [
          "app",
          "ng"
        ],
        "style": "kebab-case"
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({
  selector: \`[app-foo-bar]\`
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": "attribute",
        "prefix": [
          "app",
          "baz"
        ],
        "style": "kebab-case"
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({
  selector: 'baz-[app-bar-foo][foe].bar'
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": "element",
        "prefix": [
          "app",
          "ng"
        ],
        "style": "kebab-case"
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({
  selector: 'app-bar-foo[baz].bar'
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": [
          "element"
        ],
        "prefix": [
          "bar"
        ],
        "style": "kebab-case"
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  selector: 'app-foo-bar'
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": [
          "attribute"
        ],
        "prefix": [
          "app"
        ],
        "style": "kebab-case"
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({
  selector: \`
    [app-foo-bar]
  \`
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": [
          "attribute"
        ],
        "prefix": [
          "app"
        ],
        "style": "kebab-case"
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({
  selector: \`
    [app-foo-bar],
    [app-bar-foo]
  \`
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/directive-selector": [
      "error",
      {
        "type": [
          "attribute"
        ],
        "prefix": [
          "app"
        ],
        "style": "kebab-case"
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({
  selector: 'button[app-foo-bar]'
})
class Test {}
```

</details>

<br>
