<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-attribute-decorator.ts
  - ../../tests/rules/no-attribute-decorator/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/no-attribute-decorator`

Disallows usage of @Attribute decorator.

- Type: problem
- Category: Possible Errors

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```ts
class Test {
  constructor(@Attribute() foo: string) {}
              ~~~~~~~~~~~~
}
```

```ts
class Test {
  constructor(
    @Inject(TOKEN) token: string,
    randomNumber: number,
    @Attribute() foo: string,
    ~~~~~~~~~~~~
    @Attribute('baz') bar: string
    ~~~~~~~~~~~~~~~~~
  ) {}
}
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```ts
class Test {
  foo() {}
}
```

```ts
class Test {
  constructor() {}
}
```

```ts
class Test {
  constructor(@Optional() foo: string, @Optional() bar: string, baz: number) {}
}
```
