<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-async-lifecycle-method.ts
  - ../../tests/rules/no-async-lifecycle-method/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/no-async-lifecycle-method`

Angular Lifecycle methods should not be async. Angular does not wait for async lifecycle but the code incorrectly suggests it does.

- Type: problem

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-async-lifecycle-method": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  async ngAfterContentChecked() { }
        ~~~~~~~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-async-lifecycle-method": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  async ngAfterContentInit() { }
        ~~~~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-async-lifecycle-method": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  async ngAfterViewChecked() { }
        ~~~~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-async-lifecycle-method": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  async ngAfterViewInit() { }
        ~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-async-lifecycle-method": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Pipe()
class Test {
  async ngDoBootstrap() { }
        ~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-async-lifecycle-method": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  async ngDoCheck() { }
        ~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-async-lifecycle-method": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  async ngOnChanges() { }
        ~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-async-lifecycle-method": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  async ngOnDestroy() { }
        ~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-async-lifecycle-method": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  async ngOnInit() { }
        ~~~~~~~~
}
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-async-lifecycle-method": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  ngAfterContentChecked() { }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-async-lifecycle-method": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  ngAfterContentInit() { }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-async-lifecycle-method": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  ngAfterViewChecked() { }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-async-lifecycle-method": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  ngAfterViewInit() { }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-async-lifecycle-method": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  ngDoBootstrap() { }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-async-lifecycle-method": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  ngDoCheck() { }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-async-lifecycle-method": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  ngOnChanges() { }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-async-lifecycle-method": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  ngOnDestroy() { }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-async-lifecycle-method": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  ngOnInit() { }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-async-lifecycle-method": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
async function ngAfterContentChecked() {}
async function ngAfterContentInit() {}
async function ngAfterViewChecked() {}
async function ngAfterViewInit() {}
async function ngDoBootstrap() {}
async function ngDoCheck() {}
async function ngOnChanges() {}
async function ngOnDestroy() {}
async function ngOnInit() {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-async-lifecycle-method": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test {
  async ngOnInit() {}
}
```

</details>

<br>
