<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-conflicting-lifecycle.ts
  - ../../tests/rules/no-conflicting-lifecycle/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/no-conflicting-lifecycle`

Ensures that directives not implement conflicting lifecycle interfaces.

- Type: suggestion

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test implements DoCheck, OnChanges, run {
                      ~~~~~~~  ~~~~~~~~~
  test() {}
  test1() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test implements DoCheck, OnChanges {
                      ~~~~~~~  ~~~~~~~~~
  ngDoCheck() {}
  ~~~~~~~~~~~~~~
  ngOnChanges() {}
  ~~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test {
  ngDoCheck() {}
  ~~~~~~~~~~~~~~
  ngOnChanges() {}
  ~~~~~~~~~~~~~~~~
}
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test implements DoCheck {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test {
  ngDoCheck() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test implements DoCheck {
  ngDoCheck() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test implements OnChanges {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test {
  ngOnChanges() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test implements OnChanges {
  ngOnChanges() {}
}
```

</details>

<br>
