<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-conflicting-lifecycle.ts
  - ../../tests/rules/no-conflicting-lifecycle/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/no-conflicting-lifecycle`

## ⚠️ THIS RULE IS DEPRECATED

---

Ensures that directives do not implement conflicting lifecycle interfaces.

- Type: suggestion

<br>

## Rationale

This rule was created with the intent to prevent potential issues when both DoCheck and OnChanges lifecycle hooks are implemented together, as they both deal with change detection and could create confusing or duplicated logic. However, the rule has proven to be overly broad in practice. It triggers whenever a component or directive implements both hooks, regardless of whether they actually conflict or track the same data. In reality, there are legitimate use cases where both hooks can coexist without issues—for instance, using OnChanges to respond to specific input changes while using DoCheck for custom change detection logic on different data. Because the rule cannot accurately determine whether the hooks are actually conflicting (i.e., checking the same variables), it produces false positives and has been removed from the recommended configuration. While the rule was well-intentioned, it should not be relied upon in practice.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test implements DoCheck, OnChanges, run {
                      ~~~~~~~  ~~~~~~~~~
  test() {}
  test1() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test implements DoCheck, OnChanges {
                      ~~~~~~~  ~~~~~~~~~
  ngDoCheck() {}
  ~~~~~~~~~~~~~~
  ngOnChanges() {}
  ~~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test {
  ngDoCheck() {}
  ~~~~~~~~~~~~~~
  ngOnChanges() {}
  ~~~~~~~~~~~~~~~~
}
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test implements DoCheck {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test {
  ngDoCheck() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test implements DoCheck {
  ngDoCheck() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test implements OnChanges {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test {
  ngOnChanges() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test implements OnChanges {
  ngOnChanges() {}
}
```

</details>

<br>
