<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-conflicting-lifecycle.ts
  - ../../tests/rules/no-conflicting-lifecycle/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/no-conflicting-lifecycle`

Ensures that directives not implement conflicting lifecycle interfaces.

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```ts
class Test implements DoCheck, OnChanges, run {
                      ~~~~~~~  ~~~~~~~~~
  test() {}
  test1() {}
}
```

```ts
class Test implements DoCheck, OnChanges {
                      ~~~~~~~  ~~~~~~~~~
  ngDoCheck() {}
  ~~~~~~~~~~~~~~
  ngOnChanges() {}
  ~~~~~~~~~~~~~~~~
}
```

```ts
class Test {
  ngDoCheck() {}
  ~~~~~~~~~~~~~~
  ngOnChanges() {}
  ~~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```ts
class Test implements DoCheck {}
```

```ts
class Test {
  ngDoCheck() {}
}
```

```ts
class Test implements DoCheck {
  ngDoCheck() {}
}
```

```ts
class Test implements OnChanges {}
```

```ts
class Test {
  ngOnChanges() {}
}
```

```ts
class Test implements OnChanges {
  ngOnChanges() {}
}
```
