<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-conflicting-lifecycle.ts
  - ../../tests/rules/no-conflicting-lifecycle/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/no-conflicting-lifecycle`

Ensures that directives do not implement conflicting lifecycle interfaces.

- Type: suggestion

<br>

## Rationale

The DoCheck and OnChanges lifecycle hooks serve overlapping purposes for detecting changes, but using them together creates confusing and potentially buggy behavior. When both are implemented, ngOnChanges() runs first when input properties change, then ngDoCheck() runs on every change detection cycle. This can lead to duplicated logic, performance issues from running change detection twice, and confusion about which hook should handle which logic. Angular's documentation recommends using one or the other, not both: use OnChanges for simple input tracking, or DoCheck when you need more granular control over change detection.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test implements DoCheck, OnChanges, run {
                      ~~~~~~~  ~~~~~~~~~
  test() {}
  test1() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test implements DoCheck, OnChanges {
                      ~~~~~~~  ~~~~~~~~~
  ngDoCheck() {}
  ~~~~~~~~~~~~~~
  ngOnChanges() {}
  ~~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test {
  ngDoCheck() {}
  ~~~~~~~~~~~~~~
  ngOnChanges() {}
  ~~~~~~~~~~~~~~~~
}
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test implements DoCheck {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test {
  ngDoCheck() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test implements DoCheck {
  ngDoCheck() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test implements OnChanges {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test {
  ngOnChanges() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-conflicting-lifecycle": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test implements OnChanges {
  ngOnChanges() {}
}
```

</details>

<br>
