<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-duplicates-in-metadata-arrays.ts
  - ../../tests/rules/no-duplicates-in-metadata-arrays/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/no-duplicates-in-metadata-arrays`

Ensures that metadata arrays do not contain duplicate entries.

- Type: suggestion

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-duplicates-in-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@NgModule({
  providers: [ProviderA, ProviderB, ProviderA, ProviderB]
                                    ~~~~~~~~~  ~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-duplicates-in-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@NgModule({
  declarations: [DeclarationA, DeclarationB, DeclarationA, DeclarationB],
                                             ~~~~~~~~~~~~  ~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-duplicates-in-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@NgModule({
  imports: [ImportA, ImportB, ImportA, ImportB]
                              ~~~~~~~  ~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-duplicates-in-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@NgModule({
  exports: [ExportA, ExportB, ExportA, ExportB]
                              ~~~~~~~  ~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-duplicates-in-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({
  imports: [ImportA, ImportB, ImportA, ImportB]
                              ~~~~~~~  ~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-duplicates-in-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive({
  providers: [ProviderA, ProviderB, ProviderA, ProviderB]
                                    ~~~~~~~~~  ~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-duplicates-in-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@NgModule({
  imports: [
    ImportA,
    ImportB,
    ImportA,
    ~~~~~~~
    ImportB,
    ~~~~~~~
    ImportA
    ~~~~~~~
  ]
})
class Test {}
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-duplicates-in-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@NgModule({
  providers: [ProviderA, ProviderB, ProviderC],
  declarations: [DeclarationA, DeclarationB, DeclarationC],
  imports: [ImportA, ImportB, ImportC],
  exports: [ExportA, ExportB, ExportC]
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-duplicates-in-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  imports: [ImportA, ImportB, ImportC]
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-duplicates-in-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({
  providers: [ProviderA, ProviderB, ProviderC],
})
class Test {}
```

</details>

<br>
