<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-experimental.ts
  - ../../tests/rules/no-experimental/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/no-experimental`

Disallow using code which is marked as experimental

- Type: problem

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
interface Test {};
const test: Test = {};
            ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
interface Test {};
Partial<Test>;
        ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
interface Test {
  good?: () => void;
  /** @experimental */
  bad?: () => void;
};
const test: Test = {};
test.bad();
     ~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
interface Test {
  good?: string;
  /** @experimental */
  bad?: string;
};
const test: Test = {};
if (test.good || test.bad) return;
                      ~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const test = {
  /** @experimental */
  a: {
    b: {
      c: () => {},
    },
  },
};
const value = test.a.b.c();
                   ~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
class Test {}
const test: Test = {};
            ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
class Test {}

const test = new Test();
                 ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test {
  /** @publicApi */
  good?: string;
  /** @experimental */
  bad?: string;
}

const test = new Test();
test.good = 'good';
test.bad = 'bad';
     ~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test {
  /** @experimental */
  func() {}
}

const test = new Test();
test.func();
     ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test {
  good?: string;
  /** @experimental */
  bad?: string;
}
const test = new Test();
if (test.good || test.bad) return;
                      ~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test {
  good?: string;
  /** @experimental */
  bad?: string;
}
const { good, bad } = new Test();
              ~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test {
  good() {}
  /** @experimental */
  bad() {}
}
const test = new Test();
test.good();
test.bad();
     ~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
const Test = class {};
new Test();
    ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
type Test = 'a' | 'b';
const test: Test = 'a';
            ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
function test() {}
test();
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
const test = () => {}
test();
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
function test(param = '') {}
test();
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
function test(param = ''): any {}
test()?.a()?.b();
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental @deprecated */
const test = 'test';
const myString = test + '-suffix';
                 ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
let test = [];
for (const value of test) {}
                    ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
var test = true;
const value = test ? 'yes' : 'no';
              ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
const test = 'test';
const another = test;
                ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
const test = 'test';
console.log(test);
            ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
const test = { a: { b: {} } };
test.a.b;
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
const test = {};
const result = { ...test };
                    ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
const test = [];
const result = [ ...test ];
                    ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
type A = () => {
  /** @experimental */
  b: string;
};
declare const a: A;

const { b } = a();
        ~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
let test: number = 1;
const result = test++;
               ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
let i = 0;
while(false) {
  i++;
  ~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
const x = 1;

const { y = x } = {};
            ~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const test = {
  a: '',
  /** @experimental */
  b: '',
};
test.b;
     ~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
async function test() {}

await test();
      ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
const test = 'test';
const myString = `${test}-suffix`;
                    ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
enum Test {
  member = 1,
}
Test.member;
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
enum Test {
  member1 = 1,
  /** @experimental */
  member2 = 2,
}
Test.member1;
Test.member2;
     ~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
function $localize2(value: TemplateStringsArray) {
  return value;
}
const result = $localize2`Hello World!`;
               ~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
import { experimentalFunction } from './experimental';

experimentalFunction();
~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
import { experimentalFunction as alias } from './experimental';

alias();
~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
import { ExperimentalClass } from './experimental';

const instance = new ExperimentalClass();
                     ~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
import { experimentalConst } from './experimental';

const myConst = experimentalConst;
                ~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
import { experimentalConst } from './experimental';

const myConst = { prop: experimentalConst };
                        ~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
import { SomeInterface } from './experimental';

const obj: SomeInterface = {};
const { experimentalItem } = obj;
        ~~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
/** Not experimental */
interface Test {
  /** @publicApi */
  a1: string;
  /** @experimental */
  a2: string;
};
const test: Test = {};
test.a1 = 'value';
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
/** Not experimental */
class Test {
  /** @deprecated */
  a1: string;
  /** @experimental */
  a2: string;
}
const test = new Test();
test.a1 = 'value';
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
/** Not experimental */
const test = {};
if (test) return;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
/** Not experimental */
function test() {}
test();
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
/** @experimental */
declare module "some-module" {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
import { regularFunction, experimentalFunction } from './experimental';

regularFunction();
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
import { RegularClass, ExperimentalClass } from './experimental';

const instance = new RegularClass();
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
import { experimentalFunction } from './experimental';

export { experimentalFunction as alias };
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
export * from './experimental';
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
/** @experimental */
declare function test(): void;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
/** @experimental */
declare const test: () => void;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
import { regularConst } from './experimental';
const myConst = regularConst;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
import { regularConst } from './experimental';
const myConst = { prop: regularConst };
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
import { SomeInterface } from './experimental';
const obj: SomeInterface = {};
const { regularItem } = obj;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
import { something } from './non-existing';
const myVar = something;
```

</details>

<br>
