<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-empty-lifecycle-method.ts
  - ../../tests/rules/no-empty-lifecycle-method/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/no-empty-lifecycle-method`

Disallows declaring empty lifecycle methods

- Type: suggestion
- Category: Best Practices

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

‚ùå - Examples of **incorrect** code for this rule:

```ts
@Component()
class Test {
  ngAfterContentChecked() {}
  ~~~~~~~~~~~~~~~~~~~~~~~~~~
}
```

```ts
@Directive()
class Test {
  ngAfterContentInit() {}
  ~~~~~~~~~~~~~~~~~~~~~~~
}
```

```ts
@Injectable()
class Test {
  ngAfterViewChecked() {}
  ~~~~~~~~~~~~~~~~~~~~~~~
}
```

```ts
@NgModule()
class Test {
  ngAfterViewInit() {}
  ~~~~~~~~~~~~~~~~~~~~
}
```

```ts
import { HttpInterceptor } from '@angular/common/http';
import {
  DoCheck,
  DoBootstrap
} from '@angular/core';
@Pipe()
class Test {
  ngDoBootstrap() {}
  ~~~~~~~~~~~~~~~~~~
}
```

```ts
import {
  AfterViewChecked,
  DoCheck,
  AfterViewInit,
} from '@angular/core';
@Component()
class Test
    implements AfterViewChecked,
               AfterViewInit,
               DoCheck {
  ngDoCheck() {}
  ~~~~~~~~~~~~~~
}
```

```ts
import {OnChanges, AfterContentInit} from '@angular/core';
@Directive()
class Test implements OnChanges, AfterContentInit {
  ngOnChanges() {}
  ~~~~~~~~~~~~~~~~
}
```

```ts
import {OnDestroy} from '@angular/core';
@Injectable()
class Test implements OnDestroy {
  ngOnDestroy() {}
  ~~~~~~~~~~~~~~~~
}
```

```ts
import {DoBootstrap, OnInit} from '@angular/core';
@NgModule()
class Test implements OnInit, DoBootstrap {
  ngOnInit() {}
  ~~~~~~~~~~~~~
}
```

<br>

---

<br>

‚úÖ - Examples of **correct** code for this rule:

```ts
@Component()
class Test {
  ngAfterContentChecked() { console.log('AfterContentChecked'); }
}
```

```ts
@Directive()
class Test {
  ngAfterContentInit() { console.log('AfterContentInit'); }
}
```

```ts
@Injectable()
class Test {
  ngAfterViewChecked() { console.log('AfterViewChecked'); }
}
```

```ts
@NgModule()
class Test {
  ngAfterViewInit() { console.log('AfterViewInit'); }
}
```

```ts
@Pipe()
class Test {
  ngDoBootstrap() { console.log('DoBootstrap'); }
}
```

```ts
@Component()
class Test {
  ngDoCheck() { console.log('DoCheck'); }
}
```

```ts
@Directive()
class Test {
  ngOnChanges() { console.log('OnChanges'); }
}
```

```ts
@Injectable()
class Test {
  ngOnDestroy() { console.log('OnDestroy'); }
}
```

```ts
@NgModule()
class Test {
  ngOnInit() { console.log('OnInit'); }
}
```

```ts
class Test {
  ngOnInit() {}
}
```
