<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-experimental.ts
  - ../../tests/rules/no-experimental/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/no-experimental`

Disallow using code which is marked as experimental

- Type: problem

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
/** @experimental */
interface Test {};
const test: Test = {};
            ~~~~
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-experimental": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
/** Not experimental */
interface Test {};
const test: Test = {};
```

</details>

<br>
