<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-forward-ref.ts
  - ../../tests/rules/no-forward-ref/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/no-forward-ref`

Disallows usage of `forwardRef` references for DI

- Type: suggestion

<br>

## Rationale

The forwardRef() function is a workaround for circular dependencies in Angular's dependency injection system, but it obscures the dependency graph and can make code harder to understand and maintain. When forwardRef() is used, it's often a sign of poor architecture, such as circular dependencies between components or services that could be resolved through better abstraction. Modern Angular and TypeScript have reduced the need for forwardRef() in most cases. Instead, consider restructuring your code to eliminate circular dependencies, using interfaces for dependency injection, or moving shared logic into a separate service.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-forward-ref": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({
  selector: 'test',
  template: ''
})
export class Test {
  constructor(@Inject(forwardRef(() => TestService)) testService) {}
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
}
export class TestService {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-forward-ref": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({
  providers: [
    {
      multi: true,
      provide: NG_VALUE_ACCESSOR,
      useExisting: forwardRef(() => TagsValueAccessor)
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    }
  ],
  selector: '[tags]',
})
export class TagsValueAccessor {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-forward-ref": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const TAGS_VALUE_ACCESSOR: StaticProvider = {
  multi: true,
  provide: NG_VALUE_ACCESSOR,
  useExisting: forwardRef(() => TagsValueAccessor)
               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
};
@Directive({
  providers: [TAGS_VALUE_ACCESSOR],
  selector: '[tags]'
})
export class TagsValueAccessor {}
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-forward-ref": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  selector: 'test',
  template: ''
})
export class Test {
  constructor() {
    this.test();
  }

  test() {}
}
```

</details>

<br>
