<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-host-metadata-property.ts
  - ../../tests/rules/no-host-metadata-property/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/no-host-metadata-property`

Disallows usage of the `host` metadata property. See more at https://angular.io/styleguide#style-06-03

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  allowStatic?: boolean;
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```ts
@Component({
  host: {
  ~~~~~~~
    class: 'my-class',
    [type]: 'test',
    '(click)': 'bar()'
  },
  ~
  selector: 'app-test'
})
class Test {}
```

```ts
@Directive({
  host: {
  ~~~~~~~
    class: 'my-class',
    [type]: 'test',
    '(click)': 'bar()'
  },
  ~
  selector: 'app-test'
})
class Test {}
```

```ts
const computed = '[class]';

@Directive({
  host: {
    [computed]: 'test',
    ~~~~~~~~~~~~~~~~~~
    static: true,
    'class': 'class1',
    '(click)': 'bar()',
    ~~~~~~~~~~~~~~~~~~
    '[attr.role]': role,
    ~~~~~~~~~~~~~~~~~~~
    '[@routerTransition]': 'test'
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  },
  selector: 'app-test'
})
class Test {}
```

```ts
@Component({
  host,
  ~~~~
  selector: 'app-test'
})
class Test {}
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```ts
@Component({
  selector: 'app-test',
  template: 'Hello'
})
class Test {}
```

```ts
@Directive({
  selector: 'app-test'
})
class Test {}
```
