<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-inputs-metadata-property.ts
  - ../../tests/rules/no-inputs-metadata-property/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/no-inputs-metadata-property`

Disallows usage of the `inputs` metadata property. See more at https://angular.io/styleguide#style-05-12

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```ts
@Component({
  inputs: [
  ~~~~~~~~~
    'id: foo'
  ],
  ~
  selector: 'app-test'
})
class Test {}
```

```ts
@Directive({
  inputs: [
  ~~~~~~~~~
    'id: foo'
  ],
  ~
  selector: 'app-test'
})
class Test {}
```

```ts
@Component({
  inputs,
  ~~~~~~
})
class Test {}
```

```ts
@Component({
  inputs: [],
  ~~~~~~~~~~
})
class Test {}
```

```ts
const test = [];
@Component({
  inputs: test,
  ~~~~~~~~~~~~
})
class Test {}
```

```ts
@Component({
  inputs: undefined,
  ~~~~~~~~~~~~~~~~~
})
class Test {}
```

```ts
function inputs() {
  return [];
}

@Directive({
  'inputs': inputs(),
  ~~~~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```ts
class Test {}
```

```ts
@Component()
class Test {}
```

```ts
@Directive({})
class Test {}
```

```ts
const options = {};
@Component(options)
class Test {}
```

```ts
@Directive({
  selector: 'app-test',
  template: 'Hello'
})
class Test {}
```

```ts
@Component({
  selector: 'app-test',
  queries: {},
})
class Test {}
```

```ts
const inputs = 'providers';
@Directive({
  [inputs]: [],
})
class Test {}
```

```ts
@NgModule({
  bootstrap: [Foo]
})
class Test {}
```
