<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-implicit-take-until-destroyed.ts
  - ../../tests/rules/no-implicit-take-until-destroyed/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/no-implicit-take-until-destroyed`

Ensures that `takeUntilDestroyed()` is called with an explicit `DestroyRef` when used outside of an injection context

- Type: problem

<br>

## Rationale

The `takeUntilDestroyed()` operator can automatically infer the current component's or directive's `DestroyRef` only when called within an injection context — specifically in constructors or field initializers of classes decorated with `@Component`, `@Directive`, `@Injectable`, or `@Pipe`, or in factory functions (`useFactory` in providers or `factory` in InjectionTokens). When used in lifecycle methods like `ngOnInit()` or `ngAfterViewInit()`, in regular methods, in constructors of plain classes not managed by Angular's DI system, or in `@NgModule` classes (which don't support the `ngOnDestroy` lifecycle), the injection context is not available, and `takeUntilDestroyed()` will throw a runtime error: "NG0203: inject() must be called from an injection context." To fix this, inject `DestroyRef` using `inject(DestroyRef)` and pass it explicitly: `takeUntilDestroyed(this.destroyRef)`.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-implicit-take-until-destroyed": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  ngOnInit() {
    this.data$.pipe(takeUntilDestroyed()).subscribe();
                    ~~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-implicit-take-until-destroyed": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  ngAfterViewInit() {
    this.data$.pipe(takeUntilDestroyed()).subscribe();
                    ~~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-implicit-take-until-destroyed": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  anyMethod() {
    this.data$.pipe(takeUntilDestroyed()).subscribe();
                    ~~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-implicit-take-until-destroyed": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive()
class Test {
  anyMethod() {
    this.data$.pipe(takeUntilDestroyed()).subscribe();
                    ~~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-implicit-take-until-destroyed": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Injectable()
class TestService {
  ngOnDestroy() {
    this.data$.pipe(takeUntilDestroyed()).subscribe();
                    ~~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-implicit-take-until-destroyed": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Injectable()
class TestService {
  loadData() {
    this.data$.pipe(takeUntilDestroyed()).subscribe();
                    ~~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-implicit-take-until-destroyed": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test {
  constructor() {
    this.data$.pipe(takeUntilDestroyed()).subscribe();
                    ~~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-implicit-take-until-destroyed": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@NgModule()
class AppModule {
  constructor() {
    this.data$.pipe(takeUntilDestroyed()).subscribe();
                    ~~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-implicit-take-until-destroyed": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@NgModule()
class AppModule {
  private data = this.http.get('/api').pipe(takeUntilDestroyed());
                                            ~~~~~~~~~~~~~~~~~~~~
}
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-implicit-take-until-destroyed": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  constructor() {
    this.data$.pipe(takeUntilDestroyed()).subscribe();
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-implicit-take-until-destroyed": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive()
class Test {
  constructor() {
    this.data$.pipe(takeUntilDestroyed()).subscribe();
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-implicit-take-until-destroyed": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Injectable()
class Test {
  constructor() {
    this.data$.pipe(takeUntilDestroyed()).subscribe();
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-implicit-take-until-destroyed": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Pipe({ name: 'myPipe' })
class MyPipe {
  constructor() {
    this.data$.pipe(takeUntilDestroyed()).subscribe();
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-implicit-take-until-destroyed": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  private data = this.http.get('/api').pipe(takeUntilDestroyed());
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-implicit-take-until-destroyed": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Injectable()
class TestService {
  private data = this.http.get('/api').pipe(takeUntilDestroyed());
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-implicit-take-until-destroyed": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Pipe({ name: 'myPipe' })
class MyPipe {
  private data = this.http.get('/api').pipe(takeUntilDestroyed());
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-implicit-take-until-destroyed": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const MY_TOKEN = new InjectionToken('my-token', {
  factory: () => {
    return someObservable.pipe(takeUntilDestroyed());
  }
});
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-implicit-take-until-destroyed": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  ngOnInit() {
    this.data$.pipe(takeUntilDestroyed(this.destroyRef)).subscribe();
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-implicit-take-until-destroyed": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  doSomething() {
    this.data$.pipe(takeUntilDestroyed(this.destroyRef)).subscribe();
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-implicit-take-until-destroyed": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Injectable()
class TestService {
  loadData() {
    this.data$.pipe(takeUntilDestroyed(this.destroyRef)).subscribe();
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-implicit-take-until-destroyed": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  providers: [
    {
      provide: SomeToken,
      useFactory: () => data$.pipe(takeUntilDestroyed())
    }
  ]
})
class Test {}
```

</details>

<br>
