<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-input-rename.ts
  - ../../tests/rules/no-input-rename/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/no-input-rename`

Ensures that input bindings are not aliased

- Type: suggestion
- Category: Best Practices
- üîß Supports autofix (`--fix`)

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * A list with allowed input names
   *
   * Default: `[]`
   */
  allowedNames?: string[];
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

‚ùå - Examples of **incorrect** code for this rule:

```ts
@Component({
  inputs: ['a: b']
           ~~~~~~
})
class Test {}
```

```ts
@Directive({
  outputs: ['abort'],
  'inputs': [boundary, `test: copy`, 'check: check'],
                       ~~~~~~~~~~~~
})
class Test {}
```

```ts
@Component({
  ['inputs']: ['orientation: orientation'],
               ~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

```ts
@Directive({
  [`inputs`]: ['orientation: orientation'],
               ~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

```ts
@Component()
class Test {
  @Custom() @Input(`change`) _change = getInput();
                   ~~~~~~~~
}
```

```ts
@Directive()
class Test {
  @Input('change') change = (this.subject$ as Subject<{blur: boolean}>).pipe();
         ~~~~~~~~
}
```

```ts
@Directive({
  selector: 'foo'
})
class Test {
  @Input('aria-invalid') ariaBusy: string;
         ~~~~~~~~~~~~~~
}
```

```ts
@Component({
  selector: 'foo'
})
class Test {
  @Input('fooColor') colors: string;
         ~~~~~~~~~~
}
```

```ts
@Directive({
  'selector': 'foo'
})
class Test {
  @Input('foocolor') color: string;
         ~~~~~~~~~~
}
```

```ts
@Component({
  selector: 'click',
})
class Test {}

@Injectable()
class Test {
  @Input('click') blur = this.getInput();
         ~~~~~~~
}
```

<br>

---

<br>

‚úÖ - Examples of **correct** code for this rule:

```ts
class Test {}
```

```ts
@Page({
  inputs: ['play', popstate, `online`, 'obsolete: obsol', 'store: storage'],
})
class Test {}
```

```ts
@Component()
class Test {
  change = new EventEmitter();
}
```

```ts
@Directive()
class Test {
  @Input() buttonChange = new EventEmitter<'change'>();
}
```

```ts
@Component({
  inputs,
})
class Test {}
```

```ts
@Directive({
  inputs: [...test],
})
class Test {}
```

```ts
@Component({
  inputs: func(),
})
class Test {}
```

```ts
@Directive({
  inputs: [func(), 'a'],
})
class Test {}
```

```ts
@Component({})
class Test {
  @Input() set setter(setter: string) {}
}
```

```ts
const change = 'change';
@Component()
class Test {
  @Input(change) touchMove: EventEmitter<{ action: 'click' | 'close' }> = new EventEmitter<{ action: 'click' | 'close' }>();
}
```

```ts
const blur = 'blur';
const click = 'click';
@Directive()
class Test {
  @Input(blur) [click]: EventEmitter<Blur>;
}
```

```ts
@Component({
  selector: 'foo[bar]'
})
class Test {
  @Input() bar: string;
}
```

```ts
@Component({
  selector: '[foo], test',
})
class Test {
  @Input('foo') label: string;
}
```

```ts
@Directive({
  selector: 'foo'
})
class Test {
  @Input('aria-label') ariaLabel: string;
}
```

```ts
@Directive({
  selector: 'foo'
})
class Test {
  @Input('fooMyColor') myColor: string;
}
```
