<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-inject-outside-di-context.ts
  - ../../tests/rules/no-inject-outside-di-context/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/no-inject-outside-di-context`

Ensures that `inject()` is called in an injection context

- Type: problem

<br>

## Rationale

The `inject()` function can only be called within an injection context — specifically in constructors or field initializers of classes decorated with `@Component`, `@Directive`, `@Injectable`, or `@Pipe`, in special cases like guards, resolvers and interceptors, or in factory functions (`useFactory` in providers or `factory` in InjectionTokens). When used in lifecycle methods like `ngOnInit()` or `ngAfterViewInit()`, in regular methods, in constructors of plain classes not managed by Angular's DI system, the injection context is not available, and `inject()` will throw a runtime error: "NG0203: inject() must be called from an injection context." Also, even in allowed places, the injection context is lost if inside a nested callback or after an `await`. To fix this, move `inject()` into an injection context.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  ngOnInit() {
    inject(SomeService);
    ~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  constructor() {}
  ngAfterViewInit() {
    inject(SomeService);
    ~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  anyMethod() {
    inject(SomeService);
    ~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive()
class Test {
  anyMethod() {
    inject(SomeService);
    ~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Pipe()
class Test {
  anyMethod() {
    inject(SomeService);
    ~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Injectable()
class TestService {
  loadData() {
    inject(SomeService);
    ~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test {
  constructor() {
    inject(SomeService);
    ~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test {
  private service = inject(SomeService);
                    ~~~~~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  constructor() {
    someFunction(() => {
      inject(SomeService);
      ~~~~~~~~~~~~~~~~~~~
    });
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  constructor() {
    someFunction(() => inject(SomeService));
                       ~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  private service = someFunction(() => {
    inject(SomeService);
    ~~~~~~~~~~~~~~~~~~~
  });
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  private service = someFunction(() => inject(SomeService));
                                       ~~~~~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const myGuard: CanActivateFn = () => {
  someFunction(() => {
    inject(SomeService);
    ~~~~~~~~~~~~~~~~~~~
  });
  return true;
};
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const myGuard: CanActivateFn = async () => {
  await someAsyncFunction();
  return inject(SomeService).isOK;
         ~~~~~~~~~~~~~~~~~~~
};
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const myFunction: TestFn = () => {
  inject(SomeService);
  ~~~~~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Injectable({
  providedIn: 'root',
})
class MyService {
  canActivate(): boolean {
    return inject(SomeService).isOK;
           ~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const test = {
  canActivate: [(): boolean => {
    return inject(SomeService).isOK;
           ~~~~~~~~~~~~~~~~~~~
  }],
};
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const myObject = {
  factory: () => inject(SomeService),
                 ~~~~~~~~~~~~~~~~~~~
};
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const myObject = {
  useFactory: () => inject(SomeService),
                    ~~~~~~~~~~~~~~~~~~~
};
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Injectable({
  providedIn: 'root',
})
export class MyService {
  someMethod() {
    runInInjectionContext(this.environmentInjector, () => {
      someFunction(() => {
        inject(SomeService);
        ~~~~~~~~~~~~~~~~~~~
      });
    });
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Injectable({
  providedIn: 'root',
})
export class MyService {
  someMethod() {
    someFunction(() => {
      inject(SomeService);
      ~~~~~~~~~~~~~~~~~~~
    });
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
inject(SomeService);
~~~~~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  constructor() {
    inject(SomeService);
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive()
class Test {
  constructor() {
    inject(SomeService);
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Pipe({ name: 'myPipe' })
class MyPipe {
  constructor() {
    inject(SomeService);
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Injectable()
class Test {
  constructor() {
    inject(SomeService);
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@NgModule()
class MyModule {
  constructor() {
    inject(SomeService);
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  private service = inject(SomeService);
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive()
class Test {
  private service = inject(SomeService);
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Pipe({ name: 'myPipe' })
class MyPipe {
  private service = inject(SomeService);
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Injectable()
class TestService {
  private service = inject(SomeService);
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@NgModule()
class MyPipe {
  private service = inject(SomeService);
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const MY_TOKEN = new InjectionToken('my-token', {
  factory: () => {
    return inject(SomeService).data;
  }
});
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const MY_TOKEN = new InjectionToken('my-token', {
  factory: () => inject(SomeService).data,
});
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  providers: [
    {
      provide: SomeToken,
      useFactory: () => {
        return inject(SomeService).data;
      },
    }
  ]
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  providers: [
    {
      provide: SomeToken,
      useFactory: () => inject(SomeService).data,
    }
  ]
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Injectable({
  useFactory: () => {
    inject(SomeService);
    return () => {};
  },
})
class MyService {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Injectable({
  useFactory: () => inject(SomeService).myFunction,
})
class MyService {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const myGuard: CanActivateFn = () => {
  return inject(SomeService).isOK;
};
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const myGuard: CanActivateFn = () => inject(SomeService).isOK;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const myGuard: CanActivateChildFn = () => {
  return inject(SomeService).isOK;
};
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const myGuard: CanDeactivateFn<MyComponent> = () => {
  return inject(SomeService).isOK;
};
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const myGuard: CanMatchFn = () => {
  return inject(SomeService).isOK;
};
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const myResolver: ResolveFn<MyData> = () => {
  return inject(SomeService).data;
};
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const canRun: RunGuardsAndResolvers = () => inject(SomeService).canRun;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const children: LoadChildren = () => {
  inject(SomeService);
  return [];
};
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const children: LoadChildrenCallback = () => {
  inject(SomeService);
  return [];
};
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
export const myInterceptor: HttpInterceptorFn = (req: HttpRequest<unknown>, next: HttpHandlerFn) => {
  inject(SomeService);
  return next(req);
};
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const options: ViewTransitionsFeatureOptions = {
  onViewTransitionCreated: () => {
    inject(SomeService);
  },
};
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Injectable({
  providedIn: 'root',
})
export class MyGuard implements CanActivate {
  canActivate(): boolean {
    return inject(SomeService).isOK;
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Injectable({
  providedIn: 'root',
})
export class MyGuard implements CanActivateChild {
  canActivateChild(): boolean {
    return inject(SomeService).isOK;
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Injectable({
  providedIn: 'root',
})
export class MyGuard implements CanMatch {
  canMatch(): boolean {
    return inject(SomeService).isOK;
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Injectable({
  providedIn: 'root',
})
export class MyGuard implements CanDeactivate<SomeComponent> {
  canDeactivate(): boolean {
    return inject(SomeService).isOK;
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Injectable({
  providedIn: 'root',
})
export class MyGuard implements Resolve<SomeData> {
  resolve(): SomeData {
    return inject(SomeService).data;
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Injectable({
  providedIn: 'root',
})
export class MyGuard implements HttpInterceptor {
  intercept(req: HttpRequest<unknown>, next: HttpHandlerFn) {
    inject(SomeService);
    return next(req);
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const routes: Routes = [
  { path: 'test', canActivate: [(): boolean => {
    return inject(SomeService).isOK;
  }] },
];
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const routes: Routes = [
  { path: 'test', canActivateChild: [(): boolean => {
    return inject(SomeService).isOK;
  }] },
];
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const routes: Routes = [
  { path: 'test', canMatch: [(): boolean => {
    return inject(SomeService).isOK;
  }] },
];
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const routes: Routes = [
  { path: 'test', canDeactivate: [(component: SomeComponent): boolean => {
    return inject(SomeService).isOK;
  }] },
];
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const routes: Routes = [
  { path: 'test', resolve: [(): SomeData => {
    return inject(SomeService).data;
  }] },
];
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const routes: Routes = [
  { path: 'test', runGuardsAndResolvers: [() => inject(SomeService).canRun] },
];
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const routes: Routes = [
  {
    path: 'test',
    loadChildren: () => {
      inject(SomeService);
      return [];
    },
  },
];
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const routes: Routes = [
  {
    path: 'test',
    redirectTo: () => {
      inject(SomeService);
      return '';
    },
  },
];
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const route: Route = {
  path: 'test',
  canActivate: [(): boolean => {
    return inject(SomeService).isOK;
  }],
};
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Injectable({
  providedIn: 'root',
})
export class MyService {
  private environmentInjector = inject(EnvironmentInjector);
  someMethod() {
    runInInjectionContext(this.environmentInjector, () => {
      inject(SomeService);
    });
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
bootstrapApplication(App, {
  providers: [
    provideAppInitializer(() => {
      inject(SomeService);
    }),
  ],
});
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const platformRef = platformBrowser([
  providePlatformInitializer(() =>  {
    inject(SomeService);
  }),
]);
bootstrapApplication(App, appConfig, { platformRef })
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
createEnvironmentInjector(
  [
    provideEnvironmentInitializer(() => {
      inject(SomeService);
    }),
  ],
  parentInjector,
);
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
bootstrapApplication(App, {
  providers: [
    provideRouter(
      routes,
      withViewTransitions({
        onViewTransitionCreated: () => {
          inject(SomeService);
        },
      }),
    ),
  ],
});
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
function customOperator() {
  assertInInjectionContext(customOperator);
  inject(SomeService);
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const myGuard: CanActivateFn = async () => {
  const someService = inject(SomeService);
  await someAsyncFunction();
  return someService.isOK;
};
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-inject-outside-di-context": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
TestBed.inject(SomeService);
```

</details>

<br>
