<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-input-prefix.ts
  - ../../tests/rules/no-input-prefix/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/no-input-prefix`

Ensures that input bindings, including aliases, are not named or prefixed by the configured disallowed prefixes

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  prefixes?: string[];
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```ts
@Component({
  inputs: ['on']
           ~~~~
})
class Test {}
```

```ts
@Directive({
  outputs: [onCredit],
  inputs: [onLevel, `test: on`, onFunction()],
                    ~~~~~~~~~~
})
class Test {}
```

```ts
@Component({
  'inputs': ['onTest: test', ...onArray],
             ~~~~~~~~~~~~~~
})
class Test {}
```

```ts
@Directive()
class Test {
  @Input() on: EventEmitter<any> = new EventEmitter<{}>();
           ~~
}
```

```ts
@Component()
class Test {
  @Input() @Custom('on') 'onPrefix' = new EventEmitter<void>();
                         ~~~~~~~~~~
}
```

```ts
@Directive()
class Test {
  @Custom() @Input(`on`) _on = getInput();
                   ~~~~
}
```

```ts
@Component()
class Test {
  @Input('onPrefix') _on = (this.subject$ as Subject<{on: boolean}>).pipe();
         ~~~~~~~~~~
}
```

```ts
@Directive()
class Test {
  @Input('setter') set 'on-setter'() {}
                       ~~~~~~~~~~~
}
```

```ts
@Injectable()
class Test {
  @Input('on') isPrefix = this.getInput();
         ~~~~  ~~~~~~~~
}
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```ts
@Directive({
  selector: 'foo',
})
class Test {
  @Input() set 'setter'() {}
}
```
