<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-lifecycle-call.ts
  - ../../tests/rules/no-lifecycle-call/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/no-lifecycle-call`

Disallows explicit calls to lifecycle methods

- Type: suggestion

<br>

## Rationale

Lifecycle hooks like ngOnInit(), ngOnDestroy(), and ngOnChanges() are designed to be called automatically by Angular at the appropriate times in a component's lifecycle. Calling them manually (e.g., 'this.ngOnInit()') breaks Angular's lifecycle management, can cause hooks to run multiple times or in the wrong order, and makes the code's execution flow harder to understand. The only exception is calling super.ngOnInit() in a derived class to ensure the parent class's initialization runs. If you need to share initialization logic, extract it into a separate method that both ngOnInit() and your other code can call.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-lifecycle-call": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  test(): void {
    this.ngAfterContentChecked();
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-lifecycle-call": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive()
class Test {
  test(): void {
    this.ngAfterContentInit();
    ~~~~~~~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-lifecycle-call": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Injectable()
class Test {
  test(): void {
    this.ngAfterViewChecked();
    ~~~~~~~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-lifecycle-call": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@NgModule()
class Test {
  test(): void {
    this.ngAfterViewInit();
    ~~~~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-lifecycle-call": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Pipe()
class Test {
  test(): void {
    this.ngDoBootstrap();
    ~~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-lifecycle-call": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  test(): void {
    this.ngDoCheck();
    ~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-lifecycle-call": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive()
class Test {
  test(): void {
    this.ngOnChanges();
    ~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-lifecycle-call": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Injectable()
class Test {
  test(): void {
    this.ngOnDestroy();
    ~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-lifecycle-call": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@NgModule()
class Test {
  test(): void {
    this.ngOnInit();
    ~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-lifecycle-call": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({ template: '' })
class Test extends ParentComponent {
  test(): void {
    super.ngOnChanges();
    ~~~~~~~~~~~~~~~~~~~
  }
}
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-lifecycle-call": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  ngAfterContentChecked(): void {
    super.ngAfterContentChecked();
  }
}

@Directive()
class Test {
  test(): void {
    this.ngAfterContentChecked1();
    this.angAfterContentInit();
    this.ngAfterViewChecked2();
    this.ngAfterViewInit3();
    this.ngOnChange$();
    this.ngOnDestroyx();
    this.ngOnInitialize();
    this.ngDoChecking();
    ngOnInit();
  }
}

@MyDecorator()
class Test {
  test(): void {
    ngDoCheck();
  }
}

ngOnDestroy();
```

</details>

<br>
