<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-lifecycle-call.ts
  - ../../tests/rules/no-lifecycle-call/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/no-lifecycle-call`

Disallows explicit calls to lifecycle methods

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```ts
@Component()
class Test {
  test(): void {
    this.ngAfterContentChecked();
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  }
}
```

```ts
@Directive()
class Test {
  test(): void {
    this.ngAfterContentInit();
    ~~~~~~~~~~~~~~~~~~~~~~~~~
  }
}
```

```ts
@Injectable()
class Test {
  test(): void {
    this.ngAfterViewChecked();
    ~~~~~~~~~~~~~~~~~~~~~~~~~
  }
}
```

```ts
@NgModule()
class Test {
  test(): void {
    this.ngAfterViewInit();
    ~~~~~~~~~~~~~~~~~~~~~~
  }
}
```

```ts
@Pipe()
class Test {
  test(): void {
    this.ngDoBootstrap();
    ~~~~~~~~~~~~~~~~~~~~
  }
}
```

```ts
@Component()
class Test {
  test(): void {
    this.ngDoCheck();
    ~~~~~~~~~~~~~~~~
  }
}
```

```ts
@Directive()
class Test {
  test(): void {
    this.ngOnChanges();
    ~~~~~~~~~~~~~~~~~~
  }
}
```

```ts
@Injectable()
class Test {
  test(): void {
    this.ngOnDestroy();
    ~~~~~~~~~~~~~~~~~~
  }
}
```

```ts
@NgModule()
class Test {
  test(): void {
    this.ngOnInit();
    ~~~~~~~~~~~~~~~
  }
}
```

```ts
@Component({ template: '' })
class Test extends ParentComponent {
  test(): void {
    super.ngOnChanges();
    ~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```ts
@Component()
class Test {
  ngAfterContentChecked(): void {
    super.ngAfterContentChecked();
  }
}

@Directive()
class Test {
  test(): void {
    this.ngAfterContentChecked1();
    this.angAfterContentInit();
    this.ngAfterViewChecked2();
    this.ngAfterViewInit3();
    this.ngOnChange$();
    this.ngOnDestroyx();
    this.ngOnInitialize();
    this.ngDoChecking();
    ngOnInit();
  }
}

@MyDecorator()
class Test {
  test(): void {
    ngDoCheck();
  }
}

ngOnDestroy();
```
