<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-outputs-metadata-property.ts
  - ../../tests/rules/no-outputs-metadata-property/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/no-outputs-metadata-property`

Disallows usage of the `outputs` metadata property

- Type: suggestion

<br>

## Rationale

Using the 'outputs' metadata property (@Component({ outputs: ['change'] })) is discouraged in favor of the @Output() decorator because: (1) @Output() makes it immediately clear which properties are event emitters when reading the class, (2) @Output() allows you to rename or configure the output in one place right next to the property, (3) @Output() works better with IDE features like Find References and Rename Refactoring, and (4) the decorator syntax is more consistent with modern Angular development practices. The metadata property obscures the component's interface and requires looking in two places (decorator metadata and class body) to understand the component's public API.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-outputs-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({
  outputs: [
  ~~~~~~~~~~
    'id: foo'
  ],
  ~
  selector: 'app-test'
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-outputs-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive({
  outputs: [
  ~~~~~~~~~~
    'id: foo'
  ],
  ~
  selector: 'app-test'
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-outputs-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({
  outputs,
  ~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-outputs-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive({
  outputs: [],
  ~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-outputs-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const test = [];
@Component({
  'outputs': test,
  ~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-outputs-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive({
  ['outputs']: undefined,
  ~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-outputs-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
function outputs() {
  return [];
}

@Component({
  [`outputs`]: outputs(),
  ~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-outputs-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-outputs-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-outputs-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-outputs-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const options = {};
@Component(options)
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-outputs-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({
  selector: 'app-test',
  template: 'Hello'
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-outputs-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  selector: 'app-test',
  queries: {},
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-outputs-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const outputs = 'providers';
@Directive({
  [outputs]: [],
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-outputs-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@NgModule({
  bootstrap: [Foo]
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-outputs-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  selector: 'foo',
  hostDirectives: [{
    directive: CdkMenuItem,
    outputs: ['cdkMenuItemTriggered: triggered'],
  }]
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-outputs-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  selector: 'foo',
  'hostDirectives': [{
    directive: CdkMenuItem,
    outputs: ['cdkMenuItemTriggered: triggered'],
  }]
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-outputs-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  selector: 'foo',
  ['hostDirectives']: [{
    directive: CdkMenuItem,
    outputs: ['cdkMenuItemTriggered: triggered'],
  }]
})
class Test {}
```

</details>

<br>
