<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-output-native.ts
  - ../../tests/rules/no-output-native/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/no-output-native`

Ensures that output bindings, including aliases, are not named as standard DOM events

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```ts
@Component({
  'outputs': ['pagehide']
              ~~~~~~~~~~
})
class Test {}
```

```ts
@Directive({
  inputs: ['abort'],
  ['outputs']: [boundary, `test: copy`],
                          ~~~~~~~~~~~~
})
class Test {}
```

```ts
@Component({
  inputs: ['abort'],
  [`outputs`]: [boundary, `test: copy`],
                          ~~~~~~~~~~~~
})
class Test {}
```

```ts
@Directive({
  outputs: ['orientationchange: orientation'],
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

```ts
@Component()
class Test {
  @Output() change: EventEmitter<any> = new EventEmitter<{}>();
            ~~~~~~
}
```

```ts
@Directive()
class Test {
  @Output() @Custom('change') 'change' = new EventEmitter<void>();
                              ~~~~~~~~
}
```

```ts
@Component()
class Test {
  @Custom() @Output(`change`) _change = getOutput();
                    ~~~~~~~~
}
```

```ts
@Directive()
class Test {
  @Output('change') _change = (this.subject$ as Subject<{blur: boolean}>).pipe();
          ~~~~~~~~
}
```

```ts
@Component()
class Test {
  @Output('getter') get 'cut'() {}
                        ~~~~~
}
```

```ts
@Injectable()
class Test {
  @Output('click') blur = this.getOutput();
          ~~~~~~~  ~~~~
}
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```ts
class Test {}
```

```ts
@Page({
  outputs: ['play', popstate, `online`, 'obsolete: obsol', 'store: storage'],
})
class Test {}
```

```ts
@Component()
class Test {
  change = new EventEmitter();
}
```

```ts
@Directive()
class Test {
  @Output() buttonChange = new EventEmitter<'change'>();
}
```

```ts
@Component()
class Test {
  @Output() Drag = new EventEmitter<{ click: string }>();
}
```

```ts
@Directive()
class Test {
  @Output(`changelower`) changeText = new EventEmitter<{ bar: string, blur: string }>();
}
```

```ts
@Component()
class Test {
  @Output('buttonChange') changelower = new EventEmitter<ComplextObject>();
}
```

```ts
@Directive()
class Test<SVGScroll> {
  @Output() SVgZoom = new EventEmitter<SVGScroll>();
}
```

```ts
const change = 'change';
@Component()
class Test {
  @Output(change) touchMove: EventEmitter<{ action: 'click' | 'close' }> = new EventEmitter<{ action: 'click' | 'close' }>();
}
```

```ts
const blur = 'blur';
const click = 'click';
@Directive()
class Test {
  @Output(blur) [click]: EventEmitter<Blur>;
}
```

```ts
@Directive({
  selector: 'foo',
})
class Test {
  @Output() get 'getter'() {}
}
```
