<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-pipe-impure.ts
  - ../../tests/rules/no-pipe-impure/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/no-pipe-impure`

Disallows the declaration of impure pipes

- Type: suggestion

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-pipe-impure": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Pipe({
  pure: false
  ~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-pipe-impure": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Pipe({
  name: 'test',
  'pure': false,
  ~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-pipe-impure": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Pipe({
  name: 'test',
  ['pure']: !true
  ~~~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-pipe-impure": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Pipe({
  name: 'test',
  [`pure`]: false
  ~~~~~~~~~~~~~~~
})
class Test {}
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-pipe-impure": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-pipe-impure": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Pipe()
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-pipe-impure": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Pipe({})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-pipe-impure": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
const options = {};
@Pipe(options)
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-pipe-impure": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Pipe({
  name: 'test',
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-pipe-impure": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Pipe({
  pure: true
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-pipe-impure": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Pipe({
  'pure': !0,
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-pipe-impure": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Pipe({
  ['pure']: !!isPure(),
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-pipe-impure": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
const pure = 'pure';
@Pipe({
  [pure]: false
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-pipe-impure": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
const pure = false;
@Pipe({
  pure,
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-pipe-impure": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
function isPure() {
  return false;
}

@Pipe({
  [`pure`]: isPure(),
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-pipe-impure": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@NgModule({
  bootstrap: [Foo]
})
class Test {}
```

</details>

<br>
