<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-pipe-impure.ts
  - ../../tests/rules/no-pipe-impure/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/no-pipe-impure`

Disallows the declaration of impure pipes

- Type: suggestion
- Category: Best Practices

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

‚ùå - Examples of **incorrect** code for this rule:

```ts
@Pipe({
  pure: false
  ~~~~~~~~~~~
})
class Test {}
```

```ts
@Pipe({
  name: 'test',
  'pure': false,
  ~~~~~~~~~~~~~
})
class Test {}
```

```ts
@Pipe({
  name: 'test',
  'pure': !true
  ~~~~~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

‚úÖ - Examples of **correct** code for this rule:

```ts
class Test {}
```

```ts
@Pipe()
class Test {}
```

```ts
@Pipe({})
class Test {}
```

```ts
const options = {};
@Pipe(options)
class Test {}
```

```ts
@Pipe({
  name: 'test',
})
class Test {}
```

```ts
@Pipe({
  pure: true
})
class Test {}
```

```ts
@Pipe({
  pure: !0,
})
class Test {}
```

```ts
@Pipe({
  pure: !!isPure(),
})
class Test {}
```

```ts
const pure = 'pure';
@Pipe({
  [pure]: false
})
class Test {}
```

```ts
const pure = false;
@Pipe({
  pure,
})
class Test {}
```

```ts
function isPure() {
  return false;
}

@Pipe({
  pure: isPure(),
})
class Test {}
```

```ts
@NgModule({
  bootstrap: [Foo]
})
class Test {}
```
