<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-queries-metadata-property.ts
  - ../../tests/rules/no-queries-metadata-property/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/no-queries-metadata-property`

Disallows usage of the `queries` metadata property. See more at https://angular.io/styleguide#style-05-12.

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```ts
@Component({
  queries: {
  ~~~~~~~~~~
    contentChild: new ContentChild(ChildDirective),
    contentChildren: new ContentChildren(ChildDirective),
    viewChild: new ViewChild(ChildDirective),
    viewChildren: new ViewChildren(ChildDirective)
  },
  ~
  selector: 'app-test'
})
class TestComponent {}
```

```ts
@Directive({
  queries: {
  ~~~~~~~~~
    contentChild: new ContentChild(ChildDirective),
    contentChildren: new ContentChildren(ChildDirective),
    viewChild: new ViewChild(ChildDirective),
    viewChildren: new ViewChildren(ChildDirective)
  },
  ~
  selector: 'app-test'
})
class TestDirective {}
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```ts
@Component({
  selector: 'app-test',
  template: 'Hello'
})
class TestComponent {}
```

```ts
@Directive({
  selector: 'app-test'
})
class TestDirective {}
```
