<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-queries-metadata-property.ts
  - ../../tests/rules/no-queries-metadata-property/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/no-queries-metadata-property`

Disallows usage of the `queries` metadata property. See more at https://angular.io/styleguide#style-05-12.

- Type: suggestion

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-queries-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({
  queries: {
  ~~~~~~~~~~
    contentChild: new ContentChild(ChildDirective),
    contentChildren: new ContentChildren(ChildDirective),
    viewChild: new ViewChild(ChildDirective),
    viewChildren: new ViewChildren(ChildDirective)
  },
  ~
  selector: 'app-test'
})
class TestComponent {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-queries-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive({
  queries: {
  ~~~~~~~~~
    contentChild: new ContentChild(ChildDirective),
    contentChildren: new ContentChildren(ChildDirective),
    viewChild: new ViewChild(ChildDirective),
    viewChildren: new ViewChildren(ChildDirective)
  },
  ~
  selector: 'app-test'
})
class TestDirective {}
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-queries-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  selector: 'app-test',
  template: 'Hello'
})
class TestComponent {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-queries-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({
  selector: 'app-test'
})
class TestDirective {}
```

</details>

<br>
