<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-signal-mutable-updates.ts
  - ../../tests/rules/no-signal-mutable-updates/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/no-signal-mutable-updates`

Disallow mutable updates to signals. Signals should be read-only and WritableSignals should only be updated via .set() or .update() methods

- Type: problem

<br>

## Rationale

Angular signals are designed to be immutable references to reactive state. When you have a Signal (read-only), you should only read its value by calling it - never mutate the returned value. When you have a WritableSignal, you should update it only through the provided .set() and .update() methods, which properly notify the reactivity system of changes. Mutating signal values directly (like mySignal().property = value or mySignal().push(item)) breaks the reactivity system because Angular cannot detect these changes. This leads to bugs where the UI does not update when the underlying data changes. For WritableSignals, always use .set() to replace the entire value or .update() to compute a new value based on the current one. This ensures Angular tracks all changes and updates dependent computed signals and effects correctly.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let userData: Signal<{ name: string }>;
userData().name = 'newName';
~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let userData: WritableSignal<{ name: string }>;
userData().name = 'newName';
~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let userData: WritableSignal<{ profile: { name: string } }>;
userData().profile.name = 'newName';
~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let counter: Signal<{ count: number }>;
counter().count++;
~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let counter: WritableSignal<{ count: number }>;
counter().count++;
~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let counter: WritableSignal<{ count: number }>;
++counter().count;
  ~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let items: Signal<number[]>;
items().push(4);
~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let items: WritableSignal<number[]>;
items().push(4);
~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let items: WritableSignal<number[]>;
items().pop();
~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let items: WritableSignal<number[]>;
items().splice(0, 1);
~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let items: WritableSignal<number[]>;
items().sort();
~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let items: WritableSignal<number[]>;
items().reverse();
~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let items: WritableSignal<number[]>;
items().unshift(0);
~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let items: WritableSignal<number[]>;
items().shift();
~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let items: WritableSignal<number[]>;
items().fill(0);
~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let value: ModelSignal<{ name: string }>;
value().name = 'newName';
~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let items: ModelSignal<number[]>;
items().push(4);
~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let value: InputSignal<{ name: string }>;
value().name = 'newName';
~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let counter: WritableSignal<{ count: number }>;
counter().count += 5;
~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let counter: WritableSignal<{ count: number }>;
counter().count--;
~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
export class AppComponent {
  readonly userData: WritableSignal<{ name: string }>;

  updateName() {
    this.userData().name = 'newName';
    ~~~~~~~~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let items: WritableSignal<number[]>;
items().copyWithin(0, 1);
~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let value: InputSignalWithTransform<{ name: string }, string>;
value().name = 'newName';
~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
let obj = { name: 'test' };
obj.name = 'newName';
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
let arr = [1, 2, 3];
arr.push(4);
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
let num = 5;
num++;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
let userData: WritableSignal<{ name: string }>;
userData.set({ name: 'newName' });
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
let userData: WritableSignal<{ name: string }>;
userData.update(current => ({ ...current, name: 'newName' }));
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
let items: WritableSignal<number[]>;
items.set([...items(), 4]);
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
let items: WritableSignal<number[]>;
items.update(current => [...current, 4]);
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
let userData: Signal<{ name: string }>;
const name = userData().name;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
let userData: WritableSignal<{ name: string }>;
const name = userData().name;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
let items: Signal<number[]>;
const length = items().length;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
let items: Signal<number[]>;
const mapped = items().map(x => x * 2);
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
let items: Signal<number[]>;
const filtered = items().filter(x => x > 0);
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
let items: WritableSignal<number[]>;
const sliced = items().slice(0, 2);
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
let value: ModelSignal<{ name: string }>;
value.set({ name: 'newName' });
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
let value: ModelSignal<{ name: string }>;
value.update(current => ({ ...current, name: 'newName' }));
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
let value: InputSignal<{ name: string }>;
const name = value().name;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
let items: Signal<number[]>;
const includes = items().includes(5);
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
let userData: WritableSignal<{ profile: { name: string } }>;
userData.update(current => ({
  ...current,
  profile: { ...current.profile, name: 'newName' }
}));
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-signal-mutable-updates": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
let items: WritableSignal<number[]>;
const currentItems = items();
const newItems = [...currentItems, 4];
items.set(newItems);
```

</details>

<br>
