<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-uncalled-signals.ts
  - ../../tests/rules/no-uncalled-signals/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/no-uncalled-signals`

Warns user about unintentionally doing logic on the signal, rather than the signal's value

- Type: suggestion

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
const aSignal = createSignal();
if (aSignal) {
    ~~~~~~~
}
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
const aSignal = createSignal(false);
if (aSignal || true) {
    ~~~~~~~
}
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
const aSignal = createSignal("hello");
if (aSignal == "hello") {
    ~~~~~~~
}
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
const aSignal = createSignal();
if (false || (aSignal ?? true)) {
              ~~~~~~~
}
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
const aSignal = createSignal();
const v = aSignal ? true : false;
          ~~~~~~~
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
const aSignal = createSignal(false);
const v = (aSignal || true) ? true : false;
           ~~~~~~~
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
const aSignal = createSignal("hello");
const v = (aSignal == "hello") ? true : false;
           ~~~~~~~
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
const aSignal = createSignal();
const v = (false || (aSignal ?? true)) ? true : false;
                     ~~~~~~~
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
const aSignal = createSignal();
const v = aSignal ?? true;
          ~~~~~~~
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
const aSignal = createSignal();
const v = aSignal || true;
          ~~~~~~~
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
const arbitraryVar = 1;
if (arbitraryVar) {
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
const aSignal = createSignal();
if (aSignal()) {
}
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
const aSignal = createSignal();
if (aSignal() || true) {
}
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
const aSignal = createSignal();
if (aSignal() == "hello") {
}
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
const aSignal = createSignal();
if (false || (aSignal() ?? true)) {
}
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
const aSignal = createSignal();
if (false) {
  aSignal
}
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let aSignal: Signal | null = createSignal();
if (aSignal) {
}
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let aSignal: Signal | undefined = createSignal();
if (aSignal) {
}
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let aSignal: Signal | NonSignal = createSignal();
if (aSignal) {
}
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
interface NonSignal {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
const aSignal = createSignal();
const v = aSignal() ?? true;
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
const aSignal = createSignal();
const v = aSignal() || true;
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
const aSignal = createSignal();
const v = aSignal;
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
export class AppComponent {
  readonly test = signal<boolean>(false);

  constructor() {
    effect(() => {
      if (this.test()) {
        console.log('Hey');
      } else {
        console.log('Hoo');
      }
    });
  }
}
declare function signal<T>(value: T): Signal<T>;
declare function effect(fn: () => void): void;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
export class AppComponent {
  readonly test = signal<boolean>(false);

  constructor() {
    const t = this.test;
    effect(() => {
      if (t()) {
        console.log('Hey');
      } else {
        console.log('Hoo');
      }
    });
  }
}
declare function signal<T>(value: T): Signal<T>;
declare function effect(fn: () => void): void;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a: Signal<string>;
let b: boolean;
let c = b && a.set('');

interface Signal<T> {
  set(value: T): void;
}
```

</details>

<br>
