<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-uncalled-signals.ts
  - ../../tests/rules/no-uncalled-signals/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/no-uncalled-signals`

Warns user about unintentionally doing logic on the signal, rather than the signal's value

- Type: suggestion

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
let a: Signal<boolean>;
let b = a ? 1 : 2;
        ~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
let a: Signal<boolean>;
do {} while (a);
             ~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
let a: Signal<boolean>;
for (let i = 0; a; i++) { }
                ~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
let a: Signal<boolean>;
if (a) { }
    ~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
let a: Signal<boolean>;
switch (true) {
  case a:
       ~
    break;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
let a: Signal<boolean>;
while (a) {}
       ~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
let a: Signal<boolean>;
if (!a) { }
     ~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
let a: Signal<number>;
if (a === 1) { }
    ~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
let a: Signal<number>;
if (1 === a) { }
          ~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
let a: Signal<number>;
let b = a || 1;
        ~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
let a: Signal<number>;
let b = 1 || a;
             ~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
let a: Signal<number>;
let b = 1 || 2 || 3 || a || 4 || 5;
                       ~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
function getSignal(): Signal<number> { }
if (getSignal()) { }
    ~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
let a: { b: { c: { d: { e: Signal<number>; } } } }
if (a.b.c.d.e) { }
    ~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
let a: InputSignal<boolean>;
if (a) { }
    ~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
let a: ModelSignal<boolean>;
if (a) { }
    ~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
let a: WritableSignal<boolean>;
if (a) { }
    ~
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a = 1;
if (a) { }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a = true;
if (!a) { }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a: Signal<boolean>;
let b = a() ? 1 : 2;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a: Signal<boolean>;
do {} while (a());
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a: Signal<boolean>;
for (let i = 0; a(); i++) { }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a: Signal<boolean>;
for (;;) { } // Tests for loop without 'test' node.
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a: Signal<boolean>;
if (a()) { }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a: Signal<boolean>;
switch (true) {
  case a():
    break;
  default: // Tests switch case without 'test' node.
    break;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a: Signal<boolean>;
while (a()) { }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a: Signal<boolean>;
if (!a()) { }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a: Signal<boolean>;
if (a() || true) { }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a: Signal<boolean>;
if (a() == "hello") { }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a: Signal<boolean>;
if (false || (a() ?? true)) { }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a: Signal<boolean>;
if (false) {
  a
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a: Signal<boolean> | null;
if (a) { }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a: Signal<boolean> | undefined;
if (a) { }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a: Signal<boolean> | NonSignal;
if (a) { }
interface NonSignal {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a: Signal<boolean>;
let b = a() ?? true;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a: Signal<boolean>;
const b = a() || true;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a: Signal<boolean>;
let b = a;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
function getSignal(): Signal<boolean> {}
if (getSignal()()) { }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
export class AppComponent {
  readonly test: Signal<boolean>;

  constructor() {
    effect(() => {
      if (this.test()) { }
    });
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
export class AppComponent {
  readonly test: Signal<boolean>;

  constructor() {
    const t = this.test;
    effect(() => {
      if (t()) { }
    });
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a: WritableSignal<string>;
let b: boolean;
let c = b && a.set('');
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
let a: { b: WritableSignal<boolean> };
true && a.b.set(false);
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
function a(b: WritableSignal<boolean> | InputSignal<boolean>) {
  return 'set' in b || 'applyValueToInputSignal' in b[SIGNAL];
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/no-uncalled-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
function a(b: { c: WritableSignal<boolean> }) {
  true && ((x) => b.c.set(x))(true);
}
```

</details>

<br>
