<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/pipe-prefix.ts
  - ../../tests/rules/pipe-prefix/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/pipe-prefix`

Enforce consistent prefix for pipes.

- Type: suggestion
- Category: Stylistic Issues

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  prefixes?: string[];
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```ts
@Pipe({
  name: 'foo-bar'
        ~~~~~~~~~
})
class Test {}
```

```ts
@Pipe({
  name: 'foo-bar'
        ~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:
