<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/pipe-prefix.ts
  - ../../tests/rules/pipe-prefix/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/pipe-prefix`

Enforce consistent prefix for pipes.

- Type: suggestion

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * Default: `[]`
   */
  prefixes?: string[];
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/pipe-prefix": [
      "error",
      {
        "prefixes": [
          "ng"
        ]
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Pipe({
  name: 'foo-bar'
        ~~~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/pipe-prefix": [
      "error",
      {
        "prefixes": [
          "ng",
          "mg",
          "sg"
        ]
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Pipe({
  name: 'foo-bar'
        ~~~~~~~~~
})
class Test {}
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/pipe-prefix": [
      "error",
      {
        "prefixes": [
          "ng"
        ]
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Pipe
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/pipe-prefix": [
      "error",
      {
        "prefixes": [
          "ng"
        ]
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Pipe({})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/pipe-prefix": [
      "error",
      {
        "prefixes": [
          "ng"
        ]
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
export function mockPipe(name: string): any {
  @Pipe({ name })
  class MockPipe implements PipeTransform {
    transform(input: any): any {
      return input;
    }
  }
  return MockPipe;
}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/pipe-prefix": [
      "error",
      {
        "prefixes": []
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Pipe({
  name: 'ngBarFoo'
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/pipe-prefix": [
      "error",
      {
        "prefixes": [
          "ng"
        ]
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Pipe({
  name: 'ngBarFoo'
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/pipe-prefix": [
      "error",
      {
        "prefixes": [
          "ng",
          "sg",
          "mg"
        ]
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Pipe({
  name: 'ngBarFoo'
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/pipe-prefix": [
      "error",
      {
        "prefixes": [
          "ng",
          "sg",
          "mg"
        ]
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Pipe({
  name: \`ngBarFoo\`
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/pipe-prefix": [
      "error",
      {
        "prefixes": [
          "ng"
        ]
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/pipe-prefix": [
      "error",
      {
        "prefixes": [
          "ng"
        ]
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const pipeName = 'fooBar';
@Pipe({
  name: pipeName
})
class Test {}
```

</details>

<br>
