<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-on-push-component-change-detection.ts
  - ../../tests/rules/prefer-on-push-component-change-detection/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/prefer-on-push-component-change-detection`

Ensures component's `changeDetection` is set to `ChangeDetectionStrategy.OnPush`

- Type: suggestion

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rationale

By default, Angular's change detection checks every component on every change detection cycle, which can involve thousands of checks per second in a large application. OnPush change detection strategy is a performance optimization that tells Angular to only check a component when: (1) its input properties receive new references, (2) an event originates from the component or its children, or (3) change detection is manually triggered. This dramatically reduces the number of change detection runs, improving application performance. OnPush pairs well with immutable data patterns and Angular signals, and is considered a best practice for most components. However, you must be careful to use immutable data patterns (creating new object references when data changes) for OnPush to work correctly.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-on-push-component-change-detection": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component()
~~~~~~~~~~~~
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-on-push-component-change-detection": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
import type { ChangeDetectionStrategy } from '@angular/core';

@Component({})
~~~~~~~~~~~~~~
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-on-push-component-change-detection": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
import { Component } from '@angular/core';
const changeDetection = 'template';
@Component({ [changeDetection]: '' })
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-on-push-component-change-detection": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({ changeDetection: undefined })
                              ~~~~~~~~~
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-on-push-component-change-detection": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
import * as ng from '@angular/core';
@Component({ 'changeDetection': ChangeDetectionStrategy.Default })
                                                        ~~~~~~~
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-on-push-component-change-detection": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
import type { OnInit } from '@angular/core';
@Component({ ['changeDetection']: ChangeDetectionStrategy.Default })
                                                          ~~~~~~~
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-on-push-component-change-detection": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
import ng from '@angular/core';
@Component({ [`changeDetection`]: ChangeDetectionStrategy.Default })
                                                          ~~~~~~~
class Test {}
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-on-push-component-change-detection": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-on-push-component-change-detection": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
const options = {};
@Component(options)
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-on-push-component-change-detection": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-on-push-component-change-detection": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  'changeDetection': changeDetection,
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-on-push-component-change-detection": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
const changeDetection = ChangeDetectionStrategy.Default;
@Component({
  changeDetection,
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-on-push-component-change-detection": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
function changeDetection() {
  return ChangeDetectionStrategy.OnPush;
}

@Component({
  ['changeDetection']: changeDetection(),
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-on-push-component-change-detection": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  [`changeDetection`]: ChangeDetectionStrategy.OnPush,
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-on-push-component-change-detection": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@NgModule({
  bootstrap: [Foo]
})
class Test {}
```

</details>

<br>
