<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-inject.ts
  - ../../tests/rules/prefer-inject/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/prefer-inject`

Prefer using the inject() function over constructor parameter injection

- Type: suggestion

<br>

## Rationale

The inject() function is Angular's modern dependency injection API that offers several advantages over constructor-based injection. First, it enables dependency injection outside of constructors, allowing you to use DI in functions, factories, and even at the class field level. This makes code more flexible and composable. Second, inject() is more concise and reduces boilerplate - you don't need constructor parameter properties or manual field assignments. Third, inject() works naturally with modern TypeScript features and tree-shaking. Fourth, it's required for using many modern Angular features like functional guards, interceptors, and the new signal-based APIs. Angular provides an automated migration schematic (ng generate @angular/core:inject) to convert constructor injection to inject(), making adoption straightforward. This is the recommended approach for all new Angular code.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-inject": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Injectable()
class UserService {
  constructor(private http: HttpClient) {}
              ~~~~~~~~~~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-inject": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({})
class MyComponent {
  constructor(
    private userService: UserService,
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    private http: HttpClient
    ~~~~~~~~~~~~~~~~~~~~~~~~
  ) {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-inject": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Injectable()
class ConfigService {
  constructor(
    @Inject(CONFIG_TOKEN) private config: AppConfig,
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    @Optional() private logger?: LoggerService
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  ) {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-inject": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Injectable()
class WithPrimitiveToken {
  constructor(@Inject(LEVEL_TOKEN) public level: string) {}
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-inject": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({})
class MyComponent extends BaseComponent {
  constructor(
    private service: MyService,
    ~~~~~~~~~~~~~~~~~~~~~~~~~~
  ) {
    super();
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-inject": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({})
class MyComponent {
  constructor(elementRef: ElementRef) {}
              ~~~~~~~~~~~~~~~~~~~~~~
}
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-inject": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class PlainClass {
  constructor(private value: string) {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-inject": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Injectable()
class UserService {
  private http = inject(HttpClient);
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-inject": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({})
class MyComponent {
  constructor() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-inject": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({})
class MyComponent extends BaseComponent {
  constructor() {
    super();
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-inject": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Injectable()
class Logger {
  constructor(level: string) {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-inject": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Injectable()
class Logger {
  constructor(public level: string) {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-inject": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Injectable()
class Foo {
  constructor(foo: any, bar: unknown) {}
}
```

</details>

<br>
