<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-output-readonly.ts
  - ../../tests/rules/prefer-output-readonly/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/prefer-output-readonly`

Prefer to declare `@Output`, `OutputEmitterRef` and `OutputRef` as `readonly` since they are not supposed to be reassigned

- Type: suggestion

- ğŸ’¡ Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>âŒ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-readonly": [
      "error"
    ]
  }
}
```

<br>

#### âŒ Invalid Code

```ts
class Test {
  @Output() testEmitter = new EventEmitter<string>();
            ~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-readonly": [
      "error"
    ]
  }
}
```

<br>

#### âŒ Invalid Code

```ts
class Test {
  testEmitter: OutputEmitterRef<string>;
  ~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-readonly": [
      "error"
    ]
  }
}
```

<br>

#### âŒ Invalid Code

```ts
class Test {
  testEmitter = output();
  ~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-readonly": [
      "error"
    ]
  }
}
```

<br>

#### âŒ Invalid Code

```ts
class Test {
  testEmitter = output<string>();
  ~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-readonly": [
      "error"
    ]
  }
}
```

<br>

#### âŒ Invalid Code

```ts
class Test {
  testEmitter: OutputRef<string>;
  ~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-readonly": [
      "error"
    ]
  }
}
```

<br>

#### âŒ Invalid Code

```ts
class Test {
  testEmitter = outputFromObservable(testObservable);
  ~~~~~~~~~~~
}
```

</details>

<br>

---

<br>

<details>
<summary>âœ… - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-readonly": [
      "error"
    ]
  }
}
```

<br>

#### âœ… Valid Code

```ts
class Test {
  testEmitter = new EventEmitter<string>();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-readonly": [
      "error"
    ]
  }
}
```

<br>

#### âœ… Valid Code

```ts
class Test {
  @Output() readonly testEmitter = new EventEmitter<string>();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-readonly": [
      "error"
    ]
  }
}
```

<br>

#### âœ… Valid Code

```ts
class Test {
  testEmitter: NotOutputEmitterRef<string>;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-readonly": [
      "error"
    ]
  }
}
```

<br>

#### âœ… Valid Code

```ts
class Test {
  readonly testEmitter: OutputEmitterRef<string>;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-readonly": [
      "error"
    ]
  }
}
```

<br>

#### âœ… Valid Code

```ts
class Test {
  testEmitter = notOutput();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-readonly": [
      "error"
    ]
  }
}
```

<br>

#### âœ… Valid Code

```ts
class Test {
  readonly testEmitter = output();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-readonly": [
      "error"
    ]
  }
}
```

<br>

#### âœ… Valid Code

```ts
class Test {
  testEmitter = notOutput<string>();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-readonly": [
      "error"
    ]
  }
}
```

<br>

#### âœ… Valid Code

```ts
class Test {
  readonly testEmitter = output<string>();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-readonly": [
      "error"
    ]
  }
}
```

<br>

#### âœ… Valid Code

```ts
class Test {
  readonly testEmitter: OutputRef<string>;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-readonly": [
      "error"
    ]
  }
}
```

<br>

#### âœ… Valid Code

```ts
class Test {
  readonly testEmitter = outputFromObservable(testObservable);
}
```

</details>

<br>
