<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-signal-model.ts
  - ../../tests/rules/prefer-signal-model/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/prefer-signal-model`

Use `model` instead of `input` and `output` for two-way bindings

- Type: suggestion
- üîß Supports autofix (`--fix`)

<br>

## Rationale

The model() function is Angular's modern API for two-way bindings, combining both input and output into a single signal. When you have an input property paired with an output property that follows the naming pattern of `propertyChange` (e.g., `enabled` input with `enabledChange` output), this is the traditional pattern for two-way binding. The model() function provides a cleaner, more concise way to express this pattern with better type safety and integration with Angular's signal ecosystem. It eliminates the boilerplate of managing separate input and output properties while maintaining the same two-way binding functionality.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-model": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  readonly enabled = input();
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~
  readonly enabledChange = output();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-model": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  readonly enabled = input();
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~
  readonly enabledChange = output();
  readonly test = input();
  ~~~~~~~~~~~~~~~~~~~~~~~~
  readonly testChange = output();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-model": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  readonly value = input();
  ~~~~~~~~~~~~~~~~~~~~~~~~~
  readonly otherProp = input();
  readonly valueChange = output();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-model": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  readonly count = input<number>();
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  readonly countChange = output<number>();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-model": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  readonly enabledChange = output();
  readonly enabled = input();
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-model": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
import { model, input, output } from '@angular/core';

class Test {
  readonly enabled = input();
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~
  readonly enabledChange = output();
}
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-model": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly enabled = model();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-model": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly enabled = model<boolean>();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-model": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly enabled = input();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-model": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly enabledChange = output();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-model": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly enabled = input();
  readonly onChange = output();
}
```

</details>

<br>
