<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-signals.ts
  - ../../tests/rules/prefer-signals/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/prefer-signals`

Use readonly signals instead of `@Input()`, `@ViewChild()` and other legacy decorators

- Type: suggestion

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * Default: `true`
   */
  preferReadonlySignalProperties?: boolean;
  /**
   * Default: `true`
   */
  preferInputSignals?: boolean;
  /**
   * Default: `true`
   */
  preferQuerySignals?: boolean;
  useTypeChecking?: boolean;
  /**
   * Default: `[]`
   */
  additionalSignalCreationFunctions?: string[];
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal: Signal<number>;
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal: InputSignal<number>;
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal: ModelSignal<number>;
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal: WritableSignal<number>;
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = computed(() => 0);
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = contentChild('test');
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = contentChild.required('test');
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = contentChildren('test');
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = input('');
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = input.required('');
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = model(42);
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = model.required();
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = signal(42);
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = toSignal(source);
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = viewChild('test');
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = viewChild.required('test');
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = viewChildren('test');
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error",
      {
        "additionalSignalCreationFunctions": [
          "createSignal"
        ]
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = createSignal('test');
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error",
      {
        "useTypeChecking": true
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = createSignal();
  ~~~~~~~~~~
}
declare function createSignal(): Signal<boolean>;
interface Signal<T> {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  @Input()
  ~~~~~~~~
  value = 1;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  @ViewChild('test')
  ~~~~~~~~~~~~~~~~~~
  value: Widget;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  @ViewChildren('test')
  ~~~~~~~~~~~~~~~~~~~~~
  value: QueryList<Widget>;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  @ContentChild('test')
  ~~~~~~~~~~~~~~~~~~~~~
  value: Widget;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  @ContentChildren('test')
  ~~~~~~~~~~~~~~~~~~~~~~~~
  value: QueryList<Widget>;
}
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  testSubject = new Subject();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  testSubject = new ReplaySubject(1);
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  testValue = test();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  testValue: number;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  testValue: Widget<number>;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal: Signal<number>;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal: InputSignal<number>;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal: ModelSignal<number>;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal: WritableSignal<number>;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = computed(() => 0);
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = contentChild('test');
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = contentChild.required('test');
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = contentChildren('test');
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = input('');
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = input.required();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = model();
  readonly testRequired = model.required(42);
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = signal(true);
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = toSignal(source);
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = viewChild('test');
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = viewChild.required('test');
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = viewChildren('test');
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  testSignal = createSignal('test');
}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error",
      {
        "preferReadonlySignalProperties": false
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  testSignal = signal('test');
}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error",
      {
        "additionalSignalCreationFunctions": [
          "createSignal"
        ]
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = createSignal('test');
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  testSignal = createSignal();
}
function createSignal(): Signal<boolean> {
  return signal(true);
}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error",
      {
        "useTypeChecking": true
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = createSignal();
}
function createSignal(): Signal<boolean> {
  return signal(true);
}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error",
      {
        "useTypeChecking": true
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  testNotSignal = createNotSignal();
}
function createNotSignal(): NotSignal<boolean> {
  return true;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly value = input();
}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error",
      {
        "preferInputSignals": false
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  @Input()
  readonly value = 1;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly query = viewChild('test');
}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error",
      {
        "preferQuerySignals": false
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  @ViewChild('test')
  value: Widget;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly query = viewChildren('test');
}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error",
      {
        "preferQuerySignals": false
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  @ViewChildren('test')
  value: QueryList<Widget>;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly query = contentChild('test');
}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error",
      {
        "preferQuerySignals": false
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  @ContentChild('test')
  value: Widget;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly query = contentChildren('test');
}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signals": [
      "error",
      {
        "preferQuerySignals": false
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  @ContentChildren('test')
  value: QueryList<Widget>;
}
```

</details>

<br>
