<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-on-push-component-change-detection.ts
  - ../../tests/rules/prefer-on-push-component-change-detection/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/prefer-on-push-component-change-detection`

Enforces component's change detection to ChangeDetectionStrategy.OnPush.

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```ts
@Component({
  changeDetection: ChangeDetectionStrategy.Default
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
})
class Test {}
```

```ts
@Component({
~~~~~~~~~~~~
  selector: 'foo'
})
~~
class Test {}
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```ts
@Component({
  changeDetection: ChangeDetectionStrategy.OnPush
})
class Test {}
```
