<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-standalone-component.ts
  - ../../tests/rules/prefer-standalone-component/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/prefer-standalone-component`

## ‚ö†Ô∏è THIS RULE IS DEPRECATED

Please use `@angular-eslint/prefer-standalone` instead.

---

Ensures component `standalone` property is set to `true` in the component decorator

- Type: suggestion
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-standalone-component": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({})
~~~~~~~~~~~~~~
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-standalone-component": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({ standalone: false })
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-standalone-component": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
~~~~~~~~~~~~
standalone: false,
~~~~~~~~~~~~~~~~~~
template: '<div></div>'
~~~~~~~~~~~~~~~~~~~~~~~
})
~~
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-standalone-component": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
~~~~~~~~~~~~
template: '<div></div>'
~~~~~~~~~~~~~~~~~~~~~~~
})
~~
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-standalone-component": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
~~~~~~~~~~~~
selector: 'my-selector',
~~~~~~~~~~~~~~~~~~~~~~~
template: '<div></div>'
~~~~~~~~~~~~~~~~~~~~~~~
})
~~
class Test {}
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-standalone-component": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  standalone: true,
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-standalone-component": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  standalone: true,
  selector: 'test-selector'
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-standalone-component": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  selector: 'test-selector',
  standalone: true,
  template: '<div></div>',
  styleUrls: ['./test.css']
})
class Test {}
```

</details>

<br>
