<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-signal-readonly.ts
  - ../../tests/rules/prefer-signal-readonly/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/prefer-signal-readonly`

Prefer to declare `Signal` properties as `readonly` since they are not supposed to be reassigned

- Type: suggestion

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * Default: `[]`
   */
  signalCreationFunctions?: string[];
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal: Signal<number>;
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal: InputSignal<number>;
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal: ModelSignal<number>;
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal: WritableSignal<number>;
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = computed(() => 0);
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = contentChild('test');
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = contentChild.required('test');
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = contentChildren('test');
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = input('');
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = input.required('');
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = model(42);
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = model.required();
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = signal(42);
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = toSignal(source);
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = viewChild('test');
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = viewChild.required('test');
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = viewChildren('test');
  ~~~~~~~~~~
}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error",
      {
        "signalCreationFunctions": [
          "createSignal"
        ]
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
class Test {
  testSignal = createSignal('test');
  ~~~~~~~~~~
}
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  testValue = test();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  testValue: number;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  testValue: Widget<number>;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal: Signal<number>;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal: InputSignal<number>;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal: ModelSignal<number>;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal: WritableSignal<number>;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = computed(() => 0);
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = contentChild('test');
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = contentChild.required('test');
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = contentChildren('test');
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = input('');
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = input.required();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = model();
  readonly testRequired = model.required(42);
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = signal(true);
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = toSignal(source);
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = viewChild('test');
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = viewChild.required('test');
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = viewChildren('test');
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  testSignal = createSignal('test');
}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/prefer-signal-readonly": [
      "error",
      {
        "signalCreationFunctions": [
          "createSignal"
        ]
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {
  readonly testSignal = createSignal('test');
}
```

</details>

<br>
