<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-output-readonly.ts
  - ../../tests/rules/prefer-output-readonly/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/prefer-output-readonly`

Prefer to declare `@Output` as `readonly` since they are not supposed to be reassigned

- Type: suggestion
- Category: Best Practices

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

‚ùå - Examples of **incorrect** code for this rule:

```ts
class Test {
  @Output() testEmitter = new EventEmitter<string>();
            ~~~~~~~~~~~
}
```

<br>

---

<br>

‚úÖ - Examples of **correct** code for this rule:

```ts
class Test {
  testEmitter = new EventEmitter<string>();
}
```

```ts
class Test {
  @Output() readonly testEmitter = new EventEmitter<string>();
}
```
