<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-output-emitter-ref.ts
  - ../../tests/rules/prefer-output-emitter-ref/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/prefer-output-emitter-ref`

Use readonly `OutputEmitterRef` instead of `@Output()`

- Type: suggestion
- ğŸ”§ Supports autofix (`--fix`)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>âŒ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-emitter-ref": [
      "error"
    ]
  }
}
```

<br>

#### âŒ Invalid Code

```ts
class Test {
  @Output()
  ~~~~~~~~~
  readonly change: EventEmitter<number>;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-emitter-ref": [
      "error"
    ]
  }
}
```

<br>

#### âŒ Invalid Code

```ts
@Component()
class Test {
  change = output();
  ~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-emitter-ref": [
      "error"
    ]
  }
}
```

<br>

#### âŒ Invalid Code

```ts
@Directive()
class Test {
  change = output();
  ~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-emitter-ref": [
      "error"
    ]
  }
}
```

<br>

#### âŒ Invalid Code

```ts
@Component()
class Test {
  change = output<string>();
  ~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-emitter-ref": [
      "error"
    ]
  }
}
```

<br>

#### âŒ Invalid Code

```ts
@Directive()
class Test {
  change = output<string>();
  ~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-emitter-ref": [
      "error"
    ]
  }
}
```

<br>

#### âŒ Invalid Code

```ts
@Component()
class Test {
  change: OutputEmitterRef<string>;
  ~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-emitter-ref": [
      "error"
    ]
  }
}
```

<br>

#### âŒ Invalid Code

```ts
@Directive()
class Test {
  change: OutputEmitterRef<string>;
  ~~~~~~
}
```

</details>

<br>

---

<br>

<details>
<summary>âœ… - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-emitter-ref": [
      "error"
    ]
  }
}
```

<br>

#### âœ… Valid Code

```ts
@Component()
class Test {
  change = outputValue();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-emitter-ref": [
      "error"
    ]
  }
}
```

<br>

#### âœ… Valid Code

```ts
@Component()
class Test {
  change = new EventEmitter();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-emitter-ref": [
      "error"
    ]
  }
}
```

<br>

#### âœ… Valid Code

```ts
@Component()
class Test {
  readonly change = output();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-emitter-ref": [
      "error"
    ]
  }
}
```

<br>

#### âœ… Valid Code

```ts
@Directive()
class Test {
  readonly change = output();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-emitter-ref": [
      "error"
    ]
  }
}
```

<br>

#### âœ… Valid Code

```ts
@Component()
class Test {
  readonly change = output<string>();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-emitter-ref": [
      "error"
    ]
  }
}
```

<br>

#### âœ… Valid Code

```ts
@Directive()
class Test {
  readonly change = output<string>();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-output-emitter-ref": [
      "error"
    ]
  }
}
```

<br>

#### âœ… Valid Code

```ts
class Test {
  // This is allowed because it is not in a component or directive.
  change = output();
}
```

</details>

<br>
