<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-host-metadata-property.ts
  - ../../tests/rules/prefer-host-metadata-property/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/prefer-host-metadata-property`

Use `host` metadata property instead of `@HostBinding` and `HostListener`

- Type: suggestion

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-host-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test {
  @HostBinding('class')
  ~~~~~~~~~~~~~~~~~~~~~
  readonly cssClass = 'my-class';
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-host-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test {
  @HostListener('click')
  ~~~~~~~~~~~~~~~~~~~~~~
  bar() {}
}
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-host-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  host: {
    class: 'my-class',
  },
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/prefer-host-metadata-property": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({
  host: {
    '(click)': 'bar()'
  },
})
class Test {
  bar() {}
}
```

</details>

<br>
