<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/require-localize-metadata.ts
  - ../../tests/rules/require-localize-metadata/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/require-localize-metadata`

Ensures that $localize tagged messages contain helpful metadata to aid with translations.

- Type: suggestion

<br>

## Rationale

When internationalizing Angular applications with @angular/localize, adding metadata (description, meaning, and custom IDs) to $localize tagged strings is essential for high-quality translations. The description helps translators understand the context and purpose of the text they're translating. For example, "Submit" could mean "submit a form" or "submit to authority" - context matters for accurate translation. The meaning field distinguishes identical text that needs different translations in different contexts. Custom IDs provide stable references that persist across code changes, preventing retranslation when only code structure changes. Without this metadata, translators work blind, leading to poor translations, wasted effort on retranslating unchanged strings, and difficulty managing large translation projects. This rule ensures your $localize strings include helpful metadata for translation workflows.

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * Default: `false`
   */
  requireDescription?: boolean;
  /**
   * Default: `false`
   */
  requireMeaning?: boolean;
  /**
   * Default: `false`
   */
  requireCustomId?: boolean | string;
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireDescription": true
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const localizedText = $localize`Hello i18n!`;
                               ~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireDescription": true
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const localizedTexts = {
  helloI18n: $localize`:An introduction header for this sample:Hello i18n!`
};
localizedTexts.helloI18n = $localize`Hello i18n!`;
                                    ~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireDescription": true
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
return $localize`Hello i18n!`;
                ~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireDescription": true
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
someFunction($localize`Hello i18n!`);
                      ~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireDescription": true
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const localizedText = $localize`:site header|:Hello i18n!`;
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireDescription": true
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const localizedText = $localize`:@@custom_id:Hello i18n!`;
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireDescription": true
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const localizedText = $localize`:site header|@@custom_id:Hello i18n!`;
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireMeaning": true
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const localizedText = $localize`Hello i18n!`;
                               ~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireDescription": true,
        "requireMeaning": true
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const localizedText = $localize`:An introduction header for this sample:Hello i18n!`;
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireMeaning": true
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const localizedText = $localize`:|An introduction header for this sample:Hello i18n!`;
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireDescription": true,
        "requireMeaning": true
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const localizedText = $localize`:Hello i18n!`;
                               ~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireCustomId": true
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const localizedText = $localize`Hello i18n!`;
                               ~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireCustomId": true
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const localizedText = $localize`:@some.custom.id:Hello i18n!`;
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireCustomId": true
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const localizedText = $localize`:meaning|description:Hello i18n!`;
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireCustomId": "^some.wrong.pattern$"
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const localizedText = $localize`:@@some.custom.id:Hello i18n!`;
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const localizedText = $localize`Hello i18n!`;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const localizedText = $localize`:site header|:Hello i18n!`;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const localizedText = $localize`:@@custom_id:Hello i18n!`;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const localizedText = $localize`:site header|@@custom_id:Hello i18n!`;
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireDescription": true
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
let localizedText = $localize`:An introduction header for this sample:Hello i18n!`;
localizedText = $localize`:An introduction header for this sample modified:Hello i18n modified!`;
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireDescription": true
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const localizedTexts = {
  helloI18n: $localize`:An introduction header for this sample:Hello i18n!`
};
localizedTexts.helloI18n = $localize`:An introduction header for this sample modified:Hello i18n modified!`;
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireDescription": true
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
return $localize`:An introduction header for this sample:Hello i18n!`;
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireDescription": true
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
someFunction($localize`:An introduction header for this sample:Hello i18n!`);
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireDescription": true
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const localizedText = `Hello i18n!`;
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireMeaning": true
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const localizedText = $localize`:site header|:Hello i18n!`;
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireDescription": true,
        "requireMeaning": true
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const localizedText = $localize`:site header|An introduction header for this sample:Hello i18n!`;
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireCustomId": true
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const localizedText = $localize`:@@some.custom.id:Hello i18n!`;
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/require-localize-metadata": [
      "error",
      {
        "requireCustomId": "^some.*id$"
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const localizedText = $localize`:@@some.custom.id:Hello i18n!`;
```

</details>

<br>
