<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/require-lifecycle-on-prototype.ts
  - ../../tests/rules/require-lifecycle-on-prototype/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/require-lifecycle-on-prototype`

Ensures that lifecycle methods are defined on the object's prototype instead of on an instance.

- Type: problem

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({})
class Test {
  ngOnChanges = func;
  ~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({})
class Test {
  ngOnInit = func;
  ~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({})
class Test {
  ngDoCheck = func;
  ~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({})
class Test {
  ngAfterContentInit = func;
  ~~~~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({})
class Test {
  ngAfterContentChecked = func;
  ~~~~~~~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({})
class Test {
  ngAfterViewInit = func;
  ~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({})
class Test {
  ngAfterViewChecked = func;
  ~~~~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component({})
class Test {
  ngOnDestroy = func;
  ~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive({})
class Test {
  ngOnChanges = func;
  ~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive({})
class Test {
  ngOnInit = func;
  ~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive({})
class Test {
  ngDoCheck = func;
  ~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive({})
class Test {
  ngAfterContentInit = func;
  ~~~~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive({})
class Test {
  ngAfterContentChecked = func;
  ~~~~~~~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive({})
class Test {
  ngAfterViewInit = func;
  ~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive({})
class Test {
  ngAfterViewChecked = func;
  ~~~~~~~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Directive({})
class Test {
  ngOnDestroy = func;
  ~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Injectable({})
class Test {
  ngOnDestroy = func;
  ~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Pipe({})
class Test {
  ngOnDestroy = func;
  ~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test {
  constructor() {
    this.ngOnDestroy = func;
         ~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test {
  constructor() {
    this['ngOnDestroy'] = func;
         ~~~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test {
  run() {
    this.ngOnDestroy = func;
         ~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
function hook(component) {
  component.ngOnDestroy = func;
            ~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
function hook(component) {
  (component as any).ngOnDestroy = func;
                     ~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test {
  ['ngOnDestroy'] = func;
   ~~~~~~~~~~~~~
}
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({})
class Test {
  ngAfterContentChecked() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({})
class Test {
  ngAfterContentInit() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({})
class Test {
  ngAfterViewChecked() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({})
class Test {
  ngAfterViewInit() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({})
class Test {
  ngOnChanges() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({})
class Test {
  ngOnDestroy() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({})
class Test {
  ngOnInit() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({})
class Test {
  ngDoCheck() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({})
class Test {
  ngAfterContentChecked() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({})
class Test {
  ngAfterContentInit() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({})
class Test {
  ngAfterViewChecked() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({})
class Test {
  ngAfterViewInit() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({})
class Test {
  ngOnChanges() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({})
class Test {
  ngOnDestroy() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({})
class Test {
  ngOnInit() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Directive({})
class Test {
  ngDoCheck() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Injectable({})
class Test {
  ngOnDestroy() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Pipe({})
class Test {
  ngOnDestroy() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({})
class Test {}
function hook(type) {
  type.prototype.ngOnDestroy = () => {};
}
hook(Test);
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({})
class Test {}
function hook(type) {
  (type.prototype as any).ngOnDestroy = () => {};
}
hook(Test);
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({})
class Test {}
function hook(type) {
  type['prototype'].ngOnDestroy = () => {};
}
hook(Test);
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({})
class Test {}
function hook(type) {
  (type.prototype as unknown as { ngOnDestroy: () => void}).ngOnDestroy = () => {};
}
hook(Test);
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({})
class Test {
  onDestroy = () => {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({})
class Test {
  constructor() {
    this.onDestroy = () => {}
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({})
class Test {
  constructor() {
    this['onDestroy'] = () => {}
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/require-lifecycle-on-prototype": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component({})
class Test {
  constructor() {
    let ngOnDestroy;
    ngOnDestroy = () => {};
  }
}
```

</details>

<br>
