<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/runtime-localize.ts
  - ../../tests/rules/runtime-localize/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/runtime-localize`

Ensures that $localize tagged messages can use runtime-loaded translations.

- Type: suggestion

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/runtime-localize": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
const foo = $localize`foo`;
            ~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/runtime-localize": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
let foo = {
  x: [{
    y: `--${$localize`foo`}--`
            ~~~~~~~~~
  }]
};
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/runtime-localize": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
(x => { return x; })($localize`foo`);
                     ~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/runtime-localize": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test {
  static foo = $localize`foo`;
               ~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/runtime-localize": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
class Test {
  static foo = {
    x: [{
      y: `--${$localize`foo`}--`
              ~~~~~~~~~
    }]
  }
}
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/runtime-localize": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
function test() { return $localize`foo`; }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/runtime-localize": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const test = function () { return $localize`foo`; }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/runtime-localize": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
const test = () => $localize`foo`;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/runtime-localize": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
function test(foo = $localize`foo`) { return foo; }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/runtime-localize": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test {
  foo() { $localize`foo`; }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/runtime-localize": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test {
  foo = () => $localize`foo`;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/runtime-localize": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test {
  static foo = () => $localize`foo`;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/runtime-localize": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test {
  readonly foo = $localize`foo`;
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/runtime-localize": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
class Test {
  readonly foo = { x: [`--${$localize`foo`}--`] };
}
```

</details>

<br>
