<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/sort-ngmodule-metadata-arrays.ts
  - ../../tests/rules/sort-ngmodule-metadata-arrays/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/sort-ngmodule-metadata-arrays`

Ensures ASC alphabetical order for `NgModule` metadata arrays for easy visual scanning

- Type: suggestion
- Category: Best Practices
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

‚ùå - Examples of **incorrect** code for this rule:

```ts
@NgModule({
  imports: [aModule, bModule, DModule, cModule]
                                       ~~~~~~~
})
class Test {}
```

```ts
@NgModule({
  'declarations': [
    AComponent,
    cPipe,
    bDirective,
    ~~~~~~~~~~
    DComponent,
  ],
})
class Test {}
```

```ts
@NgModule({
  ['exports']: [
    AComponent,
    cPipe,
    bDirective,
    ~~~~~~~~~~
    DComponent,
  ],
})
class Test {}
```

```ts
@NgModule({
  [`bootstrap`]: [
    AppModule2,
    AppModule3,
    AppModule1,
    ~~~~~~~~~~
  ]
})
class Test {}
```

```ts
@NgModule({
  schemas: [
    A_SCHEMA,
    C_SCHEMA,
    B_SCHEMA,
    ~~~~~~~~
  ]
})
class Test {}
```

```ts
@NgModule({
  providers: [
    AProvider,
    {
      provide: 'myprovider',
      useClass: MyProvider,
    },
    cProvider,
    bProvider,
    ~~~~~~~~~
    DProvider,
  ]
})
class Test {}
```

```ts
@NgModule({
  bootstrap,
  declarations: declarations,
  providers: providers(),
  schemas: [],
  [imports]: [
    aModule,
    bModule,
    DModule,
    cModule,
    ~~~~~~~
  ],
})
class Test {}
```

<br>

---

<br>

‚úÖ - Examples of **correct** code for this rule:

```ts
class Test {}
```

```ts
@NgModule()
class Test {}
```

```ts
@NgModule({})
class Test {}
```

```ts
const options = {};
@NgModule(options)
class Test {}
```

```ts
@NgModule({
  bootstrap: [
    AppModule1,
    AppModule2,
    AppModule3,
  ],
  'declarations': [
    AComponent,
    bDirective,
    cPipe,
    DComponent,
    VariableComponent,
  ],
  ['imports']: [
    _foo,
    AModule,
    bModule,
    cModule,
    DModule,
  ],
  [`providers`]: [
    AProvider,
    {
      provide: 'myprovider',
      useClass: MyProvider,
    },
    bProvider,
    cProvider,
    DProvider,
  ],
})
class Test {}
```

```ts
@Component({
  providers: [
    DeclarationD,
    DeclarationA,
  ]
})
class Test {}
```
