<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/sort-keys-in-type-decorator.ts
  - ../../tests/rules/sort-keys-in-type-decorator/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/sort-keys-in-type-decorator`

Ensures that keys in type decorators (Component, Directive, NgModule, Pipe) are sorted in a consistent order

- Type: suggestion
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * Default: `["selector","imports","standalone","templateUrl","styleUrl","encapsulation","changeDetection"]`
   */
  Component?: string[];
  /**
   * Default: `["selector","standalone"]`
   */
  Directive?: string[];
  /**
   * Default: `["declarations","imports","exports","providers","bootstrap"]`
   */
  NgModule?: string[];
  /**
   * Default: `["name","standalone"]`
   */
  Pipe?: string[];
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/sort-keys-in-type-decorator": [
      "error",
      {
        "Component": [
          "selector",
          "imports",
          "standalone",
          "templateUrl",
          "styleUrl",
          "encapsulation",
          "changeDetection"
        ]
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  selector: 'app-root',
  imports: [CommonModule],
  standalone: true,
  templateUrl: './app.component.html',
  styleUrl: './app.component.css',
  encapsulation: ViewEncapsulation.None
})
class Test {
}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/sort-keys-in-type-decorator": [
      "error",
      {
        "Directive": [
          "selector",
          "standalone"
        ]
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Directive({
  standalone: true,
  ~~~~~~~~~~~~~~~~
  selector: '[app-test]'
})
class Test {
}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/sort-keys-in-type-decorator": [
      "error",
      {
        "NgModule": [
          "declarations",
          "imports"
        ]
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@NgModule({
  imports: [CommonModule],
  ~~~~~~~~~~~~~~~~~~~~~~~
  declarations: [AppComponent]
})
class Test {
}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/sort-keys-in-type-decorator": [
      "error",
      {
        "Pipe": [
          "name",
          "standalone"
        ]
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Pipe({
  standalone: true,
  ~~~~~~~~~~~~~~~~
  name: 'myPipe'
})
class Test {
}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/sort-keys-in-type-decorator": [
      "error",
      {
        "Component": [
          "selector",
          "imports",
          "standalone",
          "templateUrl",
          "styleUrl",
          "encapsulation",
          "changeDetection"
        ]
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  styleUrl: './app.component.css',
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  selector: 'app-root',
  templateUrl: './app.component.html'
})
class Test {
}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/sort-keys-in-type-decorator": [
      "error",
      {
        "NgModule": [
          "declarations",
          "imports",
          "exports",
          "providers",
          "bootstrap"
        ]
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@NgModule({
  exports: [AppComponent],
  ~~~~~~~~~~~~~~~~~~~~~~~
  declarations: [AppComponent]
})
class Test {
}
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-keys-in-type-decorator": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Type({
  a: 'a',
  b: 'b',
  c: 'c'
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-keys-in-type-decorator": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Type({})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-keys-in-type-decorator": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Type({
  a: 'a'
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/sort-keys-in-type-decorator": [
      "error",
      {
        "Component": [
          "selector",
          "imports",
          "standalone",
          "templateUrl",
          "styleUrl",
          "encapsulation",
          "changeDetection"
        ]
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  selector: 'app-root',
  imports: [CommonModule],
  standalone: true,
  templateUrl: './app.component.html',
  styleUrl: './app.component.css',
  encapsulation: ViewEncapsulation.None,
  changeDetection: ChangeDetectionStrategy.OnPush
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/sort-keys-in-type-decorator": [
      "error",
      {
        "Directive": [
          "selector",
          "standalone"
        ]
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Directive({
  selector: '[app-test]',
  standalone: true
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/sort-keys-in-type-decorator": [
      "error",
      {
        "NgModule": [
          "declarations",
          "imports"
        ]
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@NgModule({
  declarations: [AppComponent],
  imports: [CommonModule]
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/sort-keys-in-type-decorator": [
      "error",
      {
        "Pipe": [
          "name",
          "standalone"
        ]
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Pipe({
  name: 'myPipe',
  standalone: true
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/sort-keys-in-type-decorator": [
      "error",
      {
        "Component": [
          "selector",
          "imports",
          "standalone",
          "templateUrl",
          "styleUrl",
          "encapsulation",
          "changeDetection"
        ]
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrl: './app.component.css'
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/sort-keys-in-type-decorator": [
      "error",
      {
        "NgModule": [
          "declarations",
          "imports",
          "exports",
          "providers",
          "bootstrap"
        ]
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@NgModule({
  declarations: [AppComponent],
  exports: [AppComponent]
})
class Test {}
```

</details>

<br>
