<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/sort-ngmodule-metadata-arrays.ts
  - ../../tests/rules/sort-ngmodule-metadata-arrays/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/sort-ngmodule-metadata-arrays`

## ‚ö†Ô∏è THIS RULE IS DEPRECATED

---

Ensures ASC alphabetical order for `NgModule` metadata arrays for easy visual scanning

- Type: suggestion
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * A string with a BCP 47 language tag.
   *
   * Default: `"en-US"`
   */
  locale?: string;
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-ngmodule-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@NgModule({
  imports: [aModule, bModule, DModule, cModule]
                                       ~~~~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-ngmodule-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@NgModule({
  'declarations': [
    AComponent,
    cPipe,
    bDirective,
    ~~~~~~~~~~
    DComponent,
  ],
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-ngmodule-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@NgModule({
  ['exports']: [
    AComponent,
    cPipe,
    bDirective,
    ~~~~~~~~~~
    DComponent,
  ],
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-ngmodule-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@NgModule({
  [`bootstrap`]: [
    AppModule2,
    AppModule3,
    AppModule1,
    ~~~~~~~~~~
  ]
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-ngmodule-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@NgModule({
  schemas: [
    A_SCHEMA,
    C_SCHEMA,
    B_SCHEMA,
    ~~~~~~~~
  ]
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-ngmodule-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@NgModule({
  providers: [
    AProvider,
    {
      provide: 'myprovider',
      useClass: MyProvider,
    },
    cProvider,
    bProvider,
    ~~~~~~~~~
    DProvider,
  ]
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-ngmodule-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@NgModule({
  bootstrap,
  declarations: declarations,
  providers: providers(),
  schemas: [],
  [imports]: [
    aModule,
    bModule,
    DModule,
    cModule,
    ~~~~~~~
  ],
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/sort-ngmodule-metadata-arrays": [
      "error",
      {
        "locale": "cs-CZ"
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@NgModule({
  imports: [chModule, dModule]
                      ~~~~~~~
})
class Test {}
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-ngmodule-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-ngmodule-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@NgModule()
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-ngmodule-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@NgModule({})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-ngmodule-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
const options = {};
@NgModule(options)
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-ngmodule-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@NgModule({
  bootstrap: [
    AppModule1,
    AppModule2,
    AppModule3,
  ],
  'declarations': [
    AComponent,
    bDirective,
    cPipe,
    DComponent,
    VariableComponent,
  ],
  ['imports']: [
    _foo,
    AModule,
    bModule,
    cModule,
    DModule,
  ],
  [`providers`]: [
    AProvider,
    {
      provide: 'myprovider',
      useClass: MyProvider,
    },
    bProvider,
    cProvider,
    DProvider,
  ],
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-ngmodule-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  providers: [
    DeclarationD,
    DeclarationA,
  ]
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-ngmodule-metadata-arrays": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@NgModule({
  providers: [
    {
      provide: 'myprovider',
      useFactory: myProviderFactory,
      deps: [TOKEN_Z, ClassX, ClassA, TOKEN_A],
    },
  ],
})
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/sort-ngmodule-metadata-arrays": [
      "error",
      {
        "locale": "cs-CZ"
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  providers: [
    DatepickerProvider,
    ChipsProvider,
  ]
})
class Test {}
```

</details>

<br>
