<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/sort-signals.ts
  - ../../tests/rules/sort-signals/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/sort-signals`

Ensures that signals are declared in order of execution

- Type: problem

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * The order of the signal types
   *
   * Default: `["input","model","output","viewChild","viewChildren","contentChild","contentChildren","inject","signal","computed"]`
   */
  order?: string[];
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-signals": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  readonly outputMethod = output();
  readonly inputProp = input();
           ~~~~~~~~~
  readonly prop = signal();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-signals": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  readonly outputMethod = output();
  readonly inputProp = input.required();
           ~~~~~~~~~
  readonly prop = signal();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-signals": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  readonly inputProp = input();
  readonly outputMethod = output();
  readonly modelProp = model();
           ~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-signals": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  readonly inputProp = input();
  readonly modelProp = model();
  readonly prop = signal();
  readonly outputMethod = output();
           ~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-signals": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  readonly calculated = computed(() => prop + 1);
  readonly prop = signal();
           ~~~~
}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/sort-signals": [
      "error",
      {
        "order": [
          "output",
          "input"
        ]
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  readonly inputProp = input();
  readonly outputMethod = output();
           ~~~~~~~~~~~~
}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/sort-signals": [
      "error",
      {
        "order": [
          "output",
          "input"
        ]
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```ts
@Component()
class Test {
  readonly prop = signal();
  readonly outputMethod = output();
           ~~~~~~~~~~~~
  readonly inputProp = input();
}
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-signals": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  readonly inputProp = input();
  readonly outputMethod = output();
  readonly prop = signal();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-signals": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  readonly inputProp = input.required();
  readonly outputMethod = output();
  readonly prop = signal();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-signals": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  readonly inputProp = input();
  readonly outputMethod = output.required();
  readonly prop = signal();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-signals": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  readonly inputProp = input.required();
  readonly outputMethod = output.required();
  readonly prop = signal();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-signals": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  readonly inputProp = input();
  readonly modelProp = model();
  readonly outputMethod = output();
  readonly element = viewChild();
  readonly content = contentChild();
  readonly service = inject(Service);
  readonly prop = signal();
  readonly calculated = computed(() => prop + 1);
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-signals": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  readonly prop = signal();
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-signals": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/sort-signals": [
      "error",
      {
        "order": [
          "output",
          "input"
        ]
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```ts
@Component()
class Test {
  readonly outputMethod = output.required();
  readonly inputProp = input.required();
  readonly service = inject(Service);
}
```

</details>

<br>
