<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/use-injectable-provided-in.ts
  - ../../tests/rules/use-injectable-provided-in/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/use-injectable-provided-in`

Using the `providedIn` property makes `Injectables` tree-shakable

- Type: suggestion

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rationale

Using 'providedIn' in the @Injectable() decorator (like '@Injectable({ providedIn: "root" })') enables tree-shaking, which means Angular can remove the service from your production bundle if it's never actually used. Without 'providedIn', services must be registered in a module's providers array, and Angular must include them in the bundle even if they're unused. This can significantly increase bundle size. Additionally, 'providedIn' makes services easier to use (no need to add them to module providers) and makes circular dependencies less likely. The 'providedIn' syntax is the modern, recommended way to provide services.

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  ignoreClassNamePattern?: string;
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-injectable-provided-in": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Injectable()
~~~~~~~~~~~~~
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-injectable-provided-in": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Injectable({})
~~~~~~~~~~~~~~~
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-injectable-provided-in": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
const providedIn = 'anotherProperty';
@Injectable({ [providedIn]: [] })
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-injectable-provided-in": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Injectable({ providedIn: null })
                          ~~~~
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-injectable-provided-in": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Injectable({ ['providedIn']: undefined })
                              ~~~~~~~~~
class Test {}

@Injectable()
class HttpPostInterceptor implements HttpInterceptor {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/use-injectable-provided-in": [
      "error",
      {
        "ignoreClassNamePattern": "/(Effects|NgModule)$/"
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Injectable({ ['providedIn']: undefined })
                              ~~~~~~~~~
class Test {}

@Injectable()
class ProvidedInNgModule {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/use-injectable-provided-in": [
      "error",
      {
        "ignoreClassNamePattern": "/(Effects|NgModule)$/"
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Injectable({ [`providedIn`]: undefined })
                              ~~~~~~~~~
class Test {}

@Injectable()
class TestEffects {}
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-injectable-provided-in": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-injectable-provided-in": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
const options = {};
@Injectable(options)
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-injectable-provided-in": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Injectable({
  providedIn: `any`,
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-injectable-provided-in": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Injectable({
  'providedIn': 'root',
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-injectable-provided-in": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Injectable({
  ['providedIn']: SomeModule,
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-injectable-provided-in": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Injectable({
  [`providedIn`]: providedIn(),
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-injectable-provided-in": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Injectable()
class Test implements HttpInterceptor {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-injectable-provided-in": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Injectable()
class Test implements ng.HttpInterceptor {}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/use-injectable-provided-in": [
      "error",
      {
        "ignoreClassNamePattern": "/Effects$/"
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Injectable()
class TestEffects {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-injectable-provided-in": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@CustomInjectable()
class Test {}
```

</details>

<br>
