<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/use-pipe-transform-interface.ts
  - ../../tests/rules/use-pipe-transform-interface/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/use-pipe-transform-interface`

Ensures that `Pipes` implement `PipeTransform` interface

- Type: suggestion
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-pipe-transform-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Pipe({ name: 'test' })
class Test {
      ~~~~
  transform(value: string) {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-pipe-transform-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
import { HttpClient } from '@angular/common/http';
import type { PipeTransform } from '@angular/core';
import { Component,
  Pipe,
  Directive } from '@angular/core';

@Pipe({ name: 'test' })
class Test implements AnInterface {
      ~~~~
  transform(value: string) {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-pipe-transform-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
import type { OnInit } from '@angular/core';

@OtherDecorator() @Pipe({ name: 'test' })
class Test implements AnInterface, AnotherInterface {
      ~~~~
  transform(value: string) {}
}
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-pipe-transform-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({ template: 'test' })
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-pipe-transform-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Pipe({ name: 'test' })
class Test implements PipeTransform {
  transform(value: string) {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-pipe-transform-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@OtherDecorator() @Pipe({ name: 'test' })
class Test implements PipeTransform {
  transform(value: string) {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-pipe-transform-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Pipe({ name: 'test' })
class Test implements ng.PipeTransform {
  transform(value: string) {}
}
```

</details>

<br>
