<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/use-injectable-provided-in.ts
  - ../../tests/rules/use-injectable-provided-in/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/use-injectable-provided-in`

Using the `providedIn` property makes `Injectables` tree-shakable

- Type: suggestion
- Category: Best Practices

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  ignoreClassNamePattern?: string;
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

‚ùå - Examples of **incorrect** code for this rule:

```ts
@Injectable()
~~~~~~~~~~~~~
class Test {}
```

```ts
@Injectable({})
~~~~~~~~~~~~~~~
class Test {}
```

```ts
const providedIn = 'anotherProperty';
@Injectable({ [providedIn]: [] })
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
class Test {}
```

```ts
@Injectable({ providedIn: null })
                          ~~~~
class Test {}
```

```ts
@Injectable({ ['providedIn']: undefined })
                              ~~~~~~~~~
class Test {}

@Injectable()
class HttpPostInterceptor implements HttpInterceptor {}
```

```ts
@Injectable({ ['providedIn']: undefined })
                              ~~~~~~~~~
class Test {}

@Injectable()
class ProvidedInNgModule {}
```

```ts
@Injectable({ [`providedIn`]: undefined })
                              ~~~~~~~~~
class Test {}

@Injectable()
class TestEffects {}
```

<br>

---

<br>

‚úÖ - Examples of **correct** code for this rule:

```ts
class Test {}
```

```ts
const options = {};
@Injectable(options)
class Test {}
```

```ts
@Injectable({
  providedIn: `any`,
})
class Test {}
```

```ts
@Injectable({
  'providedIn': 'root',
})
class Test {}
```

```ts
@Injectable({
  ['providedIn']: SomeModule,
})
class Test {}
```

```ts
@Injectable({
  [`providedIn`]: providedIn(),
})
class Test {}
```

```ts
@Injectable()
class Test implements HttpInterceptor {}
```

```ts
@Injectable()
class Test implements ng.HttpInterceptor {}
```

```ts
@CustomInjectable()
class Test {}
```
