<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/use-component-view-encapsulation.ts
  - ../../tests/rules/use-component-view-encapsulation/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/use-component-view-encapsulation`

Disallows using `ViewEncapsulation.None`

- Type: suggestion

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rationale

Setting encapsulation to ViewEncapsulation.None disables Angular's view encapsulation, making the component's styles global and affecting the entire application. This breaks component isolation and can cause unexpected styling conflicts throughout your app. When multiple components use ViewEncapsulation.None, their styles can interfere with each other in unpredictable ways, making it extremely difficult to reason about styling. A style intended for one component might accidentally affect completely unrelated components. This defeats one of Angular's key benefits: component style isolation. ViewEncapsulation.None is occasionally necessary for specific edge cases (like styling dynamically inserted content), but it should be avoided in normal component development. Use the default Emulated encapsulation, or ShadowDom for native Shadow DOM, to keep component styles properly isolated.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-component-view-encapsulation": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  encapsulation: ViewEncapsulation.None,
                                   ~~~~
  selector: 'app-foo-bar',
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-component-view-encapsulation": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
import type { ViewEncapsulation } from '@angular/core';
import { HttpClient } from '@angular/common/http';

@Component({
  selector: 'app-foo-bar',
  'encapsulation': ViewEncapsulation.None
                                     ~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-component-view-encapsulation": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
import { ViewEncapsulation } from '@angular/core';
import { HttpClient } from '@angular/common/http';

@Component({
  selector: 'app-foo-bar',
  ['encapsulation']: ViewEncapsulation.None
                                       ~~~~
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-component-view-encapsulation": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
import { ViewEncapsulation } from '@angular/core';
import { HttpClient } from '@angular/common/http';

@Component({
  selector: 'app-foo-bar',
  [`encapsulation`]: ViewEncapsulation.None
                                       ~~~~
})
class Test {}
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-component-view-encapsulation": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  encapsulation: ViewEncapsulation.Emulated,
  selector: 'app-foo-bar'
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-component-view-encapsulation": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  'encapsulation': ViewEncapsulation.Native,
  selector: 'app-foo-bar',
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-component-view-encapsulation": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  ['encapsulation']: ViewEncapsulation.ShadowDom,
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-component-view-encapsulation": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
function encapsulation() {
  return ViewEncapsulation.None;
}

@Component({
  [`encapsulation`]: encapsulation()
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-component-view-encapsulation": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
const encapsulation = 'templateUrl';
@Component({
  [encapsulation]: '../a.html'
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-component-view-encapsulation": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
const encapsulation = 'templateUrl';
@Component({
  encapsulation
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-component-view-encapsulation": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
const test = 'test';
@Component({
  encapsulation: test,
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-component-view-encapsulation": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  encapsulation: undefined,
})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-component-view-encapsulation": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({})
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-component-view-encapsulation": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
const options = {};
@Component(options)
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-component-view-encapsulation": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@NgModule({
  bootstrap: [Foo]
})
class Test {}
```

</details>

<br>
