<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/use-injectable-provided-in.ts
  - ../../tests/rules/use-injectable-provided-in/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/use-injectable-provided-in`

"Using the 'providedIn' property makes classes decorated with @Injectable tree shakeable

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```ts
@Injectable()
~~~~~~~~~~~~~
class Test {}
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```ts
@Injectable({
  providedIn: 'root'
})
class Test {}
```

```ts
@Injectable({
  providedIn: SomeModule
})
class Test {}
```
