<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/use-component-view-encapsulation.ts
  - ../../tests/rules/use-component-view-encapsulation/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/use-component-view-encapsulation`

Disallows using `ViewEncapsulation.None`

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```ts
@Component({
  encapsulation: ViewEncapsulation.None,
                                   ~~~~
  selector: 'app-foo-bar',
})
class Test {}
```

```ts
import type { ViewEncapsulation } from '@angular/core';
import { HttpClient } from '@angular/common/http';

@Component({
  selector: 'app-foo-bar',
  'encapsulation': ViewEncapsulation.None
                                     ~~~~
})
class Test {}
```

```ts
import { ViewEncapsulation } from '@angular/core';
import { HttpClient } from '@angular/common/http';

@Component({
  selector: 'app-foo-bar',
  ['encapsulation']: ViewEncapsulation.None
                                       ~~~~
})
class Test {}
```

```ts
import { ViewEncapsulation } from '@angular/core';
import { HttpClient } from '@angular/common/http';

@Component({
  selector: 'app-foo-bar',
  [`encapsulation`]: ViewEncapsulation.None
                                       ~~~~
})
class Test {}
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```ts
@Component({
  encapsulation: ViewEncapsulation.Emulated,
  selector: 'app-foo-bar'
})
class Test {}
```

```ts
@Component({
  'encapsulation': ViewEncapsulation.Native,
  selector: 'app-foo-bar',
})
class Test {}
```

```ts
@Component({
  ['encapsulation']: ViewEncapsulation.ShadowDom,
})
class Test {}
```

```ts
function encapsulation() {
  return ViewEncapsulation.None;
}

@Component({
  [`encapsulation`]: encapsulation()
})
class Test {}
```

```ts
const encapsulation = 'templateUrl';
@Component({
  [encapsulation]: '../a.html'
})
class Test {}
```

```ts
const encapsulation = 'templateUrl';
@Component({
  encapsulation
})
class Test {}
```

```ts
const test = 'test';
@Component({
  encapsulation: test,
})
class Test {}
```

```ts
@Component({
  encapsulation: undefined,
})
class Test {}
```

```ts
@Component({})
class Test {}
```

```ts
const options = {};
@Component(options)
class Test {}
```

```ts
@NgModule({
  bootstrap: [Foo]
})
class Test {}
```
