<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/use-component-selector.ts
  - ../../tests/rules/use-component-selector/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/use-component-selector`

Component selector must be declared

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```ts
@Component()
~~~~~~~~~~~~
class Test {}
```

```ts
@Component()
~~~~~~~~~~~~
class Test {}
```

```ts
@Component({
~~~~~~~~~~~~
  selector: ''
})
~~
class Test {}
```

```ts
@Component({
~~~~~~~~~~~~
  selector: 0
})
~~
class Test {}
```

```ts
@Component({
~~~~~~~~~~~~
  selector: null
})
~~
class Test {}
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```ts
@Component({
  selector: 'sg-bar-foo'
})
class Test {}
```
