<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/use-lifecycle-interface.ts
  - ../../tests/rules/use-lifecycle-interface/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/use-lifecycle-interface`

Ensures that classes implement lifecycle interfaces corresponding to the declared lifecycle methods. See more at https://angular.dev/style-guide#use-lifecycle-hook-interfaces

- Type: suggestion
- üîß Supports autofix (`--fix`)

<br>

## Rationale

Implementing lifecycle interfaces (like OnInit, OnDestroy, AfterViewInit) provides TypeScript compile-time checking to ensure you've spelled the lifecycle method names correctly and used the right method signatures. For example, if you typo 'ngOnInit' as 'ngOninit', TypeScript will catch this error if your class implements OnInit. The interfaces also serve as self-documentation, making it immediately clear which lifecycle hooks a component uses. While Angular will still call correctly-named lifecycle methods even without the interface, using the interface leverages TypeScript's type safety to catch errors earlier in development.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-lifecycle-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component()
class Test {
  ngOnInit() {
  ~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-lifecycle-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
import { OnInit } from '@angular/core';

        @Directive()
        class Test extends Component implements OnInit {
          ngOnInit() {}

          ngOnDestroy() {
          ~~~~~~~~~~~
          }
        }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-lifecycle-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Injectable()
class Test {
  ngDoBootstrap() {}
  ~~~~~~~~~~~~~

  ngOnInit() {}
  ~~~~~~~~

  ngOnDestroy() {}
  ~~~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-lifecycle-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@NgModule()
class Test extends Component implements ng.OnInit {
  ngOnInit() {}

  ngOnDestroy() {
  ~~~~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-lifecycle-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@NgModule()
class Test extends Component {
  ngOnInit() {
  ~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-lifecycle-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Directive()
class FoobarBase implements OnDestroy {
  ngOnDestroy(): void {
    /* some base logic here */
  }
}

@Component()
class FoobarComponent extends FoobarBase {
  ngOnDestroy(): void {
  ~~~~~~~~~~~
    super.ngOnDestroy();
    /* some concrete logic here */
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-lifecycle-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component()
class Test<T> {
  ngOnInit() {
  ~~~~~~~~
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-lifecycle-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Injectable()
class Test<A, B extends C = D> {
  ngOnInit() {
  ~~~~~~~~
  }
}
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-lifecycle-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test implements OnInit {
  ngOnInit() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-lifecycle-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test implements DoBootstrap {
      ngDoBootstrap() {}
    }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-lifecycle-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test extends Component implements OnInit, OnDestroy  {
  ngOnInit() {}

  private ngOnChanges = '';

  ngOnDestroy() {}

  ngOnSmth() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-lifecycle-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test extends Component implements ng.OnInit, ng.OnDestroy  {
  ngOnInit() {}

  private ngOnChanges = '';

  ngOnDestroy() {}

  ngOnSmth() {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-lifecycle-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class Test {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-lifecycle-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Directive()
class FoobarBase implements OnDestroy {
  ngOnDestroy(): void {
    /* some base logic here */
  }
}

@Component()
class FoobarComponent extends FoobarBase {
  override ngOnDestroy(): void {
    super.ngOnDestroy();
    /* some concrete logic here */
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-lifecycle-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
class BaseClass {
  ngOnInit(): void {
    /* base initialization */
  }
}

@Component()
class DerivedComponent extends BaseClass {
  override ngOnInit(): void {
    super.ngOnInit();
    /* derived initialization */
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/use-lifecycle-interface": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Directive()
class BaseDirective implements OnInit {
  ngOnInit(): void {
    /* base initialization */
  }
}

@Component()
class DerivedComponent extends BaseDirective {
  override ngOnInit(): void {
    super.ngOnInit();
    /* derived initialization */
  }
}
```

</details>

<br>
