<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/accessibility-label-for.ts
  - ../../tests/rules/accessibility-label-for/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/accessibility-label-for`

## ⚠️ THIS RULE IS DEPRECATED

Please use `@angular-eslint/template/accessibility-label-has-associated-control` instead.

---

Ensures that a label element/component is associated with a form element

- Type: suggestion

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * Default: `["button","input","meter","output","progress","select","textarea"]`
   */
  controlComponents?: string[];
  /**
   * Default: `["for","htmlFor"]`
   */
  labelAttributes?: string[];
  /**
   * Default: `["label"]`
   */
  labelComponents?: string[];
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-label-for": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<label>Label</label>
~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-label-for": [
      "error",
      {
        "labelAttributes": [
          "id"
        ],
        "labelComponents": [
          "app-label"
        ]
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<app-label anotherAttribute="id"></app-label>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-label-for": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<ng-container *ngFor="let item of items; index as index">
  <label for="item-{{index}}">Label #{{index}</label>
  <input id="item-{{index}}" [(ngModel)]="item.name">
</ng-container>
<label for="id"></label>
<label for="{{id}}"></label>
<label [attr.for]="id"></label>
<label [htmlFor]="id"></label>
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-label-for": [
      "error",
      {
        "labelAttributes": [
          "id"
        ],
        "labelComponents": [
          "app-label"
        ]
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<app-label id="name"></app-label>
<app-label id="{{name}}"></app-label>
<app-label [id]="name"></app-label>
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-label-for": [
      "error",
      {
        "controlComponents": [
          "app-input"
        ],
        "labelComponents": [
          "app-label"
        ]
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<label><button>Button</button></label>
<label><input type="radio"></label>
<label><meter></meter></label>
<label><output></output></label>
<label><progress></progress></label>
<label><select><option>1</option></select></label>
<label><textarea></textarea></label>
<a-label><input></a-label>
<label>
  Label
  <input>
</label>
<label>
  Label
  <span><input></span>
</label>
<app-label>
  <span>
    <app-input></app-input>
  </span>
</app-label>
```

</details>

<br>
