<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/accessibility-valid-aria.ts
  - ../../tests/rules/accessibility-valid-aria/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/template/accessibility-valid-aria`

Ensures that correct ARIA attributes and respective values are used

- Type: suggestion
- Category: Best Practices

- 💡 Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```html
<div aria-roledescriptio="text">Text</div>
     ~~~~~~~~~~~~~~~~~~~~~~~~~~
<input [aria-labelby]="label">
       ^^^^^^^^^^^^^^^^^^^^^^
<input [attr.aria-requiredIf]="required">
       #################################
```

```html
<div aria-expanded="notABoolean">notABoolean</div>
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
<div aria-haspopup="notAToken">notAToken</div>
     ^^^^^^^^^^^^^^^^^^^^^^^^^
<input [attr.aria-rowcount]="{ a: 2 }">notAnInteger
       ###############################
<div aria-relevant="notATokenList">notATokenList</div>
     %%%%%%%%%%%%%%%%%%%%%%%%%%%%%
<div aria-checked="notATristate">notATristate</div>
     ¶¶¶¶¶¶¶¶¶¶¶¶¶¶¶¶¶¶¶¶¶¶¶¶¶¶¶
<div role="slider" [attr.aria-valuemin]="[1, 2]">notANumber</div>
                   ¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨
<input [attr.aria-placeholder]="4">notAPlaceholder
       @@@@@@@@@@@@@@@@@@@@@@@@@@@
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```html
<input aria-labelledby="Text">
```

```html
<div ariaselected="0"></div>
```

```html
<textarea [attr.aria-readonly]="readonly"></textarea>
```

```html
<button [variant]="variant">Text</button>
```

```html
<div aria-expanded="true">aria-expanded</div>
```

```html
<div aria-haspopup="menu">aria-haspopup</div>
```

```html
<div [attr.aria-pressed]="undefined">aria-pressed</div>
```

```html
<input [attr.aria-rowcount]="2">
```

```html
<div aria-relevant="additions">additions</div>
```

```html
<div aria-checked="false">checked</div>
```

```html
<div role="slider" [aria-valuemin]="1"></div>
```

```html
<div aria-="text">Text</div>
```

```html
<input
  aria-placeholder="Placeholder"
  aria-orientation="undefined"
  [attr.aria-checked]="test && isChecked"
  [attr.aria-hidden]="'abc' | appAria"
  [attr.aria-invalid]="hasError ? 'grammar' : 'spelling'"
  [attr.aria-label]="inputSchema!.label"
  [attr.aria-live]="inputSchema['live']"
  [attr.aria-required]="inputSchema?.isRequired">
```

```html
<app-custom aria-x="text">Text</app-custom>
```

```html
<app-test aria-expanded="notABoolean"></app-test>
```
