<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/accessibility-role-has-required-aria.ts
  - ../../tests/rules/accessibility-role-has-required-aria/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/accessibility-role-has-required-aria`

Ensures elements with ARIA roles have all required properties for that role.

- Type: suggestion

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-role-has-required-aria": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div role="combobox"></div>
     ~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-role-has-required-aria": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<span role="checkbox" aria-checked="false"></span>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-role-has-required-aria": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<input type="checkbox" role="switch">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-role-has-required-aria": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<span role="heading" aria-level="5"></span>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-role-has-required-aria": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<span role="button"></span>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-role-has-required-aria": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<app-component [role]="ADMIN"></app-component>
```

</details>

<br>
