<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/accessibility-alt-text.ts
  - ../../tests/rules/accessibility-alt-text/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/template/accessibility-alt-text`

Enforces alternate text for elements which require the alt, aria-label, aria-labelledby attributes.

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```html
<ng-template>
  <div>
    <img src="foo">
    ~~~~~~~~~~~~~~~
  </div>
</ng-template>
```

```html
<object></object>
~~~~~~~~~~~~~~~~~
```

```html
<area />
~~~~~~~~
```

```html
<input type="image">
~~~~~~~~~~~~~~~~~~~~
```

```html
<input [type]="'image'">
~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```html
<img src="foo" alt="Foo eating a sandwich.">
```

```html
<img src="foo" [attr.alt]="altText">
```

```html
<img src="foo" [attr.alt]="'Alt Text'">
```

```html
<img src="foo" alt="">
```

```html
<object aria-label="foo">
```

```html
<object aria-labelledby="id1">
```

```html
<object>Meaningful description</object>
```

```html
<object title="An object">
```

```html
<area aria-label="foo"></area>
```

```html
<area aria-labelledby="id1"></area>
```

```html
<area alt="This is descriptive!"></area>
```

```html
<input type="text">
```

```html
<input type="image" alt="This is descriptive!">
```

```html
<input type="image" aria-label="foo">
```

```html
<input type="image" aria-labelledby="id1">
```
