<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/accessibility-table-scope.ts
  - ../../tests/rules/accessibility-table-scope/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/accessibility-table-scope`

Ensures that the `scope` attribute is only used on the `<th>` element

- Type: suggestion
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-table-scope": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ test }}<div scope></div>
               ~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-table-scope": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div [attr.scope]="scope"></div><p></p>
     ~~~~~~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-table-scope": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<th></th>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-table-scope": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<th scope="col"></th>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-table-scope": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<th [scope]="'col'"></th>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-table-scope": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<th [attr.scope]="scope"></th>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-table-scope": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div Scope="col"></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-table-scope": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<button [appscope]="col"></button>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-table-scope": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<app-table scope></app-table>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-table-scope": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<app-row [scope]="row"></app-row>
```

</details>

<br>
