<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/accessibility-elements-content.ts
  - ../../tests/rules/accessibility-elements-content/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/accessibility-elements-content`

Ensures that the heading, anchor and button elements have content in it

- Type: suggestion

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-elements-content": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<h1 class="size-1"></h1>
~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-elements-content": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<a href="#" [routerLink]="['route1']"></a>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-elements-content": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<button></button>
~~~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-elements-content": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<h1>Heading Content!</h1>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-elements-content": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<h2><app-content></app-content></h2>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-elements-content": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<h3 [innerHtml]="dangerouslySetHTML"></h3>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-elements-content": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<h4 [innerText]="text"></h4>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-elements-content": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<a>Anchor Content!</a>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-elements-content": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<a><app-content></app-content></a>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-elements-content": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<a [innerHTML]="dangerouslySetHTML"></a>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-elements-content": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<a [innerText]="text"></a>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-elements-content": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<a [outerHTML]="text"></a>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-elements-content": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<a aria-hidden></a>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-elements-content": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<button [attr.aria-hidden]="true"></button>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-elements-content": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<h5 [attr.aria-label]="text"></h5>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/accessibility-elements-content": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<h6 title="text"></h6>
```

</details>

<br>
