<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/accessibility-alt-text.ts
  - ../../tests/rules/accessibility-alt-text/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/template/accessibility-alt-text`

Ensures alternate text for elements which require the `alt`, `aria-label`, `aria-labelledby` or `title` attribute

- Type: suggestion
- Category: Best Practices

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

‚ùå - Examples of **incorrect** code for this rule:

```html
<area>
~~~~~~
```

```html
<area alt>
      ~~~
```

```html
<area [alt]="null">
      ~~~~~~~~~~~~
```

```html
<area aria-label>
      ~~~~~~~~~~
```

```html
<area aria-labelledby="">
      ~~~~~~~~~~~~~~~~~~
```

```html
<img alt>
     ~~~
```

```html
<img [alt]="undefined">
     ~~~~~~~~~~~~~~~~~
```

```html
<img aria-label="">
     ~~~~~~~~~~~~~
```

```html
<img [attr.aria-labelledby]="">
     ~~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<img role="none">
     ~~~~~~~~~~~
```

```html
<img [attr.role]="'presentation'">
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<ng-template>
  <a href="/maps/nav.map"><img src="/images/navbar.gif" ismap></a>
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
</ng-template>
```

```html
<input type="image">
~~~~~~~~~~~~~~~~~~~~
```

```html
<input type="image" [alt]="null">
                    ~~~~~~~~~~~~
```

```html
<input aria-label="" [type]="'image'">
       ~~~~~~~~~~~~~
```

```html
<input [attr.aria-labelledby]="null" [attr.type]="'image'">
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<object></object>
~~~~~~~~~~~~~~~~~
```

```html
<object [innerHttp]="testing"><i aria-hidden></i></object>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<object aria-label=""></object>
        ~~~~~~~~~~~~~
```

```html
<object aria-labelledby></object>
        ~~~~~~~~~~~~~~~
```

```html
<object [title]="null"></object>
        ~~~~~~~~~~~~~~
<object title=""></object>
        ~~~~~~~~
<object [attr.title]="undefined"></object>
        ~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

‚úÖ - Examples of **correct** code for this rule:

```html
<div></div>
```

```html
<app-top-bar [attr.aria-label]="'Aria'"></app-top-bar>
```

```html
<area alt="">
```

```html
<area alt=" ">
```

```html
<area [alt]="">
```

```html
<area alt="This is descriptive!">
```

```html
<area aria-label="id1">
```

```html
<area [attr.aria-labelledby]="'foo'">
```

```html
<img alt="">
```

```html
<img alt=" ">
```

```html
<img alt="foo">
```

```html
<img alt="{{null}}">
```

```html
<img [attr.alt]="'foo'">
```

```html
<img [alt]="alt">
```

```html
<img [alt]="alt ?? 'Alt text'">
```

```html
<img [alt]="foo.bar || ''">
```

```html
<img [alt]="{{alt + 'Alt text'}}">
```

```html
<img [alt]="getAlt()">
```

```html
<img [alt]="error ? 'not working' : 'working'">
```

```html
<img alt="" role="none">
```

```html
<img [alt]="''" role="presentation">
```

```html
<img alt="" [attr.role]="none">
```

```html
<img alt="" [role]="'presentation'">
```

```html
<img alt="" aria-label="foo">
```

```html
<img alt="" [attr.aria-labelledby]="'id0'">
```

```html
<img [attr.aria-label]="'foo'">
```

```html
<img aria-labelledby="id1">
```

```html
<img src="images/solar_system.jpg" alt="Solar System" width="472" height="800" usemap="#map">
<map name="map">
  <area shape="rect" coords="115,158,276,192" href="http://en.wikipedia.org/wiki/Mercury_%28planet%29" alt="Mercury">
  <area shape="rect" coords="115,193,276,234" href="http://en.wikipedia.org/wiki/Venus" alt="Venus">
</map>
```

```html
<input>
```

```html
<input type="text">
```

```html
<input type="image" alt="">
```

```html
<input type="image" alt=" ">
```

```html
<input type="image" alt="This is descriptive!">
```

```html
<input aria-label="">
```

```html
<input type="image" aria-label="foo">
```

```html
<input type="image" [attr.aria-labelledby]="id1">
```

```html
<object aria-label="foo"></object>
```

```html
<object [attr.aria-labelledby]="id1"></object>
```

```html
<object innerHtml="foo"></object>
```

```html
<object [innerHTML]="foo"></object>
```

```html
<object [innerText]="foo"></object>
```

```html
<object outerHTML="foo"></object>
```

```html
<object title="An object"></object>
```

```html
<object>Meaningful description</object>
```

```html
<object><p>Meaningful description</p></object>
```
