<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/accessibility-table-scope.ts
  - ../../tests/rules/accessibility-table-scope/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/template/accessibility-table-scope`

Ensures that the `scope` attribute is only used on the `<th>` element

- Type: suggestion
- Category: Best Practices
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

‚ùå - Examples of **incorrect** code for this rule:

```html
{{ test }}<div scope></div>
               ~~~~~
```

```html
<div [attr.scope]="scope"></div><p></p>
     ~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

‚úÖ - Examples of **correct** code for this rule:

```html
<th></th>
```

```html
<th scope="col"></th>
```

```html
<th [scope]="'col'"></th>
```

```html
<th [attr.scope]="scope"></th>
```

```html
<div Scope="col"></div>
```

```html
<button [appscope]="col"></button>
```

```html
<app-table scope></app-table>
```

```html
<app-row [scope]="row"></app-row>
```
