<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/accessibility-label-has-associated-control.ts
  - ../../tests/rules/accessibility-label-has-associated-control/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/template/accessibility-label-has-associated-control`

Ensures that a label element/component is associated with a form element

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  controlComponents?: string[];
  labelComponents?: {
    inputs?: string[];
    selector: string;
  }[];
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```html
<label>Label</label>
~~~~~~~~~~~~~~~~~~~~
```

```html
<app-label anotherAttribute="id"></app-label>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```html
<ng-container *ngFor="let item of items; index as index">
  <label for="item-{{index}}">Label #{{index}</label>
  <input id="item-{{index}}" [(ngModel)]="item.name">
</ng-container>
<label for="id"></label>
<label for="{{id}}"></label>
<label [attr.for]="id"></label>
<label [htmlFor]="id"></label>
```
