<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/attributes-order.ts
  - ../../tests/rules/attributes-order/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/attributes-order`

Ensures that HTML attributes and Angular bindings are sorted based on an expected order

- Type: layout
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * Default: `false`
   */
  alphabetical?: boolean;
  /**
   * Default: `["STRUCTURAL_DIRECTIVE","TEMPLATE_REFERENCE","ATTRIBUTE_BINDING","INPUT_BINDING","TWO_WAY_BINDING","OUTPUT_BINDING"]`
   *
   * @minItems 6
   */
  order?: [
    (
      | "STRUCTURAL_DIRECTIVE"
      | "TEMPLATE_REFERENCE"
      | "ATTRIBUTE_BINDING"
      | "INPUT_BINDING"
      | "TWO_WAY_BINDING"
      | "OUTPUT_BINDING"
    ),
    (
      | "STRUCTURAL_DIRECTIVE"
      | "TEMPLATE_REFERENCE"
      | "ATTRIBUTE_BINDING"
      | "INPUT_BINDING"
      | "TWO_WAY_BINDING"
      | "OUTPUT_BINDING"
    ),
    (
      | "STRUCTURAL_DIRECTIVE"
      | "TEMPLATE_REFERENCE"
      | "ATTRIBUTE_BINDING"
      | "INPUT_BINDING"
      | "TWO_WAY_BINDING"
      | "OUTPUT_BINDING"
    ),
    (
      | "STRUCTURAL_DIRECTIVE"
      | "TEMPLATE_REFERENCE"
      | "ATTRIBUTE_BINDING"
      | "INPUT_BINDING"
      | "TWO_WAY_BINDING"
      | "OUTPUT_BINDING"
    ),
    (
      | "STRUCTURAL_DIRECTIVE"
      | "TEMPLATE_REFERENCE"
      | "ATTRIBUTE_BINDING"
      | "INPUT_BINDING"
      | "TWO_WAY_BINDING"
      | "OUTPUT_BINDING"
    ),
    (
      | "STRUCTURAL_DIRECTIVE"
      | "TEMPLATE_REFERENCE"
      | "ATTRIBUTE_BINDING"
      | "INPUT_BINDING"
      | "TWO_WAY_BINDING"
      | "OUTPUT_BINDING"
    ),
    ...(
      | "STRUCTURAL_DIRECTIVE"
      | "TEMPLATE_REFERENCE"
      | "ATTRIBUTE_BINDING"
      | "INPUT_BINDING"
      | "TWO_WAY_BINDING"
      | "OUTPUT_BINDING"
    )[]
  ];
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error",
      {
        "alphabetical": true
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<li><input type="text" id="input"></li>
           ~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input #inputRef *ngIf="flag" class="className">
       ~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input *ngIf="flag" class="className" #inputRef [binding]="true" [(ngModel)]="model" (output)="handleOutput($event)">
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input *ngFor="inputs" [binding]="true" class="className" (output)="handleOutput($event)">
                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input *ngIf="flag" #inputRef class="className" [(ngModel)]="model" [binding]="true" (output)="handleOutput($event)">
                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input *ngIf="flag" #inputRef class="className" (output)="handleOutput($event)" [binding]="true">
                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error",
      {
        "alphabetical": false,
        "order": [
          "TEMPLATE_REFERENCE",
          "ATTRIBUTE_BINDING",
          "STRUCTURAL_DIRECTIVE",
          "INPUT_BINDING",
          "OUTPUT_BINDING",
          "TWO_WAY_BINDING"
        ]
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input *ngIf="flag" class="className">
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error",
      {
        "alphabetical": true,
        "order": [
          "TEMPLATE_REFERENCE",
          "STRUCTURAL_DIRECTIVE",
          "ATTRIBUTE_BINDING",
          "INPUT_BINDING",
          "OUTPUT_BINDING",
          "TWO_WAY_BINDING"
        ]
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input *ngIf="flag" [(ngModel)]="model" #inputRef id="input" class="className" [binding]="true" (output)="handleOutput($event)">
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input
  [(ngModel)]="model"
  ~~~~~~~~~~~~~~~~~~~
  *ngIf="flag"
  ~~~~~~~~~~~~
  #inputRef
  ~~~~~~~~~
  id="input"
  ~~~~~~~~~~
  class="className"
  ~~~~~~~~~~~~~~~~~
  (output)="handleOutput($event)"
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  [binding]="true">
  ~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<ng-container (click)="bar = []" id="issue" *ngFor="let foo of bar"></ng-container>
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<ng-container (click)="bar = []" id="issue" *ngFor="let foo of bar; index as i; first as isFirst"></ng-container>
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div id="id" *ngIf="bar as foo"></div>
     ~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div id="id" *ngIf="condition then foo else bar"></div>
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error",
      {
        "alphabetical": true
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div [disabled]="disabled" [class.disabled]="disabled"></div>
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error",
      {
        "alphabetical": true
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<ng-template let-value #Template></ng-template>
             ~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error",
      {
        "alphabetical": true
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<ng-template let-value="something" let-anotherValue="else" #Template></ng-template>
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error",
      {
        "alphabetical": true
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<svg>
  <ng-template let-value #Template></ng-template>
               ~~~~~~~~~~~~~~~~~~~
</svg>
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error",
      {
        "alphabetical": true
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<td mat-cell *matCellDef="let element"></td>
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error",
      {
        "alphabetical": true
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<td mat-cell *matCellDef="
    ~~~~~~~~~~~~~~~~~~~~~~
  let element
  ~~~~~~~~~~~
"></td>
~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error",
      {
        "alphabetical": true
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div class="abc" *ngIf="sth.property as property "></div>
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error",
      {
        "alphabetical": true,
        "order": [
          "ATTRIBUTE_BINDING",
          "TEMPLATE_REFERENCE",
          "INPUT_BINDING",
          "OUTPUT_BINDING",
          "TWO_WAY_BINDING",
          "STRUCTURAL_DIRECTIVE"
        ]
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div *structuralDirective class="abc"></div>
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error",
      {
        "alphabetical": true
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div title="abc" *structuralDirective abbr="abc"></div>
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div class="abc" *structuralDirective></div>
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div i18n-alpha="a1" beta="b" alpha="a" i18n-beta="b1"></div>
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div i18n="i" beta="b" alpha="a" pi="p">x</div>
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div #alpha *ngIf="true" (epsilon)="e" beta="b" [gamma]="'g'" i18n-beta="b18" [(delta)]="d" pi="p"></div>
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div [beta]="'b'" alpha="a" [gamma]="g" i18n-alpha="a18" i18n-beta="b18"></div>
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<ng-template i18n="@@a:b" let-foo #template>The value is {{ foo }}.</ng-template>
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<input class="card" [value]="foo" (valueChange)="handleValueChange($event)">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<input *ngIf="flag" #inputRef id="input" class="className" [binding]="true" [(ngModel)]="model" (output)="handleOutput($event)">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<input *ngIf="flag" (output)="handleOutput($event)">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<input *ngIf="flag" required>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<input [(ngModel)]="model">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<input [(ngModel)]="model" (ngModelChange)="onChange($event)">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-template></ng-template>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-template #Template><div></div></ng-template>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-template #Template i18n="@@test"><div>a</div></ng-template>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-template [ngIf]="condition" [ngIfThen]="If" [ngIfElse]="Else"><div></div></ng-template>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-template i18n="@@test" [ngIf]="condition" [ngIfThen]="If" [ngIfElse]="Else"><div></div></ng-template>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-template #Template let-value><div></div></ng-template>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-template #Template let-value i18n="@@test">c</ng-template>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div test1="test1" test2="test2"></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div i18n="@@test" test1="test1" test2="test2"></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div i18n="@@test" test1="test1" i18n-test1="@@TEST1" test2="test2" i18n-test2="@@TEST2"></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div test1="test1" i18n-test1="@@TEST1" test2="test2" i18n-test2="@@TEST2"></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div beta="" [alpha]="" i18n-alpha="a" [gamma]="" i18n-gamma="g"></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<svg><ng-template #Template let-value><line x1="1" x2="2" y1="3" y2="4"></line></ng-template></svg>
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error",
      {
        "alphabetical": true,
        "order": []
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div alpha="a" beta="b" gamma="g"></div>
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/attributes-order": [
      "error",
      {
        "alphabetical": true,
        "order": []
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div alpha="a" i18n-alpha="a18n" beta="b" i18n-beta="b18n" gamma="g" i18n-gamma="g18n"></div>
```

</details>

<br>
