<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/banana-in-box.ts
  - ../../tests/rules/banana-in-box/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/banana-in-box`

Ensures that the two-way data binding syntax is correct

- Type: suggestion
- üîß Supports autofix (`--fix`)

<br>

## Rationale

The Angular two-way binding syntax is [(ngModel)]="value", where the parentheses are INSIDE the brackets, resembling a banana in a box. A common typo is ([ngModel])="value" with the parentheses outside, which creates a one-way binding FROM the template TO the component (the opposite of what's intended). This will make the input update when the value changes but won't update the value when the user types. This syntax error is easy to make and can be hard to spot visually, but it breaks two-way data binding. The correct syntax is: [()] for two-way binding, [] for property binding (component to template), and () for event binding (template to component).

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/banana-in-box": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input type="text" name="foo" ([ngModel])="foo">
                              ~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/banana-in-box": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<app-item ([bar])="bar" ([item])="item" [(test)]="test"></app-item>
          ~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~
<div [baz]="oneWay" (emitter)="emitter" ([twoWay])="twoWay"></div>
                                        ~~~~~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/banana-in-box": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<input type="text" name="foo" [ngModel]="foo">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/banana-in-box": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<input type="text" name="foo" [(ngModel)]="foo">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/banana-in-box": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<button type="button" (click)="navigate(['/resources'])">
  Navigate
</button>
```

</details>

<br>
