<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/banana-in-box.ts
  - ../../tests/rules/banana-in-box/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/template/banana-in-box`

Ensures that the two-way data binding syntax is correct

- Type: suggestion
- Category: Best Practices
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

‚ùå - Examples of **incorrect** code for this rule:

```html
<input type="text" name="foo" ([ngModel])="foo">
                              ~~~~~~~~~~~~~~~~~
```

```html
<app-item ([bar])="bar" ([item])="item" [(test)]="test"></app-item>
          ~~~~~~~~~~~~~ ^^^^^^^^^^^^^^^
<div [baz]="oneWay" (emitter)="emitter" ([twoWay])="twoWay"></div>
                                        ###################
```

<br>

---

<br>

‚úÖ - Examples of **correct** code for this rule:

```html
<input type="text" name="foo" [ngModel]="foo">
```

```html
<input type="text" name="foo" [(ngModel)]="foo">
```

```html
<button type="button" (click)="navigate(['/resources'])">
  Navigate
</button>
```
