<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/click-events-have-key-events.ts
  - ../../tests/rules/click-events-have-key-events/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/template/click-events-have-key-events`

Ensures that the click event is accompanied with at least one key event keyup, keydown or keypress.

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```html
<div (click)="onClick()"></div>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<header (click)="onClick()"></header>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<a (click)="onClick()"></a>
~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<div (click)="onClick()" aria-hidden="false"></div>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<div (click)="onClick()" [attr.aria-hidden]="'false'"></div>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<div (click)="onClick()" role="header"></div>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<div (click)="onClick()" role="aside"></div>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<div (click)="onClick()" [attr.role]="'header'"></div>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:
