<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/click-events-have-key-events.ts
  - ../../tests/rules/click-events-have-key-events/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/template/click-events-have-key-events`

Ensures that the `click` event is accompanied by at least one of the following: `keydown`, `keypress` or `keyup` events. Coding for the keyboard is important for users with physical disabilities who cannot use a mouse, AT compatibility, and screenreader users. This does not apply for interactive or hidden elements. See more at https://www.w3.org/WAI/WCAG21/Understanding/keyboard

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```html
<div (click)="onClick()"></div>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<a (click)="onClick()"></a>
~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<div (click)="onClick()" aria-hidden="false"></div>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<div (click)="onClick()" [attr.aria-hidden]="ariaHidden"></div>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<div (click)="onClick()" role="header"></div>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<a (click)="onClick()" [attr.role]="'header'"></a>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<div (click)="onClick()" [role]="role"></div>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```html
<div (click)="onClick()" (keyup)="onKeyup()"></div>
```

```html
<div (click)="onClick()" (keypress.enter)="onKeypress()"></div>
```

```html
<div (click)="onClick()" (keydown.shift.f)="onKeydown()"></div>
```

```html
<cui-button (click)="onClick()"></cui-button>
```

```html
<div (click)="onClick()" hidden></div>
<div style="display: none">
  <header (click)="onClick()"></header>
</div>
<div (click)="onClick()" [style.display.none]="true"></div>
<div (click)="onClick()" [ngStyle]="{visibility: 'hidden'}"></div>
<div (click)="onClick()" aria-hidden></div>
<div (click)="onClick()" aria-hidden="true"></div>
<div (click)="onClick()" [attr.aria-hidden]="true"></div>
<div (click)="onClick()" [attr.aria-hidden]="'true'"></div>
```

```html
<div (click)="onClick()" role="presentation"></div>
<div (click)="onClick()" [attr.role]="'none'"></div>
```

```html
<input (click)="onClick()">
<button (click)="onClick()"></button>
<textarea (click)="onClick()"></textarea>
<select (click)="onClick()">
  <option (click)="onClick()"></option>
</select>
<textarea (click)="onClick()"></textarea>
<a href="#" (click)="onClick()"></a>
<a [attr.href]="href" class="anchor" (click)="onClick()"></a>
<a [routerLink]="'route'" (click)="onClick()"></a>
```
