<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/cyclomatic-complexity.ts
  - ../../tests/rules/cyclomatic-complexity/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/template/cyclomatic-complexity`

Checks cyclomatic complexity against a specified limit. It is a quantitative measure of the number of linearly independent paths through a program's source code

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * Default is: `5`
   */
  maxComplexity?: number;
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```html
<div *ngIf="a === '1'">
  <div *ngFor="let person of persons; trackBy: trackByFn">
    <div *ngIf="a === '1'">{{ person.name }}</div>
    <div [ngSwitch]="person.emotion">
      <app-happy-hero    *ngSwitchCase="'happy'"    [hero]="currentHero"></app-happy-hero>
      <app-sad-hero      *ngSwitchCase="'sad'"      [hero]="currentHero"></app-sad-hero>
      <app-confused-hero *ngSwitchCase="'confused'" [hero]="currentHero"></app-confused-hero>
                          ~~~~~~~~~~~~~~~~~~~~~~~~
      <app-unknown-hero  *ngSwitchDefault           [hero]="currentHero"></app-unknown-hero>
                          ^^^^^^^^^^^^^^^
    </div>
  </div>
</div>
```

```html
<div [fakeDirective]="'test'"></div>
<ng-template ngFor let-person [ngForOf]="persons" let-i="index">
  {{ person.name }}
</ng-template>
<ng-template [ngIf]="a === '1'">
  something here
</ng-template>
<div *ngIf="a === '1'">
  <div *ngFor="let person of persons; trackBy: trackByFn">
    <div *ngIf="a === '1'">{{ person.name }}</div>
    <div [ngSwitch]="person.emotion">
      <app-happy-hero    *ngSwitchCase="'happy'"    [hero]="currentHero"></app-happy-hero>
      <app-sad-hero      *ngSwitchCase="'sad'"      [hero]="currentHero"></app-sad-hero>
                          ~~~~~~~~~~~~~~~~~~~
      <app-confused-hero *ngSwitchCase="'confused'" [hero]="currentHero"></app-confused-hero>
                          ^^^^^^^^^^^^^^^^^^^^^^^^
      <app-unknown-hero  *ngSwitchDefault           [hero]="currentHero"></app-unknown-hero>
                          ###############
    </div>
  </div>
</div>
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:
