<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/eqeqeq.ts
  - ../../tests/rules/eqeqeq/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/template/eqeqeq`

Requires `===` and `!==` in place of `==` and `!=`

- Type: suggestion
- Category: Best Practices
- üîß Supports autofix (`--fix`)

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  allowNullOrUndefined?: boolean;
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

‚ùå - Examples of **incorrect** code for this rule:

```html
{{ 'null' == test }}
   ~~~~~~~~~~~~~~
```

```html
<div [attr.disabled]="test != 'undefined' && null == '3'"></div>
                      ~~~~~~~~~~~~~~~~~~~
```

```html
<div *ngIf="test == true || test1 !== undefined"></div>
            ~~~~~~~~~~~~
```

```html
{{ one != '02' ? c > d : 'hey!' }}
   ~~~~~~~~~~~
```

```html
{{ a === b && 1 == b ? c > d : 'hey!' }}
              ~~~~~~
```

```html
{{ c > d ? a != b : 'hey!' }}
           ~~~~~~
```

```html
{{ c > d ? 'hey!' : a == false }}
                    ~~~~~~~~~~
```

```html
{{ undefined == test1 && a === b ? (c > d ? d != '0' : v === 4) : 'hey!' }}
                                            ~~~~~~~~
```

```html
{{ undefined != test1 }}
   ~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

‚úÖ - Examples of **correct** code for this rule:

```html
{{ a === 1 }}
```

```html
<div [class.testing]="b === false">
```

```html
<div *ngIf="c === test">
```
