<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/eqeqeq.ts
  - ../../tests/rules/eqeqeq/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/eqeqeq`

Requires `===` and `!==` in place of `==` and `!=`

- Type: suggestion
- üîß Supports autofix (`--fix`)

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  allowNullOrUndefined?: boolean;
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/eqeqeq": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ 'null' == test }}
   ~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/eqeqeq": [
      "error",
      {
        "allowNullOrUndefined": true
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div [attr.disabled]="test != 'undefined' && null == '3'"></div>
                      ~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/eqeqeq": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div *ngIf="test == true || test1 !== undefined"></div>
            ~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/eqeqeq": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ one != '02' ? c > d : 'hey!' }}
   ~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/eqeqeq": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ a === b && 1 == b ? c > d : 'hey!' }}
              ~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/eqeqeq": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ c > d ? a != b : 'hey!' }}
           ~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/eqeqeq": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ c > d ? 'hey!' : a == false }}
                    ~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/eqeqeq": [
      "error",
      {
        "allowNullOrUndefined": true
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ undefined == test1 && a === b ? (c > d ? d != '0' : v === 4) : 'hey!' }}
                                            ~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/eqeqeq": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ undefined != test1 }}
   ~~~~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/eqeqeq": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ a === 1 }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/eqeqeq": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div [class.testing]="b === false">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/eqeqeq": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div *ngIf="c === test">
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/eqeqeq": [
      "error",
      {
        "allowNullOrUndefined": true
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div *appShow="(d == null && e === null && (f | lowercase) == undefined) || g === undefined">
```

</details>

<br>
