<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/headings-consecutive-order.ts
  - ../../tests/rules/headings-consecutive-order/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/headings-consecutive-order`

[Accessibility] Ensures that headings have consecutive order from largest to lowest h1 -> h6

- Type: suggestion

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/headings-consecutive-order": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<h2>Heading 2</h2>
<h4>Heading 4</h4>
~~~~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/headings-consecutive-order": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<h1>Heading 1</h1>
            <h2>Heading 2</h2>
            <h2>Heading 2</h2>
            <h2>Heading 2</h2>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/headings-consecutive-order": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<h1>Heading 1</h1>
            <h2>Heading 2</h2>
            <h2>Heading 2</h2>
            <h3>Heading 3</h3>
            <h3>Heading 3</h3>
            <h3>Heading 3</h3>
            <h4>Heading 4</h4>
            <div>valid</div>
            <h2>Heading 2</h2>
            <h3>Heading 3</h3>
            <h3>Heading 3</h3>
            <h4>Heading 4</h4>
            <h2>Heading 2</h2>
            <h3>Heading 3</h3>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/headings-consecutive-order": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<h2>Heading 2</h2>
            <h2>Heading 2</h2>
            <h3>Heading 3</h3>
            <h3>Heading 3</h3>
            <h1>Heading 1</h1>
            <h2>Heading 2</h2>
            <h2>Heading 2</h2>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/headings-consecutive-order": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<h2>Heading 2</h2>
            <ng-template>
                <div>valid</div>
                <h6>ignore</h6>
            </ng-template>
            <h3>Heading 3</h3>
```

</details>

<br>
