<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/i18n.ts
  - ../../tests/rules/i18n/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/template/i18n`

Ensures following best practices for i18n. Checks for missing i18n attributes on elements and attributes containing texts. Can also check for texts without i18n attribute, elements that do not use custom ID (@@) feature and duplicate custom IDs

- Type: suggestion
- Category: Best Practices
- üîß Supports autofix (`--fix`)

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  boundTextAllowedPattern?: string;
  /**
   * Default: `true`
   */
  checkAttributes?: boolean;
  /**
   * Default: `true`
   */
  checkId?: boolean;
  /**
   * Default: `true`
   */
  checkText?: boolean;
  /**
   * Default: `["charset","class","color","colspan","fill","formControlName","height","href","id","lang","ngClass","ngProjectAs","routerLink","src","stroke","stroke-width","style","svgIcon","tabindex","target","type","viewBox","width","xmlns"]`
   */
  ignoreAttributes?: string[];
  ignoreTags?: string[];
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

‚ùå - Examples of **incorrect** code for this rule:

```html
<div tooltip="This requires translation"></div>
     ~~~~~~~
```

```html
<div>
  <span>-{{data_from_backend}}</span>
        ~~~~~~~~~~~~~~~~~~~~~~
</div>
```

```html
{ value, plural, =0 {<div>No elements</div>} =1 {111} }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<div>
  <ng-container>Some text&nbsp;t@ tr1nslate</ng-container>
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~
</div>
```

```html
<p>Lorem ipsum <em i18n="@@dolor">dolor</em> sit amet.</p>
   ~~~~~~~~~~~~                              ~~~~~~~~~
```

```html
<div tooltip="This requires translation" i18n-tooltip></div>
     ~~~~~~~
```

```html
<div>
  <span i18n label="label is ignored in 'ignoreAttributes'">
  ~
    Missing custom ID
  </span>
        ~
</div>
```

```html
<div
  i18n-tooltip="@@custom-id"
  tooltip="This requires translation"
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  label="Custom label"
  ~~~~~~~~~~~~~~~~~~~~
  i18n-label="@@custom-id"
></div>
```

```html
<h3 i18n="@@myId">Hello</h3>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<p i18n="@@myId">Good bye</p>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<div i18n-tooltip="@@custom-id" tooltip="This requires translation">
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  <span i18n="@@custom-id">Some text to translate</span>
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
</div>
```

```html
<div i18n-tooltip="@@custom-id" tooltip="This requires translation">
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  <span i18n="@@custom-id">Some text to translate</span>
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
</div>
<div i18n-label="@@custom-id" label="A label"></div>
                              ~~~~~~~~~~~~~~~
```

```html
<div
  tooltip="This requires translation"
  ~~~~~~~
  i18n-placeholder
  placeholder="More translation, please"
  ~~~~~~~~~~~
  class="red"
>
  <div
    *ngIf="true"
    width="100px"
    label="Templates need translation too."
    ~~~~~
  >
    <span i18n label="label is ignored in 'ignoreAttributes'">
    ~
      Missing custom ID
    </span>
          ~
  </div>
</div>
```

<br>

---

<br>

‚úÖ - Examples of **correct** code for this rule:

```html
<div>
  <span i18n="@@custom-id">Some text to translate</span>
</div>
```

```html
<div>
  <span class="red" i18n="@@custom-id">
    Some text to translate
  </span>
</div>
```

```html
<a
  mat-button
  ngClass="class"
  routerLink="exclusions"
  i18n="@@keywording.tools.exclusions"
>
  Exclusions
</a>
```

```html
<ng-container i18n="@@description">
  { value, plural, =0 {<div>No elements</div>} =1 {111} }
</ng-container>
```

```html
<mat-option *ngFor="let mode of modes" [value]="mode.id" i18n="@@option">
  {mode.name, select, mode {mode} other { {{mode.name}} } }
</mat-option>
```

```html
<ng-container ngProjectAs="top">&ngsp;</ng-container>
```

```html
<div label="1">
  <div matBadge="&#8288;">5</div>
</div>
```

```html
<div ariaselected="0"></div>
<div>+</div>
<span>&nbsp;</span>
<span>123</span>
<ng-content select=".content-area"></ng-content>
<ul i18n="@@list">
  <li>ItemA</li>
  <li>ItemB</li>
  <li>ItemC</li>
</ul>
```
