<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/i18n.ts
  - ../../tests/rules/i18n/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/template/i18n`

Helps to ensure following best practices for i18n. Checks for missing i18n attributes on elements and non-ignored attributes containing text. Can also highlight tags that do not use custom ID (@@) feature.

- Type: suggestion
- Category: Best Practices
- üîß Supports autofix (`--fix`)

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  boundTextAllowedPattern?: string;
  /**
   * Default: `true`
   */
  checkId?: boolean;
  /**
   * Default: `true`
   */
  checkText?: boolean;
  /**
   * Default: `true`
   */
  checkAttributes?: boolean;
  /**
   * Default: `["charset","class","color","colspan","fill","formControlName","height","href","id","lang","ngClass","routerLink","src","stroke","stroke-width","style","svgIcon","tabindex","target","type","viewBox","width","xmlns"]`
   */
  ignoreAttributes?: string[];
  ignoreTags?: string[];
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

‚ùå - Examples of **incorrect** code for this rule:

```html
<div>
  <span>Some text to translate</span>
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
</div>
```

```html
<div>
  <span>-{{data_from_backend}}</span>
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
</div>
```

<br>

---

<br>

‚úÖ - Examples of **correct** code for this rule:

```html
<div>
  <span i18n="@@my-custom-id">Some text to translate</span>
</div>
```

```html
<div i18n-tooltip="@@my-custom-id" tooltip="This also requires translation">
  <span i18n="@@my-custom-id">Some text to translate</span>
</div>
```

```html
<div>
  <span class="red" i18n="@@my-custom-id">
    Some text to translate
  </span>
</div>
```
