<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-autofocus.ts
  - ../../tests/rules/no-autofocus/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/no-autofocus`

[Accessibility] Ensures that the `autofocus` attribute is not used

- Type: suggestion
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<button autofocus>Click me!</button>
        ~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input [attr.autofocus]="false">
       ~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<app-test [autofocus]="true"></app-test>
<select autofocus></select>
        ~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<input type="text">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<textarea autoFocus></textarea>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div [autoFocus]="true"></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<button [appautofocus]="false">Click me!</button>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<app-drag-drop autofocus></app-drag-drop>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<app-textarea [autofocus]="false"></app-textarea>
```

</details>

<br>
