<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-any.ts
  - ../../tests/rules/no-any/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/no-any`

The use of "$any" nullifies the compile-time benefits of Angular's type system

- Type: suggestion

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ $any(framework).name }}
   ~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ this.$any(framework).name }}
   ~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<a [href]="$any(getHref())">Click here</a>
           ~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ $any(framework).name }}
   ~~~~~~~~~~~~~~~
<div>
  {{ this.$any(framework).name }}
     ~~~~~~~~~~~~~~~~~~~~
</div>
<a [href]="$any(getHref())">Click here</a>'
           ~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ $any }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ obj.$any() }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ obj?.x?.y!.z!.$any() }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<a [href]="$test()">Click here</a>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<button type="button" (click)="anyClick()">Click here</button>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ $any }}
{{ obj?.x?.y!.z!.$any() }}
<a [href]="$test()">Click here</a>
<button type="button" (click)="anyClick()">Click here</button>
```

</details>

<br>
