<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-any.ts
  - ../../tests/rules/no-any/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/no-any`

The use of "$any" nullifies the compile-time benefits of Angular's type system

- Type: suggestion

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rationale

The $any() type cast function in Angular templates disables TypeScript's type checking for an expression, removing all the benefits of static typing. This means typos, incorrect property access, and invalid method calls won't be caught until runtime. While $any() can be a quick fix for type errors, it's better to properly type your component properties or fix the underlying type issue. Overusing $any() makes templates less safe and can hide real bugs. If you're using $any() frequently, it often indicates that your component's types need improvement. Use $any() sparingly and only when you have a specific reason why TypeScript's type checking isn't applicable.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ $any(framework).name }}
   ~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ this.$any(framework).name }}
   ~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<a [href]="$any(getHref())">Click here</a>
           ~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div *ngIf="$any(attributeList)['NPSScore']">Content</div>
            ~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div>{{ $any(attributeList)['NPSScore'] }}</div>
        ~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div>{{ this.$any(attributeList)['NPSScore'] }}</div>
        ~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div>{{ $any(attributeList)['NPSScore']['another'] }}</div>
        ~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ $any(framework).name }}
   ~~~~~~~~~~~~~~~
<div>
  {{ this.$any(framework).name }}
     ~~~~~~~~~~~~~~~~~~~~
</div>
<a [href]="$any(getHref())">Click here</a>'
           ~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ $any }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ obj.$any() }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ obj?.x?.y!.z!.$any() }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ obj[read][$any] }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ $any['test'] }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ obj['test'].$any }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ obj['test'].$any() }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ obj.$any()['test'] }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ test()['value'] }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<a [href]="$test()">Click here</a>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<button type="button" (click)="anyClick()">Click here</button>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-any": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ $any }}
{{ obj?.x?.y!.z!.$any() }}
<a [href]="$test()">Click here</a>
<button type="button" (click)="anyClick()">Click here</button>
```

</details>

<br>
