<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-any.ts
  - ../../tests/rules/no-any/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/template/no-any`

The use of "$any" nullifies the compile-time benefits of Angular's type system

- Type: suggestion
- Category: Best Practices

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

‚ùå - Examples of **incorrect** code for this rule:

```html
{{ $any(framework).name }}
   ~~~~~~~~~~~~~~~
```

```html
{{ this.$any(framework).name }}
   ~~~~~~~~~~~~~~~~~~~~
```

```html
<a [href]="$any(getHref())">Click here</a>
           ~~~~~~~~~~~~~~~
```

```html
{{ $any(framework).name }}
   ~~~~~~~~~~~~~~~
<div>
  {{ this.$any(framework).name }}
     ~~~~~~~~~~~~~~~~~~~~
</div>
<a [href]="$any(getHref())">Click here</a>'
           ~~~~~~~~~~~~~~~
```

<br>

---

<br>

‚úÖ - Examples of **correct** code for this rule:

```html
{{ $any }}
```

```html
{{ obj.$any() }}
```

```html
{{ obj?.x?.y!.z!.$any() }}
```

```html
<a [href]="$test()">Click here</a>
```

```html
<button type="button" (click)="anyClick()">Click here</button>
```

```html
{{ $any }}
{{ obj?.x?.y!.z!.$any() }}
<a [href]="$test()">Click here</a>
<button type="button" (click)="anyClick()">Click here</button>
```
