<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-autofocus.ts
  - ../../tests/rules/no-autofocus/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/no-autofocus`

[Accessibility] Ensures that the `autofocus` attribute is not used

- Type: suggestion
- üîß Supports autofix (`--fix`)

<br>

## Rationale

The autofocus attribute automatically moves focus to an element when the page loads, which can be disorienting and problematic for accessibility. For screen reader users, autofocus disrupts their normal navigation flow and can cause them to miss important page content. For users with motor disabilities, unexpected focus changes can be confusing. For users with cognitive disabilities, the auto-focused element might grab attention before they've had a chance to understand the page structure. Additionally, autofocus can interfere with browser features like scroll restoration. If focus management is needed, implement it through Angular lifecycle hooks with proper context awareness rather than using the autofocus attribute.

However, there is an important exception: Using autofocus on elements within <dialog> elements (or on the dialog itself) is actually recommended for accessibility. According to MDN, the autofocus attribute should be added to the element the user is expected to interact with immediately upon opening a modal dialog. If no other element involves more immediate interaction, it is recommended to add autofocus to the close button inside the dialog, or the dialog itself. This rule automatically allows autofocus when used within dialog elements.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<button autofocus>Click me!</button>
        ~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input [attr.autofocus]="false">
       ~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<app-test [autofocus]="true"></app-test>
<select autofocus></select>
        ~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<input type="text">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<textarea autoFocus></textarea>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div [autoFocus]="true"></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<button [appautofocus]="false">Click me!</button>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<app-drag-drop autofocus></app-drag-drop>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<app-textarea [autofocus]="false"></app-textarea>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<dialog autofocus></dialog>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<dialog><button autofocus>Close</button></dialog>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<dialog><form><input autofocus type="text"></form></dialog>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<dialog [attr.autofocus]="true"></dialog>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<dialog><button [attr.autofocus]="true">Close</button></dialog>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-autofocus": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<dialog><div><button autofocus>Action</button></div></dialog>
```

</details>

<br>
