<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-autofocus.ts
  - ../../tests/rules/no-autofocus/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/template/no-autofocus`

Ensures that the `autofocus` attribute is not used

- Type: suggestion
- Category: Best Practices
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

‚ùå - Examples of **incorrect** code for this rule:

```html
<button autofocus>Click me!</button>
        ~~~~~~~~~
```

```html
<input [attr.autofocus]="false">
       ~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<app-test [autofocus]="true"></app-test>
<select autofocus></select>
        ~~~~~~~~~
```

<br>

---

<br>

‚úÖ - Examples of **correct** code for this rule:

```html
<input type="text">
```

```html
<textarea autoFocus></textarea>
```

```html
<div [autoFocus]="true"></div>
```

```html
<button [appautofocus]="false">Click me!</button>
```

```html
<app-drag-drop autofocus></app-drag-drop>
```

```html
<app-textarea [autofocus]="false"></app-textarea>
```
