<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-call-expression.ts
  - ../../tests/rules/no-call-expression/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/template/no-call-expression`

Disallows calling expressions in templates, except for output handlers

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```html
<div>{{ getInfo()() }}</div>
        ~~~~~~~~~~~
```

```html
<a href="{{ getUrls().user }}"></a>
            ~~~~~~~~~
```

```html
<p [test]="test?.getInfo()"></p>
           ~~~~~~~~~~~~~~~
```

```html
<a [href]="id && createUrl() && test()($any)">info</a>
                 ~~~~~~~~~~~    ~~~~~~~~~~~~
{{ id || obj?.nested1() }}
         ~~~~~~~~~~~~~~
```

```html
<a [href]="id ? a?.createUrl() : editUrl(3)">info</a>
                ~~~~~~~~~~~~~~   ~~~~~~~~~~
{{ 1 === 2 ? 3 : obj?.nested1()() }}
                 ~~~~~~~~~~~~~~~~
```

```html
{{ obj?.nested1() }} {{ obj!.nested1() }}
   ~~~~~~~~~~~~~~       ~~~~~~~~~~~~~~
<button [type]="obj!.$any(b)!.getType()()">info</button>
                ~~~~~~~~~~~~~~~~~~~~~~~~~
<a [href]="obj.propertyA?.href()">info</a>
           ~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```html
{{ info }}
```

```html
<button type="button" (click)="handleClick()">Click Here</button>
```

```html
{{ $any(info) }}
```

```html
<input (change)="obj?.changeHandler()">
```

```html
<form [formGroup]="form" (ngSubmit)="form.valid || save()"></form>
```

```html
<form [formGroup]="form" (ngSubmit)="form.valid && save()"></form>
```

```html
<form [formGroup]="form" (ngSubmit)="id ? save() : edit()"></form>
```
