<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-call-expression.ts
  - ../../tests/rules/no-call-expression/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/template/no-call-expression`

Disallows calling expressions in templates, except for output handlers

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```html
<div>{{ getInfo() }}</div>
        ~~~~~~~~~
```

```html
<a href="http://example.com">{{ getInfo().name }}</a>
                                ~~~~~~~~~
```

```html
<a [href]="getUrl()">info</a>
           ~~~~~~~~
```

```html
<a [href]="id && createUrl()">info</a>
                 ~~~~~~~~~~~
{{ id || obj?.nested1() }}
         ~~~~~~~~~~~~~~
```

```html
<a [href]="id ? a?.createUrl() : editUrl()">info</a>
                ~~~~~~~~~~~~~~   ~~~~~~~~~
{{ 1 === 2 ? 3 : obj?.nested1() }}
                 ~~~~~~~~~~~~~~
```

```html
{{ obj?.nested1() }} {{ obj!.nested1() }}
   ~~~~~~~~~~~~~~       ~~~~~~~~~~~~~~
<button [type]="obj!.$any(b)!.getType()">info</button>
                ~~~~~~~~~~~~~~~~~~~~~~~
<a [href]="obj.propertyA?.href()">info</a>
           ~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```html
{{ info }}
<button type="button" (click)="handleClick()">Click Here</button>
{{ $any(info) }}
<input (change)="obj?.changeHandler()">
<form [formGroup]="form" (ngSubmit)="form.valid || save()"></form>
<form [formGroup]="form" (ngSubmit)="form.valid && save()"></form>
<form [formGroup]="form" (ngSubmit)="id ? save() : edit()"></form>
```
