<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-distracting-elements.ts
  - ../../tests/rules/no-distracting-elements/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/no-distracting-elements`

Enforces that no distracting elements are used

- Type: suggestion
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-distracting-elements": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<marquee></marquee>{{ test }}
~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-distracting-elements": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div></div><blink></blink>
           ~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-distracting-elements": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div>Valid</div>
```

</details>

<br>
