<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-distracting-elements.ts
  - ../../tests/rules/no-distracting-elements/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/template/no-distracting-elements`

Enforces that no distracting elements are used

- Type: suggestion
- Category: Best Practices
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

‚ùå - Examples of **incorrect** code for this rule:

```html
<marquee></marquee>{{ test }}
~~~~~~~~~~~~~~~~~~~
```

```html
<div></div><blink></blink>
           ~~~~~~~~~~~~~~~
```

<br>

---

<br>

‚úÖ - Examples of **correct** code for this rule:

```html
<div>Valid</div>
```
