<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-duplicate-attributes.ts
  - ../../tests/rules/no-duplicate-attributes/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/no-duplicate-attributes`

Ensures that there are no duplicate input properties or output event listeners

- Type: problem

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * Whether or not two-way data binding is allowed as an exception to the rule.
   *
   * Default: `true`
   */
  allowTwoWayDataBinding?: boolean;
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input [name]="foo" [name]="bar">
       ~~~~~~~~~~~~ ~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input [name]="foo" name="bar">
       ~~~~~~~~~~~~ ~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input name="foo" name="bar">
       ~~~~~~~~~~ ~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input (change)="foo($event)" (change)="bar($event)">
       ~~~~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input [(ngModel)]="model" [(ngModel)]="otherModel">
       ~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input [name]="foo" [other]="bam" [name]="bar">
       ~~~~~~~~~~~~               ~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input [name]="foo" [name]="bar" [name]="bam">
       ~~~~~~~~~~~~ ~~~~~~~~~~~~ ~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input [(ngModel)]="model" [name]="foo" [(ngModel)]="otherModel" name="bar">
       ~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input (@fade.start)="animationStarted($event)" (@fade.start)="animationStarted($event)">
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input (window:resize)="windowResized($event)" (resize)="resize()" (window:resize)="windowResized2($event)">
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error",
      {
        "allowTwoWayDataBinding": false
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<input [(ngModel)]="model" (ngModelChange)="modelChanged()">
       ~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<input name="foo">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<input [name]="foo">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<input (change)="bar()">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<input [(ngModel)]="foo">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<input [(ngModel)]="model" (ngModelChange)="modelChanged()">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div (@fade.start)="animationStarted($event)" (@fade.done)="animationDone($event)"></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div (window:keydown)="windowKeydown($event)" (document:keydown)="documentKeydown($event)" (document:keyup)="documentKeyup($event)" (keyup)="keyup($event)" (keydown)="keydown($event)"></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div [style.width.px]="col.width" [width]="col.width"></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<button [class.disabled]="!enabled" [disabled]="!enabled"></button>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<button [@.disabled]="!enabled" [.disabled]="!enabled"></button>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-duplicate-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div [style.width]="col.width + 'px'" [width]="col.width"></div>
```

</details>

<br>
