<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-empty-control-flow.ts
  - ../../tests/rules/no-empty-control-flow/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/no-empty-control-flow`

Ensures that control flow blocks are not empty. Empty control flow blocks usually occur due to refactoring that wasn't completed. They can cause confusion when reading code.

- Type: suggestion

<br>

## Rationale

Empty control flow blocks (@if, @else, @for, @switch, @case, @empty) usually indicate incomplete refactoring or code that was commented out or deleted but left the block structure behind. These empty blocks add no functionality but create visual noise and confusion when reading templates. Developers may wonder whether the empty block is intentional or a bug. Removing empty blocks keeps templates clean and makes it clear that there is no conditional logic or looping in that location. This rule helps maintain template quality and catches likely mistakes during development.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@if (a) {} @else if (b) {
~~~~
  2
} @else {
  3
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@if (a) {
~~~~
} @else if (b) {
  2
} @else {
  3
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@if (a) {
  1
} @else if (b) {} @else {
  ~~~~~~~~~
  3
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@if (a) {
  1
} @else if (b) {
  ~~~~~~~~~
} @else {
  3
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@if (a) {
  1
} @else if (b) {
  2
} @else {}
  ~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@if (a) {
  1
} @else if (b) {
  2
} @else {
  ~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@for (item of items; track item.id) {} @empty {
~~~~~
  2
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@for (item of items; track item.id) {
~~~~~
} @empty {
  2
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@for (item of items; track item.id) {
  1
} @empty {}
  ~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@for (item of items; track item.id) {
  1
} @empty {
  ~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@switch (condition) {}
~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@switch (condition) {
~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@switch (condition) {
  @case (a) {}
  ~~~~~~
  @default {
    2
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@switch (condition) {
  @case (a) {
  ~~~~~~
  }
  @default {
    2
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@switch (condition) {
  @case (a) {
    1
  }
  @default {}
  ~~~~~~~~~
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@switch (condition) {
  @case (a) {
    1
  }
  @default {
  ~~~~~~~~~
  }
}
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
@if (a) {
    1
  } @else if (b) {
    2
  } @else {
    3
  }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
@if (a) {
    <!-- a -->
  } @else if (b) {
    <!-- b -->
  } @else {
    <!-- c -->
  }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
@if (a) {
     &nbsp;
  } @else if (b) {
     &nbsp;
  } @else {
     &nbsp;
  }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
@for (item of items; track item.id) {
    1
  } @empty {
    2
  }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
@for (item of items; track item.id) {
    <!-- a -->
  } @empty {
    <!-- b -->
  }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
@for (item of items; track item.id) {
    &nbsp;
  } @empty {
    &nbsp;
  }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
@switch (condition) {
    @case (a) {
      1
    }
    @default {
      2
    }
  }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
@switch (condition) {
    @case (a) {
      <!-- a -->
    }
    @default {
      <!-- b -->
    }
  }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-empty-control-flow": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
@switch (condition) {
    @case (a) {
      &nbsp;
    }
    @default {
      &nbsp;
    }
  }
```

</details>

<br>
