<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-inline-styles.ts
  - ../../tests/rules/no-inline-styles/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/no-inline-styles`

Disallows the use of inline styles in HTML templates

- Type: suggestion

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  allowNgStyle?: boolean;
  allowBindToStyle?: boolean;
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<ng-template>
  <div>
    <img style="position: relative;">
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  </div>
</ng-template>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<object style="padding: 30px;"></object>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<area style="padding: 30px;" />
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<input style="color: red;">
~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<input [attr.style]="'padding: 10px;'">
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<input [style.background-color]="'#fff'">
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error",
      {
        "allowNgStyle": true
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<input [style]="'border: 1px solid black;'" [ngStyle]="{ 'padding': '10px' }" [style.background-color]="'#fff'">
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<input [ngStyle]="'background-color: #fff'">
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error",
      {
        "allowNgStyle": true,
        "allowBindToStyle": false
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<input [style]="'border: 1px solid black;'">
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error",
      {
        "allowNgStyle": true,
        "allowBindToStyle": false
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<input [ngStyle]="{ 'padding': '10px' }" [style]="'border: 1px solid black;'">
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<img alt="Foo eating a sandwich.">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<img src="foo" [attr.alt]="altText">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<img src="foo" [attr.alt]="'Alt Text'">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<img src="foo" alt="">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<object aria-label="foo">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<object aria-labelledby="id1">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<object>Meaningful description</object>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<object title="An object">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<area aria-label="foo" />
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<area aria-labelledby="id1" />
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<area alt="This is descriptive!" />
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<input type="text">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<input type="image" alt="This is descriptive!">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<input type="image" aria-label="foo">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<input type="image" aria-labelledby="id1">
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error",
      {
        "allowNgStyle": true
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<input [ngStyle]="{ 'background-color': '#fff' }">
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error",
      {
        "allowBindToStyle": true
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<input [style.backgroundColor]="'#fff'">
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error",
      {
        "allowNgStyle": true,
        "allowBindToStyle": true
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<input [ngStyle]="{ 'background-color': 'red' }" [style.background-color]="'#fff'">
```

</details>

<br>
