<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-interpolation-in-attributes.ts
  - ../../tests/rules/no-interpolation-in-attributes/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/no-interpolation-in-attributes`

Ensures that property-binding is used instead of interpolation in attributes.

- Type: suggestion

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-interpolation-in-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<input type="text" name="{{ foo }}">
                         ~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-interpolation-in-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<input type="text" name="{{ foo }}bar">
                         ~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-interpolation-in-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<input type="text" [name]="foo">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-interpolation-in-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<input type="text" name="foo" [(ngModel)]="foo">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-interpolation-in-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<input type="text" [name]="foo + 'bar'">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-interpolation-in-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<input type="text" [name]="foo | bar">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-interpolation-in-attributes": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<div>{{ content }}</div>
```

</details>

<br>
