<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-negated-async.ts
  - ../../tests/rules/no-negated-async/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/no-negated-async`

Ensures that async pipe results are not negated

- Type: suggestion

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-negated-async": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{      !(foo | async) }}
        ~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-negated-async": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ !(foo | somethingElse | async) }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-negated-async": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div *ngIf="!(a | async)"></div>
            ~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-negated-async": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ nullable ?? !(obsVar | async) }}
               ~~~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-negated-async": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ (foo | async) }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-negated-async": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ !(foo | async | somethingElse) }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-negated-async": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ (foo | async) == null }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-negated-async": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ (foo | async) === false }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-negated-async": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ !(foo | notAnAsyncPipe) }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-negated-async": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div [class.mx-4]="!!(foo | async)"></div>
```

</details>

<br>
