<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-negated-async.ts
  - ../../tests/rules/no-negated-async/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/template/no-negated-async`

Ensures that async pipe results are not negated

- Type: suggestion
- Category: Best Practices

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

‚ùå - Examples of **incorrect** code for this rule:

```html
{{      !(foo | async) }}
        ~~~~~~~~~~~~~~
```

```html
{{ !(foo | somethingElse | async) }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<div *ngIf="!(a | async)"></div>
            ~~~~~~~~~~~~
```

```html
{{ nullable ?? !(obsVar | async) }}
               ~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

‚úÖ - Examples of **correct** code for this rule:

```html
{{ (foo | async) }}
```

```html
{{ !(foo | async | somethingElse) }}
```

```html
{{ (foo | async) == null }}
```

```html
{{ (foo | async) === false }}
```

```html
{{ !(foo | notAnAsyncPipe) }}
```

```html
<div [class.mx-4]="!!(foo | async)"></div>
```
