<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-non-null-assertion.ts
  - ../../tests/rules/no-non-null-assertion/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/no-non-null-assertion`

Disallows the non-null assertion operator (!) in templates

- Type: suggestion

<br>

## Rationale

Equivalent of @typescript-eslint/no-non-null-assertion rule for Angular templates. TypeScript's `!` non-null assertion operator asserts to the type system that an expression is non-nullable, as in not `null` or `undefined`. Using assertions to tell the type system new information is often a sign that code is not fully type-safe. It's generally better to structure program logic so that TypeScript understands when values may be nullable.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ foo! }}
   ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ foo!.bar }}
   ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ user.name! | uppercase }}
   ~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ user | uppercase : firstArg : secondArg! }}
                                 ~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ user.name | uppercase : (prefix! | lowercase) }}
                            ~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<div [title]="user!.name"></div>
              ~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@let user = a.b!.c;
            ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@let user = getUser()!;
            ~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ x.y({ param: a.b!.c }) }}
                ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ getValue()! }}
   ~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ foo.bar!.baz }}
   ~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ foo?.bar!.baz }}
   ~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<button (click)="handler(item!)">Click</button>
                         ~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<div *ngIf="foo!">Content</div>
            ~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@if (user!) { <div>Content</div> }
     ~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@if (user!.isActive) { <div>Active</div> }
     ~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@for (item of items!; track item.id) { <div>{{ item.name }}</div> }
              ~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@for (item of items; track item!.id) { <div>{{ item.name }}</div> }
                           ~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@switch (status!) { @case ('active') { <div>Active</div> } }
         ~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@switch (status) { @case (activeStatus!) { <div>Active</div> } }
                          ~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ user.getProfile()!.getName() }}
   ~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
@if ((data!.items?.length ?? 0) > 0) { <div>Has items</div> }
      ~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ user! ? user.name : 'Guest' }}
   ~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ items![0] }}
   ~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ items[0]! }}
   ~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ value! + 10 }}
   ~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ 10 + value! }}
        ~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ value! === 'test' }}
   ~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ !value! }}
    ~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ condition! ? 'yes' : 'no' }}
   ~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ condition ? value! : 'no' }}
               ~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ a + b?.c! - d == 42 }}
       ~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ a + b?.c! - d === 42 }}
       ~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ a + b?.c! - d < 42 }}
       ~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ a + b?.c! - d <= 42 }}
       ~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ a + b?.c! - d > 42 }}
       ~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ a + b?.c! - d >= 42 }}
       ~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ condition ?? value! }}
                ~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ condition ? 'yes' : value! }}
                       ~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ service!.getData() }}
   ~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<button (click)="handler(item!)">Click</button>
                         ~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ x[y]![z] }}
   ~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ x[y][z]! }}
   ~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ x.y.z!() }}
   ~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ x.y?.z!() }}
   ~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ foo }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ foo.bar }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ foo.bar.baz }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ foo?.bar }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ foo?.bar?.baz }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ user?.profile?.address?.city }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ foo ?? 'default' }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ (data?.items?.length ?? 0) > 0 }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ getValue() }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ foo?.getValue() }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ user?.getProfile()?.getName() }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ items[0] }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ items?.[0] }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ value | uppercase }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ user?.name | uppercase }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ value | pipe : arg }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<div [title]="user?.name"></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<button (click)="handler(item)">Click</button>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
@if (condition) { <div>Content</div> }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
@if (user?.isActive) { <div>Active</div> }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
@for (item of items; track item.id) { <div>{{ item.name }}</div> }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
@for (item of data?.items; track item.id) { <div>{{ item.name }}</div> }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
@switch (status) { @case ('active') { <div>Active</div> } }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
@switch (user?.status) { @case ('active') { <div>Active</div> } }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
@let userId = 42;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
@let user = getUser();
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<div *ngIf="condition">Content</div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ a + b }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ value + 10 }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ a === b }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ value === 'test' }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ !value }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ a && b }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ condition ? value : other }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-non-null-assertion": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ condition ? 'yes' : 'no' }}
```

</details>

<br>
