<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-positive-tabindex.ts
  - ../../tests/rules/no-positive-tabindex/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/no-positive-tabindex`

Ensures that the `tabindex` attribute is not positive

- Type: suggestion

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rationale

Using positive tabindex values (like tabindex="1" or tabindex="5") disrupts the natural tab order and creates a confusing navigation experience. The natural tab order follows the DOM structure, which users expect. Positive tabindex values create a complex focus order where lower numbers are focused first, then tabindex="0" elements, then remaining interactive elements. This makes the tab order unpredictable and hard to maintain. Users with motor disabilities who rely on keyboard navigation will struggle with illogical focus orders. Instead, use tabindex="0" to add elements to the natural tab order, tabindex="-1" to remove them, and structure your DOM in a logical order. This is a WCAG Level A requirement.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-positive-tabindex": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div tabindex="5"></div>
               ~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-positive-tabindex": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div [attr.tabindex]="21"></div>
                      ~~
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-positive-tabindex": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<span></span>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-positive-tabindex": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<span id="2"></span>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-positive-tabindex": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<span tabindex></span>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-positive-tabindex": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<span tabindex="-1"></span>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-positive-tabindex": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<span tabindex="0"></span>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-positive-tabindex": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<span [attr.tabindex]="-1"></span>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-positive-tabindex": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<span [attr.tabindex]="0"></span>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-positive-tabindex": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<span [attr.tabindex]="tabIndex"></span>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-positive-tabindex": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<span [attr.tabindex]="null"></span>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-positive-tabindex": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<span [attr.tabindex]="undefined"></span>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-positive-tabindex": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<app-test [tabindex]="1"></app-test>
```

</details>

<br>
