<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-restricted-syntax.ts
  - ../../tests/rules/no-restricted-syntax/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/template/no-restricted-syntax`

Disallows specified syntax.

- Type: suggestion
- Category: Stylistic Issues

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```html
<article>{{readSomethingPlease}}</article>
         ~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<button>Click me!</button>
~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```html
<button type="invalidType">Click me!</button>
        ~~~~~~~~~~~~~~~~~~
```

```html
<button [ngClass]="classes?.class1">Click me!</button>
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:
