<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/no-target-blank.ts
  - ../../tests/rules/no-target-blank/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/no-target-blank`

Disallows the use of target="\_blank" without rel="noreferrer" in HTML templates

- Type: suggestion

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-target-blank": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<ng-template>
  <div>
    <a target="_blank">link</a>
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~
  </div>
</ng-template>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-target-blank": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<ng-template>
  <div>
    <a target="_blank" rel="norefferer">link</a>
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  </div>
</ng-template>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-target-blank": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<ng-template>
  <div>
    <a [attr.target]="_blank" rel="noopener">link</a>
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  </div>
</ng-template>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-target-blank": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<ng-template>
  <div>
    <a target="_blank" [attr.rel]="noopener">link</a>
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  </div>
</ng-template>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-target-blank": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<ng-template>
  <div>
    <a [attr.target]="_blank" [attr.rel]="noopener">link</a>
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  </div>
</ng-template>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-target-blank": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<ng-template>
  <div>
    <a target="blank" rel="noreferrer">link</a>
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  </div>
</ng-template>
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-target-blank": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<a href="https://example.com" target="_blank" rel="noreferrer">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-target-blank": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<a href="https://example.com" [attr.target]="_blank" rel="noreferrer">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-target-blank": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<a href="https://example.com" target="_blank" [attr.rel]="noreferrer">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/no-target-blank": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<a href="https://example.com" [attr.target]="_blank" [attr.rel]="noreferrer">
```

</details>

<br>
