<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-static-string-properties.ts
  - ../../tests/rules/prefer-static-string-properties/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/prefer-static-string-properties`

Ensures that static string values use property assignment instead of property binding.

- Type: layout
- üîß Supports autofix (`--fix`)

<br>

## Rationale

When binding a static string literal to a property, using attribute syntax (property="value") is more efficient and simpler than property binding syntax ([property]="'value'"). Property binding with a static string creates unnecessary overhead because Angular evaluates it as an expression during change detection, even though the value never changes. Attribute syntax makes it immediately clear that the value is static and will never be updated. For example, [alt]="'Profile image'" should be alt="Profile image". This rule helps identify performance opportunities and makes templates more readable by distinguishing truly dynamic bindings from static values. The rule excludes animation bindings (@xxx), style/class sub-properties (style.color), and structural directives (\*ngIf) where property binding syntax is required.

The rule automatically excludes certain property-only DOM APIs (innerHTML, innerText, outerHTML, textContent) because these properties behave differently than their attribute counterparts. For example, textContent as a property does not HTML-encode its value, while textContent as an attribute does. Converting [textContent]="'<I>'" to textContent="<I>" would change the behavior and potentially break functionality. You can also configure additional properties to ignore using the ignore option.

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * An array of property names that should be ignored by this rule
   */
  ignore?: string[];
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<my-component [name]="'foo'"/>
              ~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<my-component [name]="   'foo'   "/>
              ~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<my-component [name]=' "foo" '/>
              ~~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component [name]="foo"/>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component [name]="42"/>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component [name]="true"/>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component [name]="null"/>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component [name]="undefined"/>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component name="foo"/>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component *name="'foo'"/>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<input *ngSwitchCase="'foo'" />
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-container *ngIf="'foo'" />
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component [style.color]="'foo'" />
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component [attr.aria-label]="'foo'" />
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component [class.foo]="'foo'" />
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component [@fade]="'foo'" />
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<code [textContent]="'<I>'"></code>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div [innerHTML]="'<strong>text</strong>'"></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<span [innerText]="'text'"></span>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div [outerHTML]="'<p>text</p>'"></div>
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error",
      {
        "ignore": [
          "customProp"
        ]
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component [customProp]="'value'"/>
```

</details>

<br>
