<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-at-empty.ts
  - ../../tests/rules/prefer-at-empty/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/prefer-at-empty`

Prefer using `@empty` with `@for` loops instead of a separate `@if` or `@else` block.

- Type: suggestion
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@for (item of items; track $index) {}
@if (items.length === 0) {
~~~~
  No items
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@for (item of items; track $index) {}
@if (items.length == 0) {
~~~~
  No items
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@for (item of items; track $index) {}
@if (0 === items.length) {
~~~~
  No items
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@for (item of items; track $index) {}
@if (0 == items.length) {
~~~~
  No items
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@for (item of items; track $index) {}
@if (!items.length) {
~~~~
  No items
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (items.length === 0) {
~~~~
  No items
}
@for (item of items; track $index) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (items.length == 0) {
~~~~
  No items
}
@for (item of items; track $index) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (0 === items.length) {
~~~~
  No items
}
@for (item of items; track $index) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (0 == items.length) {
~~~~
  No items
}
@for (item of items; track $index) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (!items.length) {
~~~~
  No items
}
@for (item of items; track $index) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (items.length > 0) {
~~~~
  @for (item of items; track $index) {}
} @else {
  No items
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (items.length !== 0) {
~~~~
  @for (item of items; track $index) {}
} @else {
  No items
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (items.length != 0) {
~~~~
  @for (item of items; track $index) {}
} @else {
  No items
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (0 < items.length) {
~~~~
  @for (item of items; track $index) {}
} @else {
  No items
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (0 !== items.length) {
~~~~
  @for (item of items; track $index) {}
} @else {
  No items
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (0 != items.length) {
~~~~
  @for (item of items; track $index) {}
} @else {
  No items
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (items.length) {
~~~~
  @for (item of items; track $index) {}
} @else {
  No items
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (items.length === 0) {
~~~~
  No items
} @else {
  @for (item of items; track $index) {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (items.length == 0) {
~~~~
  No items
} @else {
  @for (item of items; track $index) {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (0 === items.length) {
~~~~
  No items
} @else {
  @for (item of items; track $index) {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (0 == items.length) {
~~~~
  No items
} @else {
  @for (item of items; track $index) {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (!items.length) {
~~~~
  No items
} @else {
  @for (item of items; track $index) {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (items.length > 0) {
~~~~
  @for (item of items; track $index) {}
}
@if (items.length === 0) {
  No items
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (items.length === 0) {
  No items
}
@if (items.length > 0) {
~~~~
  @for (item of items; track $index) {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (items.length > 0) {
~~~~
  @for (item of items; track $index) {}
} @else if (condition) {
 Condition is true
} @else {
 Condition is false
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@for (item of items; track $index) {}
<!-- comment -->
@if (items.length === 0) {
~~~~
  No items
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<!-- The Angular compiler will ignore the space between the two blocks. -->
@for (item of items; track $index) {}
&nbsp;
@if (items.length === 0) {
~~~~
  Not empty
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (items.length === 0) {
~~~~
  Not empty
}
@for (item of items; track $index) {}
@empty {
  Existing
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@for (item of items; track $index) {}
@empty {
  Existing
}
@if (items.length === 0) {
~~~~
  Not empty
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (items.length > 0) {
~~~~
  @for (item of items; track $index) {}
  @empty {
    Existing
  }
} @else {
  Empty
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (items.length === 0) {
~~~~
  Empty
} @else {
  @for (item of items; track $index) {}
  @empty {
    Existing
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (items.length > 0) {
~~~~
  <!-- before -->
  @for (item of items; track $index) {}
  <!-- after -->
} @else {
  Empty
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (items.length === 0) {
~~~~
  Empty
} @else {
  <!-- before -->
  @for (item of items; track $index) {}
  <!-- after -->
}
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track $index) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track $index) {} @empty {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (alpha.length > 0) {
  @for (item of beta; track $index) {}
} @else {
  No items
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (items.length > 0) {
  Items:
  @for (item of items; track $index) {}
} @else {
  No items
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (items.length === 0) {
  No items
} @else {
  Items:
  @for (item of items; track $index) {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (items.length > 0) {
  @for (item of items; track $index) {}
  Total: {{ $count }}
} @else {
  No items
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (items.length === 0) {
  No items
} @else {
  @for (item of items; track $index) {}
  Total: {{ $count }}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (items.length > 0) {
  @for (item of items; track $index) {}
}
@if (items.length > 0) {
  Not empty
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (items.length > 0) {
  @for (item of items; track $index) {}
}
---
@if (items.length === 0) {
  Not empty
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track $index) {}
---
@if (items.length === 0) {
  Not empty
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track $index) {}
@if (items.length > 0) {
  Not empty
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (items.length > 0) {
  Not empty
}
@for (item of items; track $index) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-empty": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (beta.length > 0) {
  @for (item of alpha; track $index) {}
  @for (item of beta; track $index) {}
} @else {
  Empty
}
```

</details>

<br>
