<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-self-closing-tags.ts
  - ../../tests/rules/prefer-self-closing-tags/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/prefer-self-closing-tags`

Ensures that self-closing tags are used for elements with a closing tag but no content.

- Type: layout
- üîß Supports autofix (`--fix`)

<br>

## Rationale

Self-closing tags like <app-component /> are more concise and eliminate visual clutter when elements have no content. This pattern is familiar from HTML void elements (like <img /> and <br />) and from JSX/React. Using self-closing syntax makes it immediately obvious that an element is empty, whereas <app-component></app-component> requires scanning to the closing tag to confirm there is no content. The shorter syntax also reduces the chance of accidentally nesting content where none was intended. Angular v15.1+ supports self-closing tags for components and structural elements. The rule automatically skips native HTML elements (div, span, etc.) and index.html files since browser support varies. For Angular components and directives, self-closing tags improve template readability.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<my-component></my-component>
              ~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<my-component *ngIf="condition" type="text" [name]="foo"></my-component>
                                                         ~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<my-component
  type="text"
  [name]="foo"
  [items]="items">
</my-component>
~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<my-component
  type="text"
  [name]="foo"
  [items]="items"
></my-component>
 ~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<ng-template></ng-template>
             ~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<ng-template> </ng-template>
              ~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<ng-content></ng-content>
            ~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<ng-content
  selector="my-selector"
></ng-content>
 ~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<ng-content>
</ng-content>
~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<ng-content select="foo>bar">
</ng-content>
~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component type="text" [name]="foo">With some content</my-component>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component />
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component
  *ngIf="condition"
  type="text"
  [name]="foo"
  [items]="items" />
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<img />
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<img src="foo" *ngIf="condition" />
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<slot></slot><math></math><rb></rb><svg></svg><template></template>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div *ngIf="condition"></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<th scope="col"></th>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<th *ngIf="condition" scope="col"></th>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-template/>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-template>Content</ng-template>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-content/>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-content select="my-selector" />
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-content>Fallback content</ng-content>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-content>&nbsp;</ng-content>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-content> <!-- comment --> </ng-content>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-content
     select="content"
   >
    <p>Fallback content</p>
  </ng-content>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-content select="[slot='icon-only']">
    <ng-content select="[slot=text]" />
  </ng-content>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-content select="[slot='foo>bar']" />
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-content select="[slot='foo>bar']">Fallback</ng-content>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

**Filename: src/index.html**

```html
<app-root></app-root>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-container>&nbsp;</ng-container>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component>  <!-- not empty -->  </my-component>
```

</details>

<br>
