<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-built-in-pipes.ts
  - ../../tests/rules/prefer-built-in-pipes/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/prefer-built-in-pipes`

Encourages the use of Angular built-in pipes (e.g. lowercase, uppercase, titlecase) instead of certain JavaScript methods in Angular templates.

- Type: suggestion

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * Additional method names to disallow (defaults include common case/title casing helpers)
   *
   * Default: `["toLowerCase","toUpperCase","toLocaleLowerCase","toLocaleUpperCase"]`
   */
  disallowList?: string[];
  /**
   * Whether to allow these method calls inside output event handlers
   *
   * Default: `true`
   */
  allowInOutputHandlers?: boolean;
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-built-in-pipes": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ name.toLowerCase() }}
   ~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-built-in-pipes": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ title.toUpperCase() }}
   ~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-built-in-pipes": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ text.toLocaleLowerCase() }}
   ~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-built-in-pipes": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ text.toLocaleUpperCase() }}
   ~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-built-in-pipes": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<div [title]="name.toLowerCase()">Content</div>
              ~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-built-in-pipes": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<div title="{{ name.toUpperCase() }}">Content</div>
               ~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-built-in-pipes": [
      "error",
      {
        "allowInOutputHandlers": false
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<button (click)="handleClick(name.toLowerCase())">Click</button>
                             ~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-built-in-pipes": [
      "error",
      {
        "disallowList": [
          "toUpperCase",
          "toLowerCase"
        ]
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ name.toUpperCase() }}
   ~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-built-in-pipes": [
      "error",
      {
        "disallowList": [
          "getFullYear"
        ]
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
{{ date.getFullYear() }}
   ~~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-built-in-pipes": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ name }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-built-in-pipes": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ user.name }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-built-in-pipes": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<button (click)="handleClick(name.toLowerCase())">Click</button>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-built-in-pipes": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<button (change)="onChange(value.toUpperCase())">Change</button>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-built-in-pipes": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ name | lowercase }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-built-in-pipes": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ name | uppercase }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-built-in-pipes": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ title | titlecase }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-built-in-pipes": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ getValue() }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-built-in-pipes": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ items.push(item) }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-built-in-pipes": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ text.includes("test") }}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-built-in-pipes": [
      "error",
      {
        "disallowList": [
          "toUpperCase"
        ]
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
{{ name.toLowerCase() }}
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-built-in-pipes": [
      "error",
      {
        "allowInOutputHandlers": false
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<button (click)="handleClick()">Click</button>
```

</details>

<br>
