<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-class-binding.ts
  - ../../tests/rules/prefer-class-binding/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/prefer-class-binding`

Suggests using [class] bindings over ngClass where applicable

- Type: suggestion

<br>

## Rationale

For simple cases, [class] bindings offer a more straightforward syntax with better performance than ngClass. However, ngClass should still be used when you need: (1) space-separated class names in a single key, or (2) mutations on objects, as class bindings do not support these use cases (the reference must change for class bindings to detect updates). See https://angular.dev/guide/templates/binding#css-class-and-style-property-bindings for more information. This rule helps identify potential simplification opportunities but should be applied judiciously based on your specific needs.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-class-binding": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<div [ngClass]="className"></div>
     ~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-class-binding": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<div [ngClass]="getClasses()"></div>
     ~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-class-binding": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<button [ngClass]="isActive ? 'active' : 'inactive'"></button>
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-class-binding": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<div class="static" [ngClass]="dynamicClass"></div>
                    ~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-class-binding": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<input type="text" [ngClass]="inputClass" [value]="text">
                   ~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-class-binding": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<div [ngClass]="['class1', 'class2']"></div>
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-class-binding": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<div [ngClass]="{ active: true, disabled: false }"></div>
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-class-binding": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<div [class.active]="isActive"></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-class-binding": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<div class="static-class"></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-class-binding": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<div class="static" [class.active]="isActive"></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-class-binding": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<div [class.active]="isActive" [class.disabled]="isDisabled"></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-class-binding": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<button [class.active]="isActive && !isDisabled">Click</button>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-class-binding": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<input type="text" [class.error]="hasError" [value]="inputValue">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-class-binding": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<div [ngClass]="{'btn btn-primary': isPrimary}"></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-class-binding": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<div [ngClass]="{'alert alert-danger': hasError, 'alert-success': !hasError}"></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-class-binding": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<button [ngClass]="{'btn btn-lg btn-block': isLarge}">Submit</button>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-class-binding": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<div [ngClass]="{'btn btn-primary': isPrimary, 'active': isActive}"></div>
```

</details>

<br>
