<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-ngsrc.ts
  - ../../tests/rules/prefer-ngsrc/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/prefer-ngsrc`

Ensures ngSrc is used instead of src for img elements

- Type: suggestion

- üí° Provides suggestions on how to fix issues (https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

<br>

## Rationale

The ngSrc directive (part of Angular's Image directive introduced in v15) provides significant performance and user experience benefits over the standard src attribute for images. Using ngSrc enables automatic lazy loading, responsive images with srcset generation, optimized loading priority hints, prevention of layout shift by enforcing width and height attributes, and automatic image optimization when using an image loader. These features dramatically improve Largest Contentful Paint (LCP) and other Core Web Vitals metrics. The directive also provides warnings for common mistakes like missing width/height. The rule allows data: URIs with src since those are inline base64 images that don't benefit from ngSrc optimizations. For best performance and user experience, all external images should use ngSrc.

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img src="http://localhost">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img src="http://src.com">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img [src]="'http://localhost'">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img [src]="value">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img [attr.src]="value">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img [attr.src]="'http://localhost'">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img [src]="'http://' + value">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img ngSrc="http://localhost" src="http://localhost">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img ngSrc="http://src.com" src="http://src.com">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img ngSrc="http://localhost" [src]="'http://localhost'">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img ngSrc="http://localhost" [src]="value">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img [ngSrc]="otherValue" src="http://localhost">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img [ngSrc]="otherValue" [src]="'http://localhost'">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img [ngSrc]="otherValue" [src]="value">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img [src]="otherValue" [ngSrc]="value">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img src="data:image/png;base64" [ngSrc]="otherValue">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img [attr.src]="'data:image/png;base64'" [ngSrc]="otherValue">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img [src]="'data:image/png;base64'" [ngSrc]="otherValue">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img [src]="'data:' + value" [ngSrc]="otherValue">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img
      src="http://localhost">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img
      [src]="value"
      alt="test">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img
      [attr.src]="value"
      width="100">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img
      ngSrc="http://localhost"
      src="http://localhost">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img
      [ngSrc]="otherValue"
      [src]="value"
      alt="test">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<img alt="test"
      [src]="value"
    >
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<img alt="nothing">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<img ngSrc="http://localhost">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<img [ngSrc]="'http://localhost'">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<img [ngSrc]="value">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<img src="data:image/jpeg;base64">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<img [src]="'data:image/jpeg;base64'">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<img [src]="'data:' + value">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<img [attr.src]="'data:image/jpeg;base64'">
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-ngsrc": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<img [attr.src]="'data:' + value">
```

</details>

<br>
