<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-contextual-for-variables.ts
  - ../../tests/rules/prefer-contextual-for-variables/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/prefer-contextual-for-variables`

Ensures that contextual variables are used in @for blocks where possible instead of aliasing them.

- Type: suggestion
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * Default: `{"$count":[],"$index":[],"$first":[],"$last":[],"$even":[],"$odd":[]}`
   */
  allowedAliases?: {
    /**
     * Aliases for $count that are allowed to be used.
     */
    $count?: string[];
    /**
     * Aliases for $index that are allowed to be used.
     */
    $index?: string[];
    /**
     * Aliases for $first that are allowed to be used.
     */
    $first?: string[];
    /**
     * Aliases for $last that are allowed to be used.
     */
    $last?: string[];
    /**
     * Aliases for $even that are allowed to be used.
     */
    $even?: string[];
    /**
     * Aliases for $odd that are allowed to be used.
     */
    $odd?: string[];
    [k: string]: unknown;
  };
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error",
      {
        "allowedAliases": {
          "$index": [
            "a"
          ],
          "$count": [
            "b"
          ]
        }
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@for (item of items; track item.id; let x = $even, a = $index, b = $count) {,}
                                        ~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error",
      {
        "allowedAliases": {
          "$index": [
            "a"
          ],
          "$count": [
            "b"
          ]
        }
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@for (item of items; track item.id; let a = $index, x = $even, b = $count) {,}
                                                    ~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error",
      {
        "allowedAliases": {
          "$index": [
            "a"
          ],
          "$count": [
            "b"
          ]
        }
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@for (item of items; track item.id; let a = $index, b = $count, x = $even) {,}
                                                                ~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error",
      {
        "allowedAliases": {
          "$count": [
            "b"
          ]
        }
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@for (item of items; track item.id; let a = $index, b = $count, x = $even) {,}
                                        ~~~~~~~~~~              ~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error",
      {
        "allowedAliases": {
          "$count": [
            "b"
          ]
        }
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@for (item of items; track item.id; let a = $index, x = $even, b = $count) {,}
                                        ~~~~~~~~~~  ~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@for (item of items; track item.id; let a = $index, b = $even, c = $count) {,}
                                        ~~~~~~~~~~  ~~~~~~~~~  ~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@for (item of items; track item.id) {
  @if (!!($index % 2)) {
        ~~~~~~~~~~~~~
    is odd
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@for (a of outer; track a.id) {
  @if ($index === 0) {
       ~~~~~~~~~~~~
    first outer
  }
  @for (b of inner; track b.id) {
    @if ($index === 0) {
         ~~~~~~~~~~~~
      first inner
    }
  }
}
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track $index) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {}
{{ items.length }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (a of outer; track a.id) {
  @for (b of inner; track b.id) {
    {{ outer.length }}
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (a of outer(); track a.id) {
  @for (b of inner(); track b.id) {
    {{ outer().length }}
  }
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {
  {{ this.$index === 0 }}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {
  {{ $index === this.$count - 1 }}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {}
{{ $index % 2 === 0 }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {}
{{ $index % 2 !== 0 }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {}
{{ !$even }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {}
{{ !$odd }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {
  @if (this.$index % 2 === 0) {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {
  @if (this.$index % 2 !== 0) {}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {
  {{ !this.$even }}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {
  {{ !this.$odd }}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {
  {{ $index % 2 }}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {
  {{ $index % 2 === foo }}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {
  {{ $index % 2 !== foo }}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {
  {{ $index % 2 >= foo }}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {
  {{ $index % 2 > foo }}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {
  {{ $index % 2 <= foo }}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {
  {{ $index % 2 < foo }}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {
  {{ $index % 2 + foo }}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {
  {{ $index % 2 - foo }}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {
  {{ $index % 2 * foo }}
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-contextual-for-variables": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@for (item of items; track item.id) {
  {{ $index % 2 / foo }}
}
```

</details>

<br>
