<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-at-else.ts
  - ../../tests/rules/prefer-at-else/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/prefer-at-else`

Prefer using `@else` instead of a second `@if` with the opposite condition to reduce code and make it easier to read.

- Type: suggestion
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a) {}
@if (!a) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (!a) {}
@if (a) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a > b) {}
@if (a <= b) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a <= b) {}
@if (a > b) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a >= b) {}
@if (a < b) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a < b) {}
@if (a >= b) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a === b) {}
@if (a !== b) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a !== b) {}
@if (a === b) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a == b) {}
@if (a != b) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a != b) {}
@if (a == b) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a.length > 0) {}
@if (a.length === 0) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a.length === 0) {}
@if (a.length > 0) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (0 < a.length) {}
@if (a.length === 0) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a.length === 0) {}
@if (0 < a.length) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (0 < a.length) {}
@if (0 === a.length) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (0 === a.length) {}
@if (0 < a.length) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a.length > 0) {}
@if (0 === a.length) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (0 === a.length) {}
@if (a.length > 0) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a.length > 0) {}
@if (a.length == 0) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a.length == 0) {}
@if (a.length > 0) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a) {
  1
} @else {
  2
}
@if (!a) {
~~~~
  3
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a) {
  1
}
@if (!a) {
~~~~
  2
} @else {
  3
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a) {
  1
} @else {
  2
}
@if (!a) {
~~~~
  3
} @else {
  4
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a) {}

@if (!a) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a) {}
&nbsp;
@if (!a) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a) {}
<!-- comment -->
@if (!a) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a; as alias) {}
@if (!a) {}
~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (a) {}
@if (!a; as alias) {}
~~~~
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (a) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (a) {
} @else {
}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (a) {}
@if (b) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (a) {}
@if (!b) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (a.b) {}
@if (!c.b) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (a === true) {}
@if (a === false) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (a === b) {}
@if (a != b) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (a > 0) {}
@if (a < 0) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (a > 1) {}
@if (a <= 0) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (a >= b) {}
@if (a <= b) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (a.length > 1) {}
@if (a.length <= 0) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (a) {}
text
@if (!a) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (a) {}
<element/>
@if (!a) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (a) {}
@if (b) {}
@if (!a) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-at-else": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (a) {}
@else if (b) {}
@if (!a) {}
```

</details>

<br>
