<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-self-closing-tags.ts
  - ../../tests/rules/prefer-self-closing-tags/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/prefer-self-closing-tags`

Ensures that self-closing tags are used for elements with a closing tag but no content.

- Type: layout
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<my-component></my-component>
              ~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<my-component type="text" [name]="foo"></my-component>
                                       ~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<my-component
  type="text"
  [name]="foo"
  [items]="items">
</my-component>
~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<my-component
  type="text"
  [name]="foo"
  [items]="items"
></my-component>
 ~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<ng-template></ng-template>
             ~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<ng-template> </ng-template>
              ~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<ng-content></ng-content>
            ~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<ng-content
  selector="my-selector"
></ng-content>
 ~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<ng-content>
</ng-content>
~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component type="text" [name]="foo">With some content</my-component>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component />
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component
  type="text"
  [name]="foo"
  [items]="items" />
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<img />
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-template/>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-template>Content</ng-template>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-content/>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<ng-content select="my-selector" />
```

</details>

<br>
