<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-static-string-properties.ts
  - ../../tests/rules/prefer-static-string-properties/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/prefer-static-string-properties`

Ensures that static string values use property assignment instead of property binding.

- Type: layout
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<my-component [name]="'foo'"/>
              ~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<my-component [name]="   'foo'   "/>
              ~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<my-component [name]=' "foo" '/>
              ~~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component [name]="foo"/>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component [name]="42"/>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component [name]="true"/>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component [name]="null"/>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component [name]="undefined"/>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component name="foo"/>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-static-string-properties": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component *name="'foo'"/>
```

</details>

<br>
