<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/prefer-template-literal.ts
  - ../../tests/rules/prefer-template-literal/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/prefer-template-literal`

Ensure that template literals are used instead of concatenating strings or expressions.

- Type: suggestion
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ 'pre"fix-' + '-suf\'fix' }}
   ~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ "pre'fix-" + "-suf\"fix" }}
   ~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ `prefix-${value}-suffix` + `-prefix2-${value2}-suffix2` }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ 'pre"fix-' + "-suf'fix" }}
   ~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ 'pre`fix-' + `'pre\`fix"-${value}-"suf\`fix'` }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ "pre'fix-" + '-suf"fix' }}
   ~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ "pre`fix-" + `'pre\`fix"-${value}-"suf\`fix'` }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<my-component [class]="'prefix-' + myClass | pipe"></my-component>
                       ~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@if (value() + "-suffix" | pipe) {}
     ~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@defer (when value() + '-suffix' | pipe) {}
             ~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
@let letValue = value() + '-suffix';
                ~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ 'prefix-' + 42 }}
   ~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ 'prefix-' + null }}
   ~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ 'prefix-' + undefined }}
   ~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ 'prefix-' + true }}
   ~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ 'prefix-' + value }}
   ~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ 'prefix-' + value() }}
   ~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ 'prefix-' + [42] }}
   ~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ 'prefix-' + (condition ? 'true' : 'false') }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ 'prefix-' + ('value' | pipe) }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ "prefix-" + 42 }}
   ~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ "prefix-" + null }}
   ~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ "prefix-" + undefined }}
   ~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ "prefix-" + true }}
   ~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ "prefix-" + value }}
   ~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ "prefix-" + value() }}
   ~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ "prefix-" + [42] }}
   ~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ 'prefix-' + (condition ? 'true' : 'false') }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ 'prefix-' + ('value' | pipe) }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ `prefix-${value}-suffix` + 42 }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ `prefix-${value}-suffix` + null }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ `prefix-${value}-suffix` + undefined }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ `prefix-${value}-suffix` + false }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ `prefix-${value}-suffix` + value2 }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ `prefix-${value}-suffix` + value2() }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ `prefix-${value}-suffix` + [42] }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ `prefix-${value}-suffix` + (condition ? 'true' : 'false') }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ `prefix-${value}-suffix` + ('value' | pipe) }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ 42 + '-suffix' }}
   ~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ null + '-suffix' }}
   ~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ undefined + '-suffix' }}
   ~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ true + '-suffix' }}
   ~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ value + '-suffix' }}
   ~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ value() + '-suffix' }}
   ~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ [42] + '-suffix' }}
   ~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ `'pre\`fix"-${value}-"suf\`fix'` + '-suf`fix' }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ (condition ? 'true' : 'false') + '-suffix' }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ ('value' | pipe) + '-suffix' }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ 42 + "-suffix" }}
   ~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ null + "-suffix" }}
   ~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ undefined + "-suffix" }}
   ~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ true + "-suffix" }}
   ~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ value + "-suffix" }}
   ~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ value() + "-suffix" }}
   ~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ [42] + "-suffix" }}
   ~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ `'pre\`fix"-${value}-"suf\`fix'` + "-suf`fix" }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ (condition ? 'true' : 'false') + "-suffix" }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ ('value' | pipe) + "-suffix" }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ 42 + `prefix-${value}-suffix` }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ null + `prefix-${value}-suffix` }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ undefined + `prefix-${value}-suffix` }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ false + `prefix-${value}-suffix` }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ value2 + `prefix-${value}-suffix` }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ value2() + `prefix-${value}-suffix` }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ [42] + `prefix-${value}-suffix` }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ (condition ? 'true' : 'false') + `prefix-${value}-suffix` }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
{{ ('value' | pipe) + `prefix-${value}-suffix` }}
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<a [href]="'https://example.com/very-long-url-path-that-is-quite-long' + variable">Test</a>
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<a [href]="'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx' + example">Test</a>
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ `prefix-${value}-suffix` }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ 42 + 42 }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ value + value2 }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ value() + value2() }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ 'simple-quote' | pipe }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ "double-quote" }}"
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
{{ `backquote` }}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@if (`prefix-${value}-suffix`) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@defer (when `prefix-${value}-suffix`) {}
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
@let letValue = `prefix-${value}-suffix`;
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<h1>{{ `prefix-${value}-suffix` }}</h1>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component class="prefix-{{value}}-suffix"></my-component>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component [class]="`prefix-${value}-suffix`"></my-component>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/prefer-template-literal": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component *directive="`prefix-${value}-suffix` | pipe" />
```

</details>

<br>
