<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/use-track-by-function.ts
  - ../../tests/rules/use-track-by-function/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

# `@angular-eslint/template/use-track-by-function`

Ensures trackBy function is used

- Type: suggestion
- Category: Best Practices

<br>

## Rule Options

The rule does not have any configuration options.

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

❌ - Examples of **incorrect** code for this rule:

```html
<ul>
  <li *ngFor="let item of [1, 2, 3];">
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    {{ item }}
  </li>
</ul>
```

```html
<ng-template ngFor let-item [ngForOf]="[1, 2, 3]" let-i="index">
                            ~~~~~~~~~~~~~~~~~~~~~
  {{ item }}
</ng-template>
```

```html
<div *ngFor="let item of [1, 2, 3]; trackBy: trackByFn">
  {{ item }}
</div>
<ul>
  <li *ngFor="let item of [1, 2, 3];">
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    {{ item }}
  </li>
</ul>
```

```html
<div *ngFor="let item of [1, 2, 3];">
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  {{ item }}
</div>
<ng-template ngFor let-item [ngForOf]="[1, 2, 3]" let-i="index">
                            ^^^^^^^^^^^^^^^^^^^^^
  {{ item }}
</ng-template>
```

<br>

---

<br>

✅ - Examples of **correct** code for this rule:

```html
<div *ngFor="let item of [1, 2, 3]; trackBy: trackByFn">
  {{ item }}
</div>
```

```html
<div *ngFor="let item of [1, 2, 3]; let i = index; trackBy: trackByFn">
  {{ item }}
</div>
```

```html
<div *ngFor="let item of [1, 2, 3]; trackBy : trackByFn">
  {{ item }}
</div>
```

```html
<div *ngFor='let item of [1, 2, 3]; let i = index; trackBy: trackByFn'>
  {{ item }}
</div>
```

```html
<div *ngFor  =  "let item of [1, 2, 3]; let i = index; trackBy : trackByFn">
  {{ item }}
</div>
```

```html
<ng-template ngFor let-item [ngForOf]="[1, 2, 3]" let-i="index"
  [ngForTrackBy]="trackByFn">
  {{ item }}
</ng-template>
```

```html
<div *ngFor="let item of ['a', 'b', 'c']; index as i; trackBy: trackByFn">
  {{ item }}
</div>

<ng-template ngFor let-item [ngForOf]="[1, 2, 3]" let-i="index"
  [ngForTrackBy]="trackByFn">
  {{ item }}
</ng-template>
```

```html
<div *ngFor="
  let item of [1, 2, 3];
  let i = index;
  trackBy : trackByFn
">
```
