<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/use-track-by-function.ts
  - ../../tests/rules/use-track-by-function/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/use-track-by-function`

Ensures trackBy function is used

- Type: suggestion

<br>

## Rationale

When using \*ngFor to render lists, Angular needs to track which items are added, removed, or moved to efficiently update the DOM. Without a trackBy function, Angular tracks items by object identity, meaning any change to the array (like sorting, filtering, or fetching fresh data from an API) causes Angular to destroy and recreate all DOM elements, even for items that haven't changed. This is inefficient and causes problems like: (1) poor performance with large lists, (2) loss of component state (like form input values), and (3) loss of CSS animations or focus. A trackBy function tells Angular how to identify unique items (usually by an id property), allowing Angular to reuse DOM elements for unchanged items. Use trackBy for all lists, especially those that update frequently or contain components with state.

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * Default: `[]`
   */
  alias?: string[];
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>❌ - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/use-track-by-function": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<ul>
  <li *ngFor="let item of [1, 2, 3];">
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    {{ item }}
  </li>
</ul>
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/use-track-by-function": [
      "error",
      {
        "alias": [
          "ngForTrackByProperty"
        ]
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<ul>
  <li *ngFor="let item of [1, 2, 3];">
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    {{ item }}
  </li>
</ul>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/use-track-by-function": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<ng-template ngFor let-item [ngForOf]="[1, 2, 3]" let-i="index">
                            ~~~~~~~~~~~~~~~~~~~~~
  {{ item }}
</ng-template>
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/use-track-by-function": [
      "error",
      {
        "alias": [
          "ngForTrackByProperty"
        ]
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<ng-template ngFor let-item [ngForOf]="[1, 2, 3]" let-i="index">
                            ~~~~~~~~~~~~~~~~~~~~~
  {{ item }}
</ng-template>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/use-track-by-function": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<div *ngFor="let item of [1, 2, 3]; trackBy: trackByFn">
  {{ item }}
</div>
<ul>
  <li *ngFor="let item of [1, 2, 3];">
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    {{ item }}
  </li>
</ul>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/use-track-by-function": [
      "error"
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<div *ngFor="let item of [1, 2, 3];">
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  {{ item }}
</div>
<ng-template ngFor let-item [ngForOf]="[1, 2, 3]" let-i="index">
                            ~~~~~~~~~~~~~~~~~~~~~
  {{ item }}
</ng-template>
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/use-track-by-function": [
      "error",
      {
        "alias": [
          "ngForTrackByProperty"
        ]
      }
    ]
  }
}
```

<br>

#### ❌ Invalid Code

```html
<div *ngFor="let item of [1, 2, 3]; trackBy: trackByFn">
  {{ item }}
</div>
<div *ngFor="let item of [1, 2, 3]; trackByProperty: trackByFn">
  {{ item }}
</div>
<ul>
  <li *ngFor="let item of [1, 2, 3];">
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    {{ item }}
  </li>
</ul>
```

</details>

<br>

---

<br>

<details>
<summary>✅ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/use-track-by-function": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<div *ngFor="let item of [1, 2, 3]; trackBy: trackByFn">
  {{ item }}
</div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/use-track-by-function": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<div *ngFor="let item of [1, 2, 3]; let i = index; trackBy: trackByFn">
  {{ item }}
</div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/use-track-by-function": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<div *ngFor="let item of [1, 2, 3]; trackBy : trackByFn">
  {{ item }}
</div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/use-track-by-function": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<div *ngFor='let item of [1, 2, 3]; let i = index; trackBy: trackByFn'>
  {{ item }}
</div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/use-track-by-function": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<div *ngFor  =  "let item of [1, 2, 3]; let i = index; trackBy : trackByFn">
  {{ item }}
</div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/use-track-by-function": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<ng-template ngFor let-item [ngForOf]="[1, 2, 3]" let-i="index"
  [ngForTrackBy]="trackByFn">
  {{ item }}
</ng-template>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/use-track-by-function": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<div *ngFor="let item of ['a', 'b', 'c']; index as i; trackBy: trackByFn">
  {{ item }}
</div>

<ng-template ngFor let-item [ngForOf]="[1, 2, 3]" let-i="index"
  [ngForTrackBy]="trackByFn">
  {{ item }}
</ng-template>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/use-track-by-function": [
      "error"
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<div *ngFor="
  let item of [1, 2, 3];
  let i = index;
  trackBy : trackByFn
">
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/use-track-by-function": [
      "error",
      {
        "alias": [
          "ngForTrackByProperty"
        ]
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<div *ngFor="
  let item of [1, 2, 3];
  let i = index;
  trackByProperty: 'id'
">
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/use-track-by-function": [
      "error",
      {
        "alias": [
          "ngForTrackById"
        ]
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<div *ngFor="let photo of photos; trackById"></div>
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/use-track-by-function": [
      "error",
      {
        "alias": [
          "ngForTrackByProperty"
        ]
      }
    ]
  }
}
```

<br>

#### ✅ Valid Code

```html
<ng-template ngFor let-item [ngForOf]="[1, 2, 3]" let-i="index" [ngForTrackByProperty]="trackByFn">
  {{ item }}
</ng-template>
```

</details>

<br>
